{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##  Part 1: Compute mRNA-protein correlation for older tumour studies"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Input:** Transcriptomics and Proteomics data of the tumour studies listed below.                  \n",
    "1. <a href=https://www.nature.com/articles/nature13438>Colorectal Cancer (2014)</a>                   \n",
    "2. <a href=https://www.nature.com/articles/nature18003>Breast Cancer (2016)</a>      \n",
    "3. <a href=https://linkinghub.elsevier.com/retrieve/pii/S0092-8674(16)30673-0>Ovarian Cancer (2016)</a>     \n",
    "4. <a href=https://tinyurl.com/3m8239bk>Colon Cancer (2019)</a>        \n",
    "\n",
    "**Output:** Gene-wise correlation between mRNA and protein abundances "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-info\">\n",
    "<b>Note:</b> \n",
    "<ul>                    \n",
    "<li> The input data are downloaded from the links specified in data_sources.ipynb </li>\n",
    "<li> The transcriptomic and the proteomic data are preprocessed to get a matrix of proteins (rows) and samples (columns)</li>\n",
    "<li> The standardised pipeline is applied before computing the mRNA-protein correlation for each study.</li>\n",
    "</div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Import packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import standardised_pipeline_utils"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "get_data_path = lambda folders, fname: os.path.normpath(os.environ['DATA_PATH']+'/'+'/'.join(folders) +'/'+ fname)\n",
    "get_local_data_path = lambda folders, fname: os.path.normpath('../local_data/'+'/'.join(folders) +'/'+ fname)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### I/O Files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Input Files\n",
    "file_crc_transcriptomics =  get_data_path(['tumour_studies','crc','coadread_tcga_pub'],\n",
    "                                          'data_RNA_Seq_expression_median.txt')\n",
    "file_crc_proteomics = get_data_path(['tumour_studies','crc', 'zhang_2014'], 'Supplemental_Table.xlsx')\n",
    "\n",
    "file_brca_transcriptomics = get_data_path(['tumour_studies','brca', 'brca_tcga_pub2015'], \n",
    "                                          'data_RNA_Seq_v2_expression_median.txt')\n",
    "file_brca_proteomics = get_data_path(['tumour_studies','brca', 'mertins_2016'], 'CPTAC_BC_SupplementaryTable03.xlsx')\n",
    "\n",
    "file_ovca_transcriptomics = get_data_path(['tumour_studies','ovca', 'firebrowse'], 'OV.medianexp.txt')\n",
    "file_ovca_proteomics = get_data_path(['tumour_studies','ovca', 'zhang_2016'], 'Table_S2.xlsx')\n",
    "\n",
    "file_colon_transcriptomics = get_data_path(['tumour_studies','colon', 'vasaikar_2019'], 'RNAseq.cct')\n",
    "file_colon_proteomics = get_data_path(['tumour_studies','colon', 'vasaikar_2019'], 'Proteome_TMT.cct')\n",
    "file_colon_proteomics_lf = get_data_path(['tumour_studies','colon', 'vasaikar_2019'], 'Proteome_LF.cct')\n",
    "\n",
    "                 \n",
    "# Output File\n",
    "file_tumour_correlation = get_local_data_path(['processed', 'correlation_mRNA_protein'], 'older_tumour_studies.csv')\n",
    "file_samples_info = get_local_data_path(['processed', 'correlation_mRNA_protein'], 'samples_info.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Information collected for the Supplemental Table S1B\n",
    "common_samples = {}\n",
    "common_genes = {}\n",
    "transcriptomic_samples = {}\n",
    "proteomic_samples = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Pre-processing the data before applying the standardised pipeline\n",
    "def preprocess(dataframe, index_col, columns_to_drop=[]):\n",
    "    dataframe = dataframe.dropna(subset=[index_col]).set_index(index_col)\n",
    "    dataframe.drop(columns_to_drop, axis=1, inplace=True)\n",
    "    # choose the first 12 characters to identify the patient id\n",
    "    # that is consistent across transcriptomic and proteomic data\n",
    "    dataframe.rename(columns= lambda x: str(x)[:12], inplace=True)\n",
    "    dataframe = dataframe.apply(pd.to_numeric)\n",
    "    print(\"Dimensions: \", dataframe.shape)\n",
    "    print(\"Null values count: \", dataframe.isnull().sum().sum())\n",
    "    return dataframe"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"ColoRectal_Cancer\"></a>\n",
    "### 1. ColoRectal Cancer"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Transcriptomic Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (19489, 246)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>TCGA-A6-2670-01</th>\n",
       "      <th>TCGA-A6-2671-01</th>\n",
       "      <th>TCGA-A6-2672-01</th>\n",
       "      <th>TCGA-A6-2674-01</th>\n",
       "      <th>TCGA-A6-2676-01</th>\n",
       "      <th>TCGA-A6-2677-01</th>\n",
       "      <th>TCGA-A6-2678-01</th>\n",
       "      <th>TCGA-A6-2679-01</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-AY-4071-01</th>\n",
       "      <th>TCGA-AZ-4308-01</th>\n",
       "      <th>TCGA-AZ-4313-01</th>\n",
       "      <th>TCGA-AZ-4315-01</th>\n",
       "      <th>TCGA-AZ-4614-01</th>\n",
       "      <th>TCGA-AZ-4615-01</th>\n",
       "      <th>TCGA-AZ-4684-01</th>\n",
       "      <th>TCGA-CA-5256-01</th>\n",
       "      <th>TCGA-CK-4951-01</th>\n",
       "      <th>TCGA-CM-4747-01</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>1</td>\n",
       "      <td>0.523167</td>\n",
       "      <td>0.371355</td>\n",
       "      <td>0.925661</td>\n",
       "      <td>0.750148</td>\n",
       "      <td>0.524965</td>\n",
       "      <td>0.045138</td>\n",
       "      <td>0.163163</td>\n",
       "      <td>2.267252</td>\n",
       "      <td>...</td>\n",
       "      <td>0.960731</td>\n",
       "      <td>0.988869</td>\n",
       "      <td>0.285177</td>\n",
       "      <td>0.419773</td>\n",
       "      <td>0.134522</td>\n",
       "      <td>0.337758</td>\n",
       "      <td>0.714598</td>\n",
       "      <td>0.808955</td>\n",
       "      <td>0.308311</td>\n",
       "      <td>0.718461</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows Ã— 246 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Hugo_Symbol  Entrez_Gene_Id  TCGA-A6-2670-01  TCGA-A6-2671-01  \\\n",
       "0        A1BG               1         0.523167         0.371355   \n",
       "\n",
       "   TCGA-A6-2672-01  TCGA-A6-2674-01  TCGA-A6-2676-01  TCGA-A6-2677-01  \\\n",
       "0         0.925661         0.750148         0.524965         0.045138   \n",
       "\n",
       "   TCGA-A6-2678-01  TCGA-A6-2679-01  ...  TCGA-AY-4071-01  TCGA-AZ-4308-01  \\\n",
       "0         0.163163         2.267252  ...         0.960731         0.988869   \n",
       "\n",
       "   TCGA-AZ-4313-01  TCGA-AZ-4315-01  TCGA-AZ-4614-01  TCGA-AZ-4615-01  \\\n",
       "0         0.285177         0.419773         0.134522         0.337758   \n",
       "\n",
       "   TCGA-AZ-4684-01  TCGA-CA-5256-01  TCGA-CK-4951-01  TCGA-CM-4747-01  \n",
       "0         0.714598         0.808955         0.308311         0.718461  \n",
       "\n",
       "[1 rows x 246 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# colorectal cancer transcriptomics data\n",
    "crc_transcriptomics = pd.read_csv(file_crc_transcriptomics, sep='\\t',  low_memory=False)\n",
    "print('Dimensions: ', crc_transcriptomics.shape)\n",
    "crc_transcriptomics[:1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (19489, 244)\n",
      "Null values count:  0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TCGA-A6-2670</th>\n",
       "      <th>TCGA-A6-2671</th>\n",
       "      <th>TCGA-A6-2672</th>\n",
       "      <th>TCGA-A6-2674</th>\n",
       "      <th>TCGA-A6-2676</th>\n",
       "      <th>TCGA-A6-2677</th>\n",
       "      <th>TCGA-A6-2678</th>\n",
       "      <th>TCGA-A6-2679</th>\n",
       "      <th>TCGA-A6-2680</th>\n",
       "      <th>TCGA-A6-2681</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-AY-4071</th>\n",
       "      <th>TCGA-AZ-4308</th>\n",
       "      <th>TCGA-AZ-4313</th>\n",
       "      <th>TCGA-AZ-4315</th>\n",
       "      <th>TCGA-AZ-4614</th>\n",
       "      <th>TCGA-AZ-4615</th>\n",
       "      <th>TCGA-AZ-4684</th>\n",
       "      <th>TCGA-CA-5256</th>\n",
       "      <th>TCGA-CK-4951</th>\n",
       "      <th>TCGA-CM-4747</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>0.523167</td>\n",
       "      <td>0.371355</td>\n",
       "      <td>0.925661</td>\n",
       "      <td>0.750148</td>\n",
       "      <td>0.524965</td>\n",
       "      <td>0.045138</td>\n",
       "      <td>0.163163</td>\n",
       "      <td>2.267252</td>\n",
       "      <td>0.556225</td>\n",
       "      <td>0.619890</td>\n",
       "      <td>...</td>\n",
       "      <td>0.960731</td>\n",
       "      <td>0.988869</td>\n",
       "      <td>0.285177</td>\n",
       "      <td>0.419773</td>\n",
       "      <td>0.134522</td>\n",
       "      <td>0.337758</td>\n",
       "      <td>0.714598</td>\n",
       "      <td>0.808955</td>\n",
       "      <td>0.308311</td>\n",
       "      <td>0.718461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>3.269805</td>\n",
       "      <td>0.240150</td>\n",
       "      <td>0.297705</td>\n",
       "      <td>1.956353</td>\n",
       "      <td>1.960370</td>\n",
       "      <td>4.923709</td>\n",
       "      <td>4.173823</td>\n",
       "      <td>1.151729</td>\n",
       "      <td>4.542303</td>\n",
       "      <td>1.670868</td>\n",
       "      <td>...</td>\n",
       "      <td>4.423130</td>\n",
       "      <td>1.619616</td>\n",
       "      <td>0.201777</td>\n",
       "      <td>0.459658</td>\n",
       "      <td>4.037968</td>\n",
       "      <td>0.412526</td>\n",
       "      <td>2.298243</td>\n",
       "      <td>4.482096</td>\n",
       "      <td>0.433493</td>\n",
       "      <td>1.821577</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 244 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             TCGA-A6-2670  TCGA-A6-2671  TCGA-A6-2672  TCGA-A6-2674  \\\n",
       "Hugo_Symbol                                                           \n",
       "A1BG             0.523167      0.371355      0.925661      0.750148   \n",
       "A1CF             3.269805      0.240150      0.297705      1.956353   \n",
       "\n",
       "             TCGA-A6-2676  TCGA-A6-2677  TCGA-A6-2678  TCGA-A6-2679  \\\n",
       "Hugo_Symbol                                                           \n",
       "A1BG             0.524965      0.045138      0.163163      2.267252   \n",
       "A1CF             1.960370      4.923709      4.173823      1.151729   \n",
       "\n",
       "             TCGA-A6-2680  TCGA-A6-2681  ...  TCGA-AY-4071  TCGA-AZ-4308  \\\n",
       "Hugo_Symbol                              ...                               \n",
       "A1BG             0.556225      0.619890  ...      0.960731      0.988869   \n",
       "A1CF             4.542303      1.670868  ...      4.423130      1.619616   \n",
       "\n",
       "             TCGA-AZ-4313  TCGA-AZ-4315  TCGA-AZ-4614  TCGA-AZ-4615  \\\n",
       "Hugo_Symbol                                                           \n",
       "A1BG             0.285177      0.419773      0.134522      0.337758   \n",
       "A1CF             0.201777      0.459658      4.037968      0.412526   \n",
       "\n",
       "             TCGA-AZ-4684  TCGA-CA-5256  TCGA-CK-4951  TCGA-CM-4747  \n",
       "Hugo_Symbol                                                          \n",
       "A1BG             0.714598      0.808955      0.308311      0.718461  \n",
       "A1CF             2.298243      4.482096      0.433493      1.821577  \n",
       "\n",
       "[2 rows x 244 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Clean data\n",
    "crc_transcriptomics = preprocess(crc_transcriptomics, index_col='Hugo_Symbol', columns_to_drop=['Entrez_Gene_Id'])\n",
    "crc_transcriptomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert len(crc_transcriptomics.columns[crc_transcriptomics.columns.duplicated()]) == 0, \"columns contain duplicates\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-warning\">\n",
    "<b>Note:</b> The older transcriptomic and proteomic data such as this contain 0s and no null values. In such cases, the standardised pipeline obtains proteins where the number of 0s is not measured in >20% of the samples.\n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (15961, 244)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TCGA-A6-2670</th>\n",
       "      <th>TCGA-A6-2671</th>\n",
       "      <th>TCGA-A6-2672</th>\n",
       "      <th>TCGA-A6-2674</th>\n",
       "      <th>TCGA-A6-2676</th>\n",
       "      <th>TCGA-A6-2677</th>\n",
       "      <th>TCGA-A6-2678</th>\n",
       "      <th>TCGA-A6-2679</th>\n",
       "      <th>TCGA-A6-2680</th>\n",
       "      <th>TCGA-A6-2681</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-AY-4071</th>\n",
       "      <th>TCGA-AZ-4308</th>\n",
       "      <th>TCGA-AZ-4313</th>\n",
       "      <th>TCGA-AZ-4315</th>\n",
       "      <th>TCGA-AZ-4614</th>\n",
       "      <th>TCGA-AZ-4615</th>\n",
       "      <th>TCGA-AZ-4684</th>\n",
       "      <th>TCGA-CA-5256</th>\n",
       "      <th>TCGA-CK-4951</th>\n",
       "      <th>TCGA-CM-4747</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>0.523167</td>\n",
       "      <td>0.371355</td>\n",
       "      <td>0.925661</td>\n",
       "      <td>0.750148</td>\n",
       "      <td>0.524965</td>\n",
       "      <td>0.045138</td>\n",
       "      <td>0.163163</td>\n",
       "      <td>2.267252</td>\n",
       "      <td>0.556225</td>\n",
       "      <td>0.619890</td>\n",
       "      <td>...</td>\n",
       "      <td>0.960731</td>\n",
       "      <td>0.988869</td>\n",
       "      <td>0.285177</td>\n",
       "      <td>0.419773</td>\n",
       "      <td>0.134522</td>\n",
       "      <td>0.337758</td>\n",
       "      <td>0.714598</td>\n",
       "      <td>0.808955</td>\n",
       "      <td>0.308311</td>\n",
       "      <td>0.718461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>3.269805</td>\n",
       "      <td>0.240150</td>\n",
       "      <td>0.297705</td>\n",
       "      <td>1.956353</td>\n",
       "      <td>1.960370</td>\n",
       "      <td>4.923709</td>\n",
       "      <td>4.173823</td>\n",
       "      <td>1.151729</td>\n",
       "      <td>4.542303</td>\n",
       "      <td>1.670868</td>\n",
       "      <td>...</td>\n",
       "      <td>4.423130</td>\n",
       "      <td>1.619616</td>\n",
       "      <td>0.201777</td>\n",
       "      <td>0.459658</td>\n",
       "      <td>4.037968</td>\n",
       "      <td>0.412526</td>\n",
       "      <td>2.298243</td>\n",
       "      <td>4.482096</td>\n",
       "      <td>0.433493</td>\n",
       "      <td>1.821577</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 244 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             TCGA-A6-2670  TCGA-A6-2671  TCGA-A6-2672  TCGA-A6-2674  \\\n",
       "Hugo_Symbol                                                           \n",
       "A1BG             0.523167      0.371355      0.925661      0.750148   \n",
       "A1CF             3.269805      0.240150      0.297705      1.956353   \n",
       "\n",
       "             TCGA-A6-2676  TCGA-A6-2677  TCGA-A6-2678  TCGA-A6-2679  \\\n",
       "Hugo_Symbol                                                           \n",
       "A1BG             0.524965      0.045138      0.163163      2.267252   \n",
       "A1CF             1.960370      4.923709      4.173823      1.151729   \n",
       "\n",
       "             TCGA-A6-2680  TCGA-A6-2681  ...  TCGA-AY-4071  TCGA-AZ-4308  \\\n",
       "Hugo_Symbol                              ...                               \n",
       "A1BG             0.556225      0.619890  ...      0.960731      0.988869   \n",
       "A1CF             4.542303      1.670868  ...      4.423130      1.619616   \n",
       "\n",
       "             TCGA-AZ-4313  TCGA-AZ-4315  TCGA-AZ-4614  TCGA-AZ-4615  \\\n",
       "Hugo_Symbol                                                           \n",
       "A1BG             0.285177      0.419773      0.134522      0.337758   \n",
       "A1CF             0.201777      0.459658      4.037968      0.412526   \n",
       "\n",
       "             TCGA-AZ-4684  TCGA-CA-5256  TCGA-CK-4951  TCGA-CM-4747  \n",
       "Hugo_Symbol                                                          \n",
       "A1BG             0.714598      0.808955      0.308311      0.718461  \n",
       "A1CF             2.298243      4.482096      0.433493      1.821577  \n",
       "\n",
       "[2 rows x 244 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#eliminate rows with >20% of null/zero values and compute mean for the isoforms \n",
    "crc_transcriptomics_processed = standardised_pipeline_utils.process(crc_transcriptomics)\n",
    "crc_transcriptomics_processed[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# suppress warnings while reading the colorectal proteomics file \n",
    "import warnings \n",
    "warnings.simplefilter(\"ignore\", category=UserWarning)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Proteomic Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (7211, 96)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>TCGA-A6-3807-01A-22</th>\n",
       "      <th>TCGA-A6-3808-01A-22</th>\n",
       "      <th>TCGA-A6-3810-01A-22</th>\n",
       "      <th>TCGA-AA-3518-01A-11</th>\n",
       "      <th>TCGA-AA-3525-01A-12</th>\n",
       "      <th>TCGA-AA-3526-01A-11</th>\n",
       "      <th>TCGA-AA-3529-01A-12</th>\n",
       "      <th>TCGA-AA-3531-01A-22</th>\n",
       "      <th>TCGA-AA-3534-01A-22</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-AG-A01L-01A-22</th>\n",
       "      <th>TCGA-AG-A01N-01A-23</th>\n",
       "      <th>TCGA-AG-A01W-01A-23</th>\n",
       "      <th>TCGA-AG-A01Y-01A-43</th>\n",
       "      <th>TCGA-AG-A020-01A-23</th>\n",
       "      <th>TCGA-AG-A026-01A-71</th>\n",
       "      <th>TCGA-AG-A02N-01A-31</th>\n",
       "      <th>TCGA-AG-A02X-01A-32</th>\n",
       "      <th>TCGA-AG-A032-01A-31</th>\n",
       "      <th>TCGA-AG-A036-01A-22</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>3.168237</td>\n",
       "      <td>4.53212</td>\n",
       "      <td>4.460122</td>\n",
       "      <td>4.638684</td>\n",
       "      <td>3.161463</td>\n",
       "      <td>3.740757</td>\n",
       "      <td>3.250323</td>\n",
       "      <td>3.463567</td>\n",
       "      <td>4.249525</td>\n",
       "      <td>...</td>\n",
       "      <td>3.598817</td>\n",
       "      <td>2.640816</td>\n",
       "      <td>3.68281</td>\n",
       "      <td>4.406709</td>\n",
       "      <td>3.753199</td>\n",
       "      <td>3.735066</td>\n",
       "      <td>3.471804</td>\n",
       "      <td>4.367886</td>\n",
       "      <td>2.514953</td>\n",
       "      <td>4.111383</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows Ã— 96 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  GeneSymbol  TCGA-A6-3807-01A-22  TCGA-A6-3808-01A-22  TCGA-A6-3810-01A-22  \\\n",
       "0       A1BG             3.168237              4.53212             4.460122   \n",
       "\n",
       "   TCGA-AA-3518-01A-11  TCGA-AA-3525-01A-12  TCGA-AA-3526-01A-11  \\\n",
       "0             4.638684             3.161463             3.740757   \n",
       "\n",
       "   TCGA-AA-3529-01A-12  TCGA-AA-3531-01A-22  TCGA-AA-3534-01A-22  ...  \\\n",
       "0             3.250323             3.463567             4.249525  ...   \n",
       "\n",
       "   TCGA-AG-A01L-01A-22  TCGA-AG-A01N-01A-23  TCGA-AG-A01W-01A-23  \\\n",
       "0             3.598817             2.640816              3.68281   \n",
       "\n",
       "   TCGA-AG-A01Y-01A-43  TCGA-AG-A020-01A-23  TCGA-AG-A026-01A-71  \\\n",
       "0             4.406709             3.753199             3.735066   \n",
       "\n",
       "   TCGA-AG-A02N-01A-31  TCGA-AG-A02X-01A-32  TCGA-AG-A032-01A-31  \\\n",
       "0             3.471804             4.367886             2.514953   \n",
       "\n",
       "   TCGA-AG-A036-01A-22  \n",
       "0             4.111383  \n",
       "\n",
       "[1 rows x 96 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# colorectal cancer proteomics data\n",
    "crc_proteomics = pd.read_excel(file_crc_proteomics, sheet_name=\"S4_sample95_count_quantilelog\")\n",
    "print('Dimensions: ', crc_proteomics.shape)\n",
    "crc_proteomics[:1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (7211, 95)\n",
      "Null values count:  0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TCGA-A6-3807</th>\n",
       "      <th>TCGA-A6-3808</th>\n",
       "      <th>TCGA-A6-3810</th>\n",
       "      <th>TCGA-AA-3518</th>\n",
       "      <th>TCGA-AA-3525</th>\n",
       "      <th>TCGA-AA-3526</th>\n",
       "      <th>TCGA-AA-3529</th>\n",
       "      <th>TCGA-AA-3531</th>\n",
       "      <th>TCGA-AA-3534</th>\n",
       "      <th>TCGA-AA-3552</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-AG-A01L</th>\n",
       "      <th>TCGA-AG-A01N</th>\n",
       "      <th>TCGA-AG-A01W</th>\n",
       "      <th>TCGA-AG-A01Y</th>\n",
       "      <th>TCGA-AG-A020</th>\n",
       "      <th>TCGA-AG-A026</th>\n",
       "      <th>TCGA-AG-A02N</th>\n",
       "      <th>TCGA-AG-A02X</th>\n",
       "      <th>TCGA-AG-A032</th>\n",
       "      <th>TCGA-AG-A036</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>3.168237</td>\n",
       "      <td>4.53212</td>\n",
       "      <td>4.460122</td>\n",
       "      <td>4.638684</td>\n",
       "      <td>3.161463</td>\n",
       "      <td>3.740757</td>\n",
       "      <td>3.250323</td>\n",
       "      <td>3.463567</td>\n",
       "      <td>4.249525</td>\n",
       "      <td>3.482712</td>\n",
       "      <td>...</td>\n",
       "      <td>3.598817</td>\n",
       "      <td>2.640816</td>\n",
       "      <td>3.682810</td>\n",
       "      <td>4.406709</td>\n",
       "      <td>3.753199</td>\n",
       "      <td>3.735066</td>\n",
       "      <td>3.471804</td>\n",
       "      <td>4.367886</td>\n",
       "      <td>2.514953</td>\n",
       "      <td>4.111383</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>1.549085</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1.178337</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.889576</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.485427</td>\n",
       "      <td>1.864773</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.992387</td>\n",
       "      <td>2.381429</td>\n",
       "      <td>1.796467</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.297423</td>\n",
       "      <td>2.297423</td>\n",
       "      <td>0.992387</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.144390</td>\n",
       "      <td>2.102570</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 95 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            TCGA-A6-3807  TCGA-A6-3808  TCGA-A6-3810  TCGA-AA-3518  \\\n",
       "GeneSymbol                                                           \n",
       "A1BG            3.168237       4.53212      4.460122      4.638684   \n",
       "A1CF            1.549085       0.00000      1.178337      0.000000   \n",
       "\n",
       "            TCGA-AA-3525  TCGA-AA-3526  TCGA-AA-3529  TCGA-AA-3531  \\\n",
       "GeneSymbol                                                           \n",
       "A1BG            3.161463      3.740757      3.250323      3.463567   \n",
       "A1CF            0.000000      0.889576      0.000000      2.485427   \n",
       "\n",
       "            TCGA-AA-3534  TCGA-AA-3552  ...  TCGA-AG-A01L  TCGA-AG-A01N  \\\n",
       "GeneSymbol                              ...                               \n",
       "A1BG            4.249525      3.482712  ...      3.598817      2.640816   \n",
       "A1CF            1.864773      0.000000  ...      0.992387      2.381429   \n",
       "\n",
       "            TCGA-AG-A01W  TCGA-AG-A01Y  TCGA-AG-A020  TCGA-AG-A026  \\\n",
       "GeneSymbol                                                           \n",
       "A1BG            3.682810      4.406709      3.753199      3.735066   \n",
       "A1CF            1.796467      0.000000      2.297423      2.297423   \n",
       "\n",
       "            TCGA-AG-A02N  TCGA-AG-A02X  TCGA-AG-A032  TCGA-AG-A036  \n",
       "GeneSymbol                                                          \n",
       "A1BG            3.471804      4.367886      2.514953      4.111383  \n",
       "A1CF            0.992387      0.000000      1.144390      2.102570  \n",
       "\n",
       "[2 rows x 95 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Clean data\n",
    "crc_proteomics = preprocess(crc_proteomics, index_col='GeneSymbol')\n",
    "crc_proteomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert len(crc_proteomics.columns[crc_proteomics.columns.duplicated()]) == 5, \"columns contain no duplicates\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-warning\">\n",
    "<b>Note:</b> Since the Colorectal cancer proteomics data contains 5 duplicated sample IDs, drop the duplicated sample IDs. The same procedure is followed for the other studies too in this context. \n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "crc_proteomics = crc_proteomics.loc[:, ~crc_proteomics.columns.duplicated(keep='first')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (3506, 90)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TCGA-A6-3807</th>\n",
       "      <th>TCGA-A6-3808</th>\n",
       "      <th>TCGA-A6-3810</th>\n",
       "      <th>TCGA-AA-3518</th>\n",
       "      <th>TCGA-AA-3525</th>\n",
       "      <th>TCGA-AA-3526</th>\n",
       "      <th>TCGA-AA-3529</th>\n",
       "      <th>TCGA-AA-3531</th>\n",
       "      <th>TCGA-AA-3534</th>\n",
       "      <th>TCGA-AA-3552</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-AG-A01L</th>\n",
       "      <th>TCGA-AG-A01N</th>\n",
       "      <th>TCGA-AG-A01W</th>\n",
       "      <th>TCGA-AG-A01Y</th>\n",
       "      <th>TCGA-AG-A020</th>\n",
       "      <th>TCGA-AG-A026</th>\n",
       "      <th>TCGA-AG-A02N</th>\n",
       "      <th>TCGA-AG-A02X</th>\n",
       "      <th>TCGA-AG-A032</th>\n",
       "      <th>TCGA-AG-A036</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>3.168237</td>\n",
       "      <td>4.53212</td>\n",
       "      <td>4.460122</td>\n",
       "      <td>4.638684</td>\n",
       "      <td>3.161463</td>\n",
       "      <td>3.740757</td>\n",
       "      <td>3.250323</td>\n",
       "      <td>3.463567</td>\n",
       "      <td>4.249525</td>\n",
       "      <td>3.482712</td>\n",
       "      <td>...</td>\n",
       "      <td>3.598817</td>\n",
       "      <td>2.640816</td>\n",
       "      <td>3.682810</td>\n",
       "      <td>4.406709</td>\n",
       "      <td>3.753199</td>\n",
       "      <td>3.735066</td>\n",
       "      <td>3.471804</td>\n",
       "      <td>4.367886</td>\n",
       "      <td>2.514953</td>\n",
       "      <td>4.111383</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>7.194187</td>\n",
       "      <td>7.74494</td>\n",
       "      <td>6.507127</td>\n",
       "      <td>7.973176</td>\n",
       "      <td>6.890216</td>\n",
       "      <td>6.481183</td>\n",
       "      <td>6.177288</td>\n",
       "      <td>6.364757</td>\n",
       "      <td>7.194187</td>\n",
       "      <td>6.616646</td>\n",
       "      <td>...</td>\n",
       "      <td>6.219169</td>\n",
       "      <td>5.679850</td>\n",
       "      <td>6.976798</td>\n",
       "      <td>6.289679</td>\n",
       "      <td>6.815130</td>\n",
       "      <td>5.924375</td>\n",
       "      <td>5.876065</td>\n",
       "      <td>6.704523</td>\n",
       "      <td>6.574803</td>\n",
       "      <td>6.161251</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 90 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            TCGA-A6-3807  TCGA-A6-3808  TCGA-A6-3810  TCGA-AA-3518  \\\n",
       "GeneSymbol                                                           \n",
       "A1BG            3.168237       4.53212      4.460122      4.638684   \n",
       "A2M             7.194187       7.74494      6.507127      7.973176   \n",
       "\n",
       "            TCGA-AA-3525  TCGA-AA-3526  TCGA-AA-3529  TCGA-AA-3531  \\\n",
       "GeneSymbol                                                           \n",
       "A1BG            3.161463      3.740757      3.250323      3.463567   \n",
       "A2M             6.890216      6.481183      6.177288      6.364757   \n",
       "\n",
       "            TCGA-AA-3534  TCGA-AA-3552  ...  TCGA-AG-A01L  TCGA-AG-A01N  \\\n",
       "GeneSymbol                              ...                               \n",
       "A1BG            4.249525      3.482712  ...      3.598817      2.640816   \n",
       "A2M             7.194187      6.616646  ...      6.219169      5.679850   \n",
       "\n",
       "            TCGA-AG-A01W  TCGA-AG-A01Y  TCGA-AG-A020  TCGA-AG-A026  \\\n",
       "GeneSymbol                                                           \n",
       "A1BG            3.682810      4.406709      3.753199      3.735066   \n",
       "A2M             6.976798      6.289679      6.815130      5.924375   \n",
       "\n",
       "            TCGA-AG-A02N  TCGA-AG-A02X  TCGA-AG-A032  TCGA-AG-A036  \n",
       "GeneSymbol                                                          \n",
       "A1BG            3.471804      4.367886      2.514953      4.111383  \n",
       "A2M             5.876065      6.704523      6.574803      6.161251  \n",
       "\n",
       "[2 rows x 90 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#eliminate rows with >20% of null/zero values\n",
    "crc_proteomics_processed = standardised_pipeline_utils.process(crc_proteomics)\n",
    "crc_proteomics_processed[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "transcriptomic_samples['CRC (2014)'] = crc_transcriptomics_processed.shape[1]\n",
    "proteomic_samples['CRC (2014)'] = crc_proteomics_processed.shape[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of common proteins:  3238\n",
      "Number of common samples:  77\n"
     ]
    }
   ],
   "source": [
    "crc_transcriptomics_processed, crc_proteomics_processed = standardised_pipeline_utils.match_proteins_samples( \\\n",
    "                                                                            crc_transcriptomics_processed, \n",
    "                                                                            crc_proteomics_processed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "common_samples['CRC (2014)'] = crc_transcriptomics_processed.shape[1]\n",
    "common_genes['CRC (2014)'] = crc_transcriptomics_processed.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Spearman Correlation:  0.22\n"
     ]
    }
   ],
   "source": [
    "correlation_crc = standardised_pipeline_utils.correlate_genewise(crc_transcriptomics_processed, \n",
    "                                                                 crc_proteomics_processed, 'CRC (2014)')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Pearson Correlation:  0.2233\n"
     ]
    }
   ],
   "source": [
    "correlation_crc_pearson = standardised_pipeline_utils.correlate_genewise(crc_transcriptomics_processed, \n",
    "                                                                 crc_proteomics_processed, 'CRC (2014)', method='pearson')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"Breast_Cancer\"></a>\n",
    "### 2. Breast Cancer (2016)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Transcriptomic Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (20440, 819)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>TCGA-A1-A0SB-01</th>\n",
       "      <th>TCGA-A1-A0SD-01</th>\n",
       "      <th>TCGA-A1-A0SE-01</th>\n",
       "      <th>TCGA-A1-A0SF-01</th>\n",
       "      <th>TCGA-A1-A0SH-01</th>\n",
       "      <th>TCGA-A1-A0SI-01</th>\n",
       "      <th>TCGA-A1-A0SJ-01</th>\n",
       "      <th>TCGA-A1-A0SK-01</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-LL-A5YM-01</th>\n",
       "      <th>TCGA-LL-A5YN-01</th>\n",
       "      <th>TCGA-LL-A5YO-01</th>\n",
       "      <th>TCGA-LL-A5YP-01</th>\n",
       "      <th>TCGA-LQ-A4E4-01</th>\n",
       "      <th>TCGA-MS-A51U-01</th>\n",
       "      <th>TCGA-OL-A66H-01</th>\n",
       "      <th>TCGA-OL-A66I-01</th>\n",
       "      <th>TCGA-OL-A66J-01</th>\n",
       "      <th>TCGA-OL-A66K-01</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>UBE2Q2P2</td>\n",
       "      <td>100134869</td>\n",
       "      <td>14.3935</td>\n",
       "      <td>11.3241</td>\n",
       "      <td>4.4426</td>\n",
       "      <td>10.7401</td>\n",
       "      <td>3.0048</td>\n",
       "      <td>2.9782</td>\n",
       "      <td>4.9419</td>\n",
       "      <td>28.856</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.9624</td>\n",
       "      <td>3.6899</td>\n",
       "      <td>6.302</td>\n",
       "      <td>14.1288</td>\n",
       "      <td>7.9343</td>\n",
       "      <td>2.2519</td>\n",
       "      <td>1.2603</td>\n",
       "      <td>5.0428</td>\n",
       "      <td>4.3892</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows Ã— 819 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Hugo_Symbol  Entrez_Gene_Id  TCGA-A1-A0SB-01  TCGA-A1-A0SD-01  \\\n",
       "0    UBE2Q2P2       100134869          14.3935          11.3241   \n",
       "\n",
       "   TCGA-A1-A0SE-01  TCGA-A1-A0SF-01  TCGA-A1-A0SH-01  TCGA-A1-A0SI-01  \\\n",
       "0           4.4426          10.7401           3.0048           2.9782   \n",
       "\n",
       "   TCGA-A1-A0SJ-01  TCGA-A1-A0SK-01  ...  TCGA-LL-A5YM-01  TCGA-LL-A5YN-01  \\\n",
       "0           4.9419           28.856  ...              0.0           2.9624   \n",
       "\n",
       "   TCGA-LL-A5YO-01  TCGA-LL-A5YP-01  TCGA-LQ-A4E4-01  TCGA-MS-A51U-01  \\\n",
       "0           3.6899            6.302          14.1288           7.9343   \n",
       "\n",
       "   TCGA-OL-A66H-01  TCGA-OL-A66I-01  TCGA-OL-A66J-01  TCGA-OL-A66K-01  \n",
       "0           2.2519           1.2603           5.0428           4.3892  \n",
       "\n",
       "[1 rows x 819 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca_transcriptomics = pd.read_csv(file_brca_transcriptomics, sep='\\t')\n",
    "print('Dimensions: ', brca_transcriptomics.shape)\n",
    "brca_transcriptomics[:1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (20440, 817)\n",
      "Null values count:  0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TCGA-A1-A0SB</th>\n",
       "      <th>TCGA-A1-A0SD</th>\n",
       "      <th>TCGA-A1-A0SE</th>\n",
       "      <th>TCGA-A1-A0SF</th>\n",
       "      <th>TCGA-A1-A0SH</th>\n",
       "      <th>TCGA-A1-A0SI</th>\n",
       "      <th>TCGA-A1-A0SJ</th>\n",
       "      <th>TCGA-A1-A0SK</th>\n",
       "      <th>TCGA-A1-A0SM</th>\n",
       "      <th>TCGA-A1-A0SN</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-LL-A5YM</th>\n",
       "      <th>TCGA-LL-A5YN</th>\n",
       "      <th>TCGA-LL-A5YO</th>\n",
       "      <th>TCGA-LL-A5YP</th>\n",
       "      <th>TCGA-LQ-A4E4</th>\n",
       "      <th>TCGA-MS-A51U</th>\n",
       "      <th>TCGA-OL-A66H</th>\n",
       "      <th>TCGA-OL-A66I</th>\n",
       "      <th>TCGA-OL-A66J</th>\n",
       "      <th>TCGA-OL-A66K</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>UBE2Q2P2</th>\n",
       "      <td>14.3935</td>\n",
       "      <td>11.3241</td>\n",
       "      <td>4.4426</td>\n",
       "      <td>10.7401</td>\n",
       "      <td>3.0048</td>\n",
       "      <td>2.9782</td>\n",
       "      <td>4.9419</td>\n",
       "      <td>28.8560</td>\n",
       "      <td>7.6484</td>\n",
       "      <td>5.6992</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.9624</td>\n",
       "      <td>3.6899</td>\n",
       "      <td>6.3020</td>\n",
       "      <td>14.1288</td>\n",
       "      <td>7.9343</td>\n",
       "      <td>2.2519</td>\n",
       "      <td>1.2603</td>\n",
       "      <td>5.0428</td>\n",
       "      <td>4.3892</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HMGB1P1</th>\n",
       "      <td>116.3870</td>\n",
       "      <td>60.2630</td>\n",
       "      <td>153.1452</td>\n",
       "      <td>141.1933</td>\n",
       "      <td>79.8003</td>\n",
       "      <td>63.5491</td>\n",
       "      <td>134.8733</td>\n",
       "      <td>1119.1932</td>\n",
       "      <td>119.4760</td>\n",
       "      <td>95.0898</td>\n",
       "      <td>...</td>\n",
       "      <td>101.2865</td>\n",
       "      <td>100.3083</td>\n",
       "      <td>278.5626</td>\n",
       "      <td>206.4376</td>\n",
       "      <td>117.0300</td>\n",
       "      <td>150.6834</td>\n",
       "      <td>115.3378</td>\n",
       "      <td>158.3599</td>\n",
       "      <td>124.6327</td>\n",
       "      <td>106.3475</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 817 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             TCGA-A1-A0SB  TCGA-A1-A0SD  TCGA-A1-A0SE  TCGA-A1-A0SF  \\\n",
       "Hugo_Symbol                                                           \n",
       "UBE2Q2P2          14.3935       11.3241        4.4426       10.7401   \n",
       "HMGB1P1          116.3870       60.2630      153.1452      141.1933   \n",
       "\n",
       "             TCGA-A1-A0SH  TCGA-A1-A0SI  TCGA-A1-A0SJ  TCGA-A1-A0SK  \\\n",
       "Hugo_Symbol                                                           \n",
       "UBE2Q2P2           3.0048        2.9782        4.9419       28.8560   \n",
       "HMGB1P1           79.8003       63.5491      134.8733     1119.1932   \n",
       "\n",
       "             TCGA-A1-A0SM  TCGA-A1-A0SN  ...  TCGA-LL-A5YM  TCGA-LL-A5YN  \\\n",
       "Hugo_Symbol                              ...                               \n",
       "UBE2Q2P2           7.6484        5.6992  ...        0.0000        2.9624   \n",
       "HMGB1P1          119.4760       95.0898  ...      101.2865      100.3083   \n",
       "\n",
       "             TCGA-LL-A5YO  TCGA-LL-A5YP  TCGA-LQ-A4E4  TCGA-MS-A51U  \\\n",
       "Hugo_Symbol                                                           \n",
       "UBE2Q2P2           3.6899        6.3020       14.1288        7.9343   \n",
       "HMGB1P1          278.5626      206.4376      117.0300      150.6834   \n",
       "\n",
       "             TCGA-OL-A66H  TCGA-OL-A66I  TCGA-OL-A66J  TCGA-OL-A66K  \n",
       "Hugo_Symbol                                                          \n",
       "UBE2Q2P2           2.2519        1.2603        5.0428        4.3892  \n",
       "HMGB1P1          115.3378      158.3599      124.6327      106.3475  \n",
       "\n",
       "[2 rows x 817 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Clean data\n",
    "brca_transcriptomics = preprocess(brca_transcriptomics, index_col='Hugo_Symbol', columns_to_drop=['Entrez_Gene_Id'])\n",
    "brca_transcriptomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert len(brca_transcriptomics.columns[brca_transcriptomics.columns.duplicated()]) == 0, \\\n",
    "       \"columns contain duplicates\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (16558, 817)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TCGA-A1-A0SB</th>\n",
       "      <th>TCGA-A1-A0SD</th>\n",
       "      <th>TCGA-A1-A0SE</th>\n",
       "      <th>TCGA-A1-A0SF</th>\n",
       "      <th>TCGA-A1-A0SH</th>\n",
       "      <th>TCGA-A1-A0SI</th>\n",
       "      <th>TCGA-A1-A0SJ</th>\n",
       "      <th>TCGA-A1-A0SK</th>\n",
       "      <th>TCGA-A1-A0SM</th>\n",
       "      <th>TCGA-A1-A0SN</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-LL-A5YM</th>\n",
       "      <th>TCGA-LL-A5YN</th>\n",
       "      <th>TCGA-LL-A5YO</th>\n",
       "      <th>TCGA-LL-A5YP</th>\n",
       "      <th>TCGA-LQ-A4E4</th>\n",
       "      <th>TCGA-MS-A51U</th>\n",
       "      <th>TCGA-OL-A66H</th>\n",
       "      <th>TCGA-OL-A66I</th>\n",
       "      <th>TCGA-OL-A66J</th>\n",
       "      <th>TCGA-OL-A66K</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>49.1992</td>\n",
       "      <td>142.2976</td>\n",
       "      <td>192.8194</td>\n",
       "      <td>326.0194</td>\n",
       "      <td>180.3235</td>\n",
       "      <td>128.4550</td>\n",
       "      <td>226.7801</td>\n",
       "      <td>86.6436</td>\n",
       "      <td>307.4937</td>\n",
       "      <td>87.5786</td>\n",
       "      <td>...</td>\n",
       "      <td>1507.1099</td>\n",
       "      <td>452.4033</td>\n",
       "      <td>295.5528</td>\n",
       "      <td>136.0353</td>\n",
       "      <td>498.7862</td>\n",
       "      <td>362.5874</td>\n",
       "      <td>51.2469</td>\n",
       "      <td>207.2697</td>\n",
       "      <td>267.0482</td>\n",
       "      <td>422.2195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <td>75.0039</td>\n",
       "      <td>143.6440</td>\n",
       "      <td>148.6170</td>\n",
       "      <td>187.2980</td>\n",
       "      <td>157.7045</td>\n",
       "      <td>147.4985</td>\n",
       "      <td>115.2922</td>\n",
       "      <td>65.2821</td>\n",
       "      <td>121.0795</td>\n",
       "      <td>66.0327</td>\n",
       "      <td>...</td>\n",
       "      <td>317.9009</td>\n",
       "      <td>352.0115</td>\n",
       "      <td>159.2180</td>\n",
       "      <td>63.2459</td>\n",
       "      <td>250.2883</td>\n",
       "      <td>291.5545</td>\n",
       "      <td>54.4412</td>\n",
       "      <td>167.8665</td>\n",
       "      <td>174.9752</td>\n",
       "      <td>227.4997</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 817 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             TCGA-A1-A0SB  TCGA-A1-A0SD  TCGA-A1-A0SE  TCGA-A1-A0SF  \\\n",
       "Hugo_Symbol                                                           \n",
       "A1BG              49.1992      142.2976      192.8194      326.0194   \n",
       "A1BG-AS1          75.0039      143.6440      148.6170      187.2980   \n",
       "\n",
       "             TCGA-A1-A0SH  TCGA-A1-A0SI  TCGA-A1-A0SJ  TCGA-A1-A0SK  \\\n",
       "Hugo_Symbol                                                           \n",
       "A1BG             180.3235      128.4550      226.7801       86.6436   \n",
       "A1BG-AS1         157.7045      147.4985      115.2922       65.2821   \n",
       "\n",
       "             TCGA-A1-A0SM  TCGA-A1-A0SN  ...  TCGA-LL-A5YM  TCGA-LL-A5YN  \\\n",
       "Hugo_Symbol                              ...                               \n",
       "A1BG             307.4937       87.5786  ...     1507.1099      452.4033   \n",
       "A1BG-AS1         121.0795       66.0327  ...      317.9009      352.0115   \n",
       "\n",
       "             TCGA-LL-A5YO  TCGA-LL-A5YP  TCGA-LQ-A4E4  TCGA-MS-A51U  \\\n",
       "Hugo_Symbol                                                           \n",
       "A1BG             295.5528      136.0353      498.7862      362.5874   \n",
       "A1BG-AS1         159.2180       63.2459      250.2883      291.5545   \n",
       "\n",
       "             TCGA-OL-A66H  TCGA-OL-A66I  TCGA-OL-A66J  TCGA-OL-A66K  \n",
       "Hugo_Symbol                                                          \n",
       "A1BG              51.2469      207.2697      267.0482      422.2195  \n",
       "A1BG-AS1          54.4412      167.8665      174.9752      227.4997  \n",
       "\n",
       "[2 rows x 817 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca_transcriptomics_processed = standardised_pipeline_utils.process(brca_transcriptomics)\n",
    "brca_transcriptomics_processed[:2]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Proteomic Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (12553, 95)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_number</th>\n",
       "      <th>numITRAQExperimentsProteinObserved</th>\n",
       "      <th>numSpectraProteinObserved</th>\n",
       "      <th>protein_mw</th>\n",
       "      <th>species</th>\n",
       "      <th>accession_numbers</th>\n",
       "      <th>numPepsUnique</th>\n",
       "      <th>scoreUnique</th>\n",
       "      <th>total percent Coverage across all experiments</th>\n",
       "      <th>subgroupNum</th>\n",
       "      <th>...</th>\n",
       "      <th>AO-A12B.34TCGA</th>\n",
       "      <th>A2-A0SW.35TCGA</th>\n",
       "      <th>AO-A0JL.35TCGA</th>\n",
       "      <th>BH-A0BV.35TCGA</th>\n",
       "      <th>A2-A0YM.36TCGA</th>\n",
       "      <th>BH-A0C7.36TCGA</th>\n",
       "      <th>A2-A0SX.36TCGA</th>\n",
       "      <th>263d3f-I.CPTAC</th>\n",
       "      <th>blcdb9-I.CPTAC</th>\n",
       "      <th>c4155b-C.CPTAC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NP_958782</td>\n",
       "      <td>37</td>\n",
       "      <td>43425</td>\n",
       "      <td>533778.0</td>\n",
       "      <td>Human</td>\n",
       "      <td>NP_958782</td>\n",
       "      <td>678</td>\n",
       "      <td>11158.42</td>\n",
       "      <td>89.1</td>\n",
       "      <td>1.1</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.963904</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.065838</td>\n",
       "      <td>0.655850</td>\n",
       "      <td>-0.552212</td>\n",
       "      <td>-0.398560</td>\n",
       "      <td>0.598585</td>\n",
       "      <td>-0.191285</td>\n",
       "      <td>0.566975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NP_958785</td>\n",
       "      <td>37</td>\n",
       "      <td>42994</td>\n",
       "      <td>518637.5</td>\n",
       "      <td>Human</td>\n",
       "      <td>NP_958785</td>\n",
       "      <td>670</td>\n",
       "      <td>11042.11</td>\n",
       "      <td>90.0</td>\n",
       "      <td>1.2</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.938210</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.055893</td>\n",
       "      <td>0.658143</td>\n",
       "      <td>-0.547749</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.606697</td>\n",
       "      <td>-0.183918</td>\n",
       "      <td>0.578702</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 95 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  accession_number  numITRAQExperimentsProteinObserved  \\\n",
       "0        NP_958782                                  37   \n",
       "1        NP_958785                                  37   \n",
       "\n",
       "   numSpectraProteinObserved  protein_mw species accession_numbers  \\\n",
       "0                      43425    533778.0   Human         NP_958782   \n",
       "1                      42994    518637.5   Human         NP_958785   \n",
       "\n",
       "   numPepsUnique  scoreUnique  total percent Coverage across all experiments  \\\n",
       "0            678     11158.42                                           89.1   \n",
       "1            670     11042.11                                           90.0   \n",
       "\n",
       "   subgroupNum  ... AO-A12B.34TCGA A2-A0SW.35TCGA  AO-A0JL.35TCGA  \\\n",
       "0          1.1  ...      -0.963904      -0.487772        -0.10668   \n",
       "1          1.2  ...      -0.938210      -0.487772        -0.10668   \n",
       "\n",
       "   BH-A0BV.35TCGA  A2-A0YM.36TCGA  BH-A0C7.36TCGA  A2-A0SX.36TCGA  \\\n",
       "0       -0.065838        0.655850       -0.552212       -0.398560   \n",
       "1       -0.055893        0.658143       -0.547749       -0.392601   \n",
       "\n",
       "   263d3f-I.CPTAC  blcdb9-I.CPTAC  c4155b-C.CPTAC  \n",
       "0        0.598585       -0.191285        0.566975  \n",
       "1        0.606697       -0.183918        0.578702  \n",
       "\n",
       "[2 rows x 95 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca_proteomics = pd.read_excel(io=file_brca_proteomics, sheet_name=\"Global-Proteome-G3\")\n",
    "print('Dimensions: ', brca_proteomics.shape)\n",
    "brca_proteomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>geneName</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>plectin isoform 1</td>\n",
       "      <td>PLEC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>plectin isoform 1g</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>plectin isoform 1a</td>\n",
       "      <td>PLEC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Name geneName\n",
       "0   plectin isoform 1     PLEC\n",
       "1  plectin isoform 1g      NaN\n",
       "2  plectin isoform 1a     PLEC"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca_proteomics[['Name', 'geneName']].head(3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-warning\">\n",
    "<b>Note:</b> Since some proteins contain NaN in the geneName column although other isoforms of it exists with correct geneName, proteomics data needs to be processed to fill in the NaN values with the appropriate geneName\n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Null values count:  80573\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TCGA-A2-A0CM</th>\n",
       "      <th>TCGA-A2-A0D2</th>\n",
       "      <th>TCGA-A2-A0EQ</th>\n",
       "      <th>TCGA-A2-A0EV</th>\n",
       "      <th>TCGA-A2-A0EX</th>\n",
       "      <th>TCGA-A2-A0EY</th>\n",
       "      <th>TCGA-A2-A0SW</th>\n",
       "      <th>TCGA-A2-A0SX</th>\n",
       "      <th>TCGA-A2-A0T3</th>\n",
       "      <th>TCGA-A2-A0T6</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-C8-A12Z</th>\n",
       "      <th>TCGA-C8-A130</th>\n",
       "      <th>TCGA-C8-A131</th>\n",
       "      <th>TCGA-C8-A134</th>\n",
       "      <th>TCGA-C8-A135</th>\n",
       "      <th>TCGA-C8-A138</th>\n",
       "      <th>TCGA-D8-A142</th>\n",
       "      <th>TCGA-E2-A154</th>\n",
       "      <th>TCGA-E2-A158</th>\n",
       "      <th>TCGA-E2-A15A</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>geneName</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-0.899882</td>\n",
       "      <td>-0.105400</td>\n",
       "      <td>-0.380704</td>\n",
       "      <td>1.699253</td>\n",
       "      <td>-0.390673</td>\n",
       "      <td>1.849433</td>\n",
       "      <td>1.560915</td>\n",
       "      <td>2.369127</td>\n",
       "      <td>-0.003410</td>\n",
       "      <td>2.519514</td>\n",
       "      <td>...</td>\n",
       "      <td>1.775668</td>\n",
       "      <td>0.850687</td>\n",
       "      <td>-0.294419</td>\n",
       "      <td>0.320314</td>\n",
       "      <td>2.185114</td>\n",
       "      <td>-2.018457</td>\n",
       "      <td>1.539879</td>\n",
       "      <td>1.856193</td>\n",
       "      <td>-1.987341</td>\n",
       "      <td>-1.277926</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2BP1</th>\n",
       "      <td>0.911162</td>\n",
       "      <td>0.882545</td>\n",
       "      <td>-0.874399</td>\n",
       "      <td>-2.084358</td>\n",
       "      <td>1.796662</td>\n",
       "      <td>-0.578121</td>\n",
       "      <td>1.506240</td>\n",
       "      <td>-0.520707</td>\n",
       "      <td>0.120195</td>\n",
       "      <td>-1.671573</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.859222</td>\n",
       "      <td>1.008120</td>\n",
       "      <td>0.737591</td>\n",
       "      <td>1.945039</td>\n",
       "      <td>-2.473849</td>\n",
       "      <td>0.329426</td>\n",
       "      <td>0.957906</td>\n",
       "      <td>0.520191</td>\n",
       "      <td>0.291579</td>\n",
       "      <td>-2.101058</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 77 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          TCGA-A2-A0CM  TCGA-A2-A0D2  TCGA-A2-A0EQ  TCGA-A2-A0EV  \\\n",
       "geneName                                                           \n",
       "A1BG         -0.899882     -0.105400     -0.380704      1.699253   \n",
       "A2BP1         0.911162      0.882545     -0.874399     -2.084358   \n",
       "\n",
       "          TCGA-A2-A0EX  TCGA-A2-A0EY  TCGA-A2-A0SW  TCGA-A2-A0SX  \\\n",
       "geneName                                                           \n",
       "A1BG         -0.390673      1.849433      1.560915      2.369127   \n",
       "A2BP1         1.796662     -0.578121      1.506240     -0.520707   \n",
       "\n",
       "          TCGA-A2-A0T3  TCGA-A2-A0T6  ...  TCGA-C8-A12Z  TCGA-C8-A130  \\\n",
       "geneName                              ...                               \n",
       "A1BG         -0.003410      2.519514  ...      1.775668      0.850687   \n",
       "A2BP1         0.120195     -1.671573  ...     -1.859222      1.008120   \n",
       "\n",
       "          TCGA-C8-A131  TCGA-C8-A134  TCGA-C8-A135  TCGA-C8-A138  \\\n",
       "geneName                                                           \n",
       "A1BG         -0.294419      0.320314      2.185114     -2.018457   \n",
       "A2BP1         0.737591      1.945039     -2.473849      0.329426   \n",
       "\n",
       "          TCGA-D8-A142  TCGA-E2-A154  TCGA-E2-A158  TCGA-E2-A15A  \n",
       "geneName                                                          \n",
       "A1BG          1.539879      1.856193     -1.987341     -1.277926  \n",
       "A2BP1         0.957906      0.520191      0.291579     -2.101058  \n",
       "\n",
       "[2 rows x 77 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#split based on the name and then identify the right geneNames for each protein record\n",
    "brca_proteomics.loc[:, 'modifiedName'] = brca_proteomics['Name'].str.split('isoform').str[0] \n",
    "\n",
    "brca_proteomics['geneName'] = brca_proteomics.groupby(['modifiedName'])['geneName'] \\\n",
    "                                             .apply(lambda x: x.ffill().bfill())\n",
    "\n",
    "# #Even after filling if there exists NaN values for gene - we drop them\n",
    "brca_proteomics.dropna(subset=['geneName'], inplace=True)\n",
    "brca_proteomics = brca_proteomics.groupby(['geneName']).mean()\n",
    "\n",
    "# Modifying sample names to suit the samples in transcriptomic data\n",
    "brca_proteomics.columns = [i.replace('.', '-') for i in brca_proteomics.columns]\n",
    "brca_proteomics = brca_proteomics.loc[:, brca_proteomics.columns.str.contains('TCGA')]\n",
    "brca_proteomics = brca_proteomics.rename(columns = lambda x : 'TCGA-' + str(x)[:7])\n",
    "# Aggregate the duplicated columns present\n",
    "brca_proteomics = brca_proteomics.groupby(brca_proteomics.columns, axis=1).mean()\n",
    "\n",
    "print(\"Null values count: \", brca_proteomics.isnull().sum().sum())\n",
    "brca_proteomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert len(brca_proteomics.columns[brca_proteomics.columns.duplicated()]) == 0, \\\n",
    "       \"columns contain duplicates\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (8052, 77)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TCGA-A2-A0CM</th>\n",
       "      <th>TCGA-A2-A0D2</th>\n",
       "      <th>TCGA-A2-A0EQ</th>\n",
       "      <th>TCGA-A2-A0EV</th>\n",
       "      <th>TCGA-A2-A0EX</th>\n",
       "      <th>TCGA-A2-A0EY</th>\n",
       "      <th>TCGA-A2-A0SW</th>\n",
       "      <th>TCGA-A2-A0SX</th>\n",
       "      <th>TCGA-A2-A0T3</th>\n",
       "      <th>TCGA-A2-A0T6</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-C8-A12Z</th>\n",
       "      <th>TCGA-C8-A130</th>\n",
       "      <th>TCGA-C8-A131</th>\n",
       "      <th>TCGA-C8-A134</th>\n",
       "      <th>TCGA-C8-A135</th>\n",
       "      <th>TCGA-C8-A138</th>\n",
       "      <th>TCGA-D8-A142</th>\n",
       "      <th>TCGA-E2-A154</th>\n",
       "      <th>TCGA-E2-A158</th>\n",
       "      <th>TCGA-E2-A15A</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>geneName</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-0.899882</td>\n",
       "      <td>-0.105400</td>\n",
       "      <td>-0.380704</td>\n",
       "      <td>1.699253</td>\n",
       "      <td>-0.390673</td>\n",
       "      <td>1.849433</td>\n",
       "      <td>1.560915</td>\n",
       "      <td>2.369127</td>\n",
       "      <td>-0.003410</td>\n",
       "      <td>2.519514</td>\n",
       "      <td>...</td>\n",
       "      <td>1.775668</td>\n",
       "      <td>0.850687</td>\n",
       "      <td>-0.294419</td>\n",
       "      <td>0.320314</td>\n",
       "      <td>2.185114</td>\n",
       "      <td>-2.018457</td>\n",
       "      <td>1.539879</td>\n",
       "      <td>1.856193</td>\n",
       "      <td>-1.987341</td>\n",
       "      <td>-1.277926</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2BP1</th>\n",
       "      <td>0.911162</td>\n",
       "      <td>0.882545</td>\n",
       "      <td>-0.874399</td>\n",
       "      <td>-2.084358</td>\n",
       "      <td>1.796662</td>\n",
       "      <td>-0.578121</td>\n",
       "      <td>1.506240</td>\n",
       "      <td>-0.520707</td>\n",
       "      <td>0.120195</td>\n",
       "      <td>-1.671573</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.859222</td>\n",
       "      <td>1.008120</td>\n",
       "      <td>0.737591</td>\n",
       "      <td>1.945039</td>\n",
       "      <td>-2.473849</td>\n",
       "      <td>0.329426</td>\n",
       "      <td>0.957906</td>\n",
       "      <td>0.520191</td>\n",
       "      <td>0.291579</td>\n",
       "      <td>-2.101058</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 77 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          TCGA-A2-A0CM  TCGA-A2-A0D2  TCGA-A2-A0EQ  TCGA-A2-A0EV  \\\n",
       "geneName                                                           \n",
       "A1BG         -0.899882     -0.105400     -0.380704      1.699253   \n",
       "A2BP1         0.911162      0.882545     -0.874399     -2.084358   \n",
       "\n",
       "          TCGA-A2-A0EX  TCGA-A2-A0EY  TCGA-A2-A0SW  TCGA-A2-A0SX  \\\n",
       "geneName                                                           \n",
       "A1BG         -0.390673      1.849433      1.560915      2.369127   \n",
       "A2BP1         1.796662     -0.578121      1.506240     -0.520707   \n",
       "\n",
       "          TCGA-A2-A0T3  TCGA-A2-A0T6  ...  TCGA-C8-A12Z  TCGA-C8-A130  \\\n",
       "geneName                              ...                               \n",
       "A1BG         -0.003410      2.519514  ...      1.775668      0.850687   \n",
       "A2BP1         0.120195     -1.671573  ...     -1.859222      1.008120   \n",
       "\n",
       "          TCGA-C8-A131  TCGA-C8-A134  TCGA-C8-A135  TCGA-C8-A138  \\\n",
       "geneName                                                           \n",
       "A1BG         -0.294419      0.320314      2.185114     -2.018457   \n",
       "A2BP1         0.737591      1.945039     -2.473849      0.329426   \n",
       "\n",
       "          TCGA-D8-A142  TCGA-E2-A154  TCGA-E2-A158  TCGA-E2-A15A  \n",
       "geneName                                                          \n",
       "A1BG          1.539879      1.856193     -1.987341     -1.277926  \n",
       "A2BP1         0.957906      0.520191      0.291579     -2.101058  \n",
       "\n",
       "[2 rows x 77 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca_proteomics_processed = standardised_pipeline_utils.process(brca_proteomics)\n",
    "brca_proteomics_processed[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "transcriptomic_samples['BrCa (2016)'] = brca_transcriptomics_processed.shape[1]\n",
    "proteomic_samples['BrCa (2016)'] = brca_proteomics_processed.shape[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of common proteins:  7283\n",
      "Number of common samples:  59\n"
     ]
    }
   ],
   "source": [
    "brca_transcriptomics_processed, brca_proteomics_processed =  standardised_pipeline_utils.match_proteins_samples( \\\n",
    "                                                                                brca_transcriptomics_processed, \n",
    "                                                                                brca_proteomics_processed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "common_samples['BrCa (2016)'] = brca_transcriptomics_processed.shape[1]\n",
    "common_genes['BrCa (2016)'] = brca_transcriptomics_processed.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Spearman Correlation:  0.4175\n"
     ]
    }
   ],
   "source": [
    "correlation_brca = standardised_pipeline_utils.correlate_genewise(brca_transcriptomics_processed, \n",
    "                                                                  brca_proteomics_processed, 'BrCa (2016)')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Pearson Correlation:  0.4222\n"
     ]
    }
   ],
   "source": [
    "correlation_brca_pearson = standardised_pipeline_utils.correlate_genewise(brca_transcriptomics_processed, \n",
    "                                                                  brca_proteomics_processed, 'BrCa (2016)', method='pearson')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"Ovarian_Cancer\"></a>\n",
    "### 3. Ovarian Cancer "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Transcriptomic Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (18633, 595)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hybridization REF</th>\n",
       "      <th>TCGA-01-0628-11A-01R-0361-03</th>\n",
       "      <th>TCGA-01-0630-11A-01R-0361-03</th>\n",
       "      <th>TCGA-01-0631-11A-01R-0361-03</th>\n",
       "      <th>TCGA-01-0633-11A-01R-0361-03</th>\n",
       "      <th>TCGA-01-0636-11A-01R-0361-03</th>\n",
       "      <th>TCGA-01-0637-11A-01R-0361-03</th>\n",
       "      <th>TCGA-01-0639-11A-01R-0361-03</th>\n",
       "      <th>TCGA-01-0642-11A-02R-0361-03</th>\n",
       "      <th>TCGA-04-1331-01A-01R-0435-03</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-61-2104-01A-01R-0669-03</th>\n",
       "      <th>TCGA-61-2109-01A-01R-0669-03</th>\n",
       "      <th>TCGA-61-2110-01A-01R-0669-03</th>\n",
       "      <th>TCGA-61-2111-01A-01R-0669-03</th>\n",
       "      <th>TCGA-61-2113-01A-01R-0669-03</th>\n",
       "      <th>TCGA-61-2610-02A-01R-1140-03</th>\n",
       "      <th>TCGA-61-2611-02A-01R-1140-03</th>\n",
       "      <th>TCGA-61-2612-01A-01R-1140-03</th>\n",
       "      <th>TCGA-61-2613-01A-01R-1140-03</th>\n",
       "      <th>TCGA-61-2614-01A-01R-1140-03</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Composite Element REF</td>\n",
       "      <td>Signal</td>\n",
       "      <td>Signal</td>\n",
       "      <td>Signal</td>\n",
       "      <td>Signal</td>\n",
       "      <td>Signal</td>\n",
       "      <td>Signal</td>\n",
       "      <td>Signal</td>\n",
       "      <td>Signal</td>\n",
       "      <td>Signal</td>\n",
       "      <td>...</td>\n",
       "      <td>Signal</td>\n",
       "      <td>Signal</td>\n",
       "      <td>Signal</td>\n",
       "      <td>Signal</td>\n",
       "      <td>Signal</td>\n",
       "      <td>Signal</td>\n",
       "      <td>Signal</td>\n",
       "      <td>Signal</td>\n",
       "      <td>Signal</td>\n",
       "      <td>Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C9orf152</td>\n",
       "      <td>5.68406070638714</td>\n",
       "      <td>5.74059738582655</td>\n",
       "      <td>5.37909487049308</td>\n",
       "      <td>5.02811262452816</td>\n",
       "      <td>4.90491745435555</td>\n",
       "      <td>5.91024526125034</td>\n",
       "      <td>6.03056527156875</td>\n",
       "      <td>5.87840655167558</td>\n",
       "      <td>5.1293974840562</td>\n",
       "      <td>...</td>\n",
       "      <td>4.46343743805034</td>\n",
       "      <td>4.53738193913162</td>\n",
       "      <td>4.06865889851126</td>\n",
       "      <td>4.58072953660696</td>\n",
       "      <td>3.64353233073353</td>\n",
       "      <td>4.85965771708381</td>\n",
       "      <td>3.69981049486697</td>\n",
       "      <td>4.68639184341207</td>\n",
       "      <td>3.98064149167257</td>\n",
       "      <td>4.12777538628098</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 595 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Hybridization REF TCGA-01-0628-11A-01R-0361-03  \\\n",
       "0  Composite Element REF                       Signal   \n",
       "1               C9orf152             5.68406070638714   \n",
       "\n",
       "  TCGA-01-0630-11A-01R-0361-03 TCGA-01-0631-11A-01R-0361-03  \\\n",
       "0                       Signal                       Signal   \n",
       "1             5.74059738582655             5.37909487049308   \n",
       "\n",
       "  TCGA-01-0633-11A-01R-0361-03 TCGA-01-0636-11A-01R-0361-03  \\\n",
       "0                       Signal                       Signal   \n",
       "1             5.02811262452816             4.90491745435555   \n",
       "\n",
       "  TCGA-01-0637-11A-01R-0361-03 TCGA-01-0639-11A-01R-0361-03  \\\n",
       "0                       Signal                       Signal   \n",
       "1             5.91024526125034             6.03056527156875   \n",
       "\n",
       "  TCGA-01-0642-11A-02R-0361-03 TCGA-04-1331-01A-01R-0435-03  ...  \\\n",
       "0                       Signal                       Signal  ...   \n",
       "1             5.87840655167558              5.1293974840562  ...   \n",
       "\n",
       "  TCGA-61-2104-01A-01R-0669-03 TCGA-61-2109-01A-01R-0669-03  \\\n",
       "0                       Signal                       Signal   \n",
       "1             4.46343743805034             4.53738193913162   \n",
       "\n",
       "  TCGA-61-2110-01A-01R-0669-03 TCGA-61-2111-01A-01R-0669-03  \\\n",
       "0                       Signal                       Signal   \n",
       "1             4.06865889851126             4.58072953660696   \n",
       "\n",
       "  TCGA-61-2113-01A-01R-0669-03 TCGA-61-2610-02A-01R-1140-03  \\\n",
       "0                       Signal                       Signal   \n",
       "1             3.64353233073353             4.85965771708381   \n",
       "\n",
       "  TCGA-61-2611-02A-01R-1140-03 TCGA-61-2612-01A-01R-1140-03  \\\n",
       "0                       Signal                       Signal   \n",
       "1             3.69981049486697             4.68639184341207   \n",
       "\n",
       "  TCGA-61-2613-01A-01R-1140-03 TCGA-61-2614-01A-01R-1140-03  \n",
       "0                       Signal                       Signal  \n",
       "1             3.98064149167257             4.12777538628098  \n",
       "\n",
       "[2 rows x 595 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ovca_transcriptomics = pd.read_csv(file_ovca_transcriptomics, sep=\"\\t\", low_memory=False)\n",
    "print('Dimensions: ', ovca_transcriptomics.shape)\n",
    "ovca_transcriptomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (18632, 594)\n",
      "Null values count:  0\n"
     ]
    }
   ],
   "source": [
    "ovca_transcriptomics.drop(ovca_transcriptomics.index[0], inplace=True)\n",
    "ovca_transcriptomics = preprocess(ovca_transcriptomics, 'Hybridization REF', [])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert len(ovca_transcriptomics.columns[ovca_transcriptomics.columns.duplicated()]) != 0, \"columns contain no duplicates\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "ovca_transcriptomics = ovca_transcriptomics.loc[:, ~ovca_transcriptomics.columns.duplicated(keep='first')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (18632, 579)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TCGA-01-0628</th>\n",
       "      <th>TCGA-01-0630</th>\n",
       "      <th>TCGA-01-0631</th>\n",
       "      <th>TCGA-01-0633</th>\n",
       "      <th>TCGA-01-0636</th>\n",
       "      <th>TCGA-01-0637</th>\n",
       "      <th>TCGA-01-0639</th>\n",
       "      <th>TCGA-01-0642</th>\n",
       "      <th>TCGA-04-1331</th>\n",
       "      <th>TCGA-04-1332</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-61-2104</th>\n",
       "      <th>TCGA-61-2109</th>\n",
       "      <th>TCGA-61-2110</th>\n",
       "      <th>TCGA-61-2111</th>\n",
       "      <th>TCGA-61-2113</th>\n",
       "      <th>TCGA-61-2610</th>\n",
       "      <th>TCGA-61-2611</th>\n",
       "      <th>TCGA-61-2612</th>\n",
       "      <th>TCGA-61-2613</th>\n",
       "      <th>TCGA-61-2614</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hybridization REF</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15E1.2</th>\n",
       "      <td>8.518316</td>\n",
       "      <td>7.370906</td>\n",
       "      <td>7.708458</td>\n",
       "      <td>7.786219</td>\n",
       "      <td>8.224809</td>\n",
       "      <td>7.956135</td>\n",
       "      <td>7.197224</td>\n",
       "      <td>7.949850</td>\n",
       "      <td>8.704958</td>\n",
       "      <td>8.882890</td>\n",
       "      <td>...</td>\n",
       "      <td>8.731404</td>\n",
       "      <td>9.531594</td>\n",
       "      <td>8.583059</td>\n",
       "      <td>8.847043</td>\n",
       "      <td>8.976926</td>\n",
       "      <td>8.046281</td>\n",
       "      <td>9.136807</td>\n",
       "      <td>8.983845</td>\n",
       "      <td>9.509617</td>\n",
       "      <td>9.763647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2'-PDE</th>\n",
       "      <td>6.252983</td>\n",
       "      <td>6.297369</td>\n",
       "      <td>6.206035</td>\n",
       "      <td>6.278075</td>\n",
       "      <td>6.339588</td>\n",
       "      <td>6.165869</td>\n",
       "      <td>6.159611</td>\n",
       "      <td>6.257575</td>\n",
       "      <td>6.264530</td>\n",
       "      <td>6.662592</td>\n",
       "      <td>...</td>\n",
       "      <td>6.576362</td>\n",
       "      <td>6.345743</td>\n",
       "      <td>7.163527</td>\n",
       "      <td>6.374608</td>\n",
       "      <td>6.481556</td>\n",
       "      <td>6.845093</td>\n",
       "      <td>6.108022</td>\n",
       "      <td>6.476601</td>\n",
       "      <td>6.654893</td>\n",
       "      <td>6.042050</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 579 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   TCGA-01-0628  TCGA-01-0630  TCGA-01-0631  TCGA-01-0633  \\\n",
       "Hybridization REF                                                           \n",
       "15E1.2                 8.518316      7.370906      7.708458      7.786219   \n",
       "2'-PDE                 6.252983      6.297369      6.206035      6.278075   \n",
       "\n",
       "                   TCGA-01-0636  TCGA-01-0637  TCGA-01-0639  TCGA-01-0642  \\\n",
       "Hybridization REF                                                           \n",
       "15E1.2                 8.224809      7.956135      7.197224      7.949850   \n",
       "2'-PDE                 6.339588      6.165869      6.159611      6.257575   \n",
       "\n",
       "                   TCGA-04-1331  TCGA-04-1332  ...  TCGA-61-2104  \\\n",
       "Hybridization REF                              ...                 \n",
       "15E1.2                 8.704958      8.882890  ...      8.731404   \n",
       "2'-PDE                 6.264530      6.662592  ...      6.576362   \n",
       "\n",
       "                   TCGA-61-2109  TCGA-61-2110  TCGA-61-2111  TCGA-61-2113  \\\n",
       "Hybridization REF                                                           \n",
       "15E1.2                 9.531594      8.583059      8.847043      8.976926   \n",
       "2'-PDE                 6.345743      7.163527      6.374608      6.481556   \n",
       "\n",
       "                   TCGA-61-2610  TCGA-61-2611  TCGA-61-2612  TCGA-61-2613  \\\n",
       "Hybridization REF                                                           \n",
       "15E1.2                 8.046281      9.136807      8.983845      9.509617   \n",
       "2'-PDE                 6.845093      6.108022      6.476601      6.654893   \n",
       "\n",
       "                   TCGA-61-2614  \n",
       "Hybridization REF                \n",
       "15E1.2                 9.763647  \n",
       "2'-PDE                 6.042050  \n",
       "\n",
       "[2 rows x 579 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ovca_transcriptomics_processed = standardised_pipeline_utils.process(ovca_transcriptomics)\n",
    "ovca_transcriptomics_processed[:2]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Proteomic Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (9600, 208)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>refseq_peptide</th>\n",
       "      <th>hgnc_symbol</th>\n",
       "      <th>PNNL-TCGA-09-1664</th>\n",
       "      <th>PNNL-TCGA-13-1484</th>\n",
       "      <th>PNNL-TCGA-13-1488</th>\n",
       "      <th>PNNL-TCGA-13-1489</th>\n",
       "      <th>PNNL-TCGA-13-1494</th>\n",
       "      <th>PNNL-TCGA-13-1495</th>\n",
       "      <th>PNNL-TCGA-13-1499</th>\n",
       "      <th>PNNL-TCGA-13-2071</th>\n",
       "      <th>...</th>\n",
       "      <th>JHU-TCGA-61-1911</th>\n",
       "      <th>JHU-TCGA-61-1914</th>\n",
       "      <th>JHU-TCGA-61-1915</th>\n",
       "      <th>JHU-TCGA-61-1918</th>\n",
       "      <th>JHU-TCGA-61-1995</th>\n",
       "      <th>JHU-TCGA-61-2008</th>\n",
       "      <th>JHU-TCGA-61-2087</th>\n",
       "      <th>JHU-TCGA-61-2088</th>\n",
       "      <th>JHU-TCGA-61-2094</th>\n",
       "      <th>JHU-TCGA-61-2613</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NP_000005</td>\n",
       "      <td>A2M</td>\n",
       "      <td>-0.0581</td>\n",
       "      <td>-0.688</td>\n",
       "      <td>0.5180</td>\n",
       "      <td>-0.432</td>\n",
       "      <td>0.333</td>\n",
       "      <td>0.109</td>\n",
       "      <td>-1.230</td>\n",
       "      <td>0.172</td>\n",
       "      <td>...</td>\n",
       "      <td>0.265750</td>\n",
       "      <td>-0.688250</td>\n",
       "      <td>-0.071250</td>\n",
       "      <td>-0.767250</td>\n",
       "      <td>-0.821250</td>\n",
       "      <td>-0.360550</td>\n",
       "      <td>-1.022250</td>\n",
       "      <td>-1.172250</td>\n",
       "      <td>-0.132250</td>\n",
       "      <td>1.165750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NP_000007</td>\n",
       "      <td>ACADM</td>\n",
       "      <td>-0.1040</td>\n",
       "      <td>0.404</td>\n",
       "      <td>0.0714</td>\n",
       "      <td>0.272</td>\n",
       "      <td>-0.369</td>\n",
       "      <td>0.175</td>\n",
       "      <td>-0.295</td>\n",
       "      <td>-0.425</td>\n",
       "      <td>...</td>\n",
       "      <td>0.073225</td>\n",
       "      <td>-0.326775</td>\n",
       "      <td>-1.069775</td>\n",
       "      <td>-0.383775</td>\n",
       "      <td>0.105225</td>\n",
       "      <td>0.365225</td>\n",
       "      <td>-0.277775</td>\n",
       "      <td>0.401225</td>\n",
       "      <td>0.056225</td>\n",
       "      <td>-0.372775</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 208 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  refseq_peptide hgnc_symbol  PNNL-TCGA-09-1664  PNNL-TCGA-13-1484  \\\n",
       "0      NP_000005         A2M            -0.0581             -0.688   \n",
       "1      NP_000007       ACADM            -0.1040              0.404   \n",
       "\n",
       "   PNNL-TCGA-13-1488  PNNL-TCGA-13-1489  PNNL-TCGA-13-1494  PNNL-TCGA-13-1495  \\\n",
       "0             0.5180             -0.432              0.333              0.109   \n",
       "1             0.0714              0.272             -0.369              0.175   \n",
       "\n",
       "   PNNL-TCGA-13-1499  PNNL-TCGA-13-2071  ...  JHU-TCGA-61-1911  \\\n",
       "0             -1.230              0.172  ...          0.265750   \n",
       "1             -0.295             -0.425  ...          0.073225   \n",
       "\n",
       "   JHU-TCGA-61-1914  JHU-TCGA-61-1915  JHU-TCGA-61-1918  JHU-TCGA-61-1995  \\\n",
       "0         -0.688250         -0.071250         -0.767250         -0.821250   \n",
       "1         -0.326775         -1.069775         -0.383775          0.105225   \n",
       "\n",
       "   JHU-TCGA-61-2008  JHU-TCGA-61-2087  JHU-TCGA-61-2088  JHU-TCGA-61-2094  \\\n",
       "0         -0.360550         -1.022250         -1.172250         -0.132250   \n",
       "1          0.365225         -0.277775          0.401225          0.056225   \n",
       "\n",
       "   JHU-TCGA-61-2613  \n",
       "0          1.165750  \n",
       "1         -0.372775  \n",
       "\n",
       "[2 rows x 208 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ovca_proteomics = pd.read_excel(io=file_ovca_proteomics, sheet_name=\"FullOvarianProteomeData\")\n",
    "print('Dimensions: ', ovca_proteomics.shape)\n",
    "ovca_proteomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Clean data\n",
    "ovca_proteomics = ovca_proteomics.set_index('hgnc_symbol').drop('refseq_peptide', axis=1)\n",
    "ovca_proteomics.columns = ['-'.join(index) for index in ovca_proteomics.columns.str.split('-').str[1:4]]\n",
    "# taking mean of the replicates the replicates \n",
    "ovca_proteomics = ovca_proteomics.groupby(ovca_proteomics.columns, axis=1).mean()\n",
    "# taking mean of the isoforms \n",
    "ovca_proteomics = ovca_proteomics.groupby(ovca_proteomics.index).mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert len(ovca_proteomics.columns[ovca_proteomics.columns.duplicated()]) == 0, \"columns contain duplicates\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (6627, 174)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TCGA-09-1664</th>\n",
       "      <th>TCGA-09-2056</th>\n",
       "      <th>TCGA-13-1404</th>\n",
       "      <th>TCGA-13-1409</th>\n",
       "      <th>TCGA-13-1410</th>\n",
       "      <th>TCGA-13-1482</th>\n",
       "      <th>TCGA-13-1483</th>\n",
       "      <th>TCGA-13-1484</th>\n",
       "      <th>TCGA-13-1485</th>\n",
       "      <th>TCGA-13-1487</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-61-1919</th>\n",
       "      <th>TCGA-61-1995</th>\n",
       "      <th>TCGA-61-2008</th>\n",
       "      <th>TCGA-61-2087</th>\n",
       "      <th>TCGA-61-2088</th>\n",
       "      <th>TCGA-61-2094</th>\n",
       "      <th>TCGA-61-2095</th>\n",
       "      <th>TCGA-61-2096</th>\n",
       "      <th>TCGA-61-2612</th>\n",
       "      <th>TCGA-61-2613</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hgnc_symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>0.316850</td>\n",
       "      <td>-0.63230</td>\n",
       "      <td>-1.01930</td>\n",
       "      <td>-0.69730</td>\n",
       "      <td>-0.42900</td>\n",
       "      <td>-0.42700</td>\n",
       "      <td>-0.40200</td>\n",
       "      <td>-1.060</td>\n",
       "      <td>-0.21430</td>\n",
       "      <td>-0.64430</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.00193</td>\n",
       "      <td>-0.122350</td>\n",
       "      <td>0.634850</td>\n",
       "      <td>-1.11530</td>\n",
       "      <td>-0.83830</td>\n",
       "      <td>-0.06730</td>\n",
       "      <td>0.635</td>\n",
       "      <td>1.910</td>\n",
       "      <td>0.0689</td>\n",
       "      <td>0.90970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>-0.138175</td>\n",
       "      <td>0.00675</td>\n",
       "      <td>-0.86525</td>\n",
       "      <td>-0.98125</td>\n",
       "      <td>-0.00425</td>\n",
       "      <td>-0.46865</td>\n",
       "      <td>-1.12625</td>\n",
       "      <td>-0.688</td>\n",
       "      <td>-0.58425</td>\n",
       "      <td>-1.38325</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.00360</td>\n",
       "      <td>-0.706125</td>\n",
       "      <td>-0.549775</td>\n",
       "      <td>-1.02225</td>\n",
       "      <td>-1.17225</td>\n",
       "      <td>-0.13225</td>\n",
       "      <td>0.910</td>\n",
       "      <td>0.353</td>\n",
       "      <td>0.3500</td>\n",
       "      <td>1.16575</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 174 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             TCGA-09-1664  TCGA-09-2056  TCGA-13-1404  TCGA-13-1409  \\\n",
       "hgnc_symbol                                                           \n",
       "A1BG             0.316850      -0.63230      -1.01930      -0.69730   \n",
       "A2M             -0.138175       0.00675      -0.86525      -0.98125   \n",
       "\n",
       "             TCGA-13-1410  TCGA-13-1482  TCGA-13-1483  TCGA-13-1484  \\\n",
       "hgnc_symbol                                                           \n",
       "A1BG             -0.42900      -0.42700      -0.40200        -1.060   \n",
       "A2M              -0.00425      -0.46865      -1.12625        -0.688   \n",
       "\n",
       "             TCGA-13-1485  TCGA-13-1487  ...  TCGA-61-1919  TCGA-61-1995  \\\n",
       "hgnc_symbol                              ...                               \n",
       "A1BG             -0.21430      -0.64430  ...      -0.00193     -0.122350   \n",
       "A2M              -0.58425      -1.38325  ...      -0.00360     -0.706125   \n",
       "\n",
       "             TCGA-61-2008  TCGA-61-2087  TCGA-61-2088  TCGA-61-2094  \\\n",
       "hgnc_symbol                                                           \n",
       "A1BG             0.634850      -1.11530      -0.83830      -0.06730   \n",
       "A2M             -0.549775      -1.02225      -1.17225      -0.13225   \n",
       "\n",
       "             TCGA-61-2095  TCGA-61-2096  TCGA-61-2612  TCGA-61-2613  \n",
       "hgnc_symbol                                                          \n",
       "A1BG                0.635         1.910        0.0689       0.90970  \n",
       "A2M                 0.910         0.353        0.3500       1.16575  \n",
       "\n",
       "[2 rows x 174 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ovca_proteomics_processed = standardised_pipeline_utils.process(ovca_proteomics)\n",
    "ovca_proteomics_processed[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "transcriptomic_samples['Ovarian'] = ovca_transcriptomics_processed.shape[1]\n",
    "proteomic_samples['Ovarian'] = ovca_proteomics_processed.shape[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of common proteins:  5841\n",
      "Number of common samples:  174\n"
     ]
    }
   ],
   "source": [
    "ovca_transcriptomics_processed, ovca_proteomics_processed =  standardised_pipeline_utils.match_proteins_samples( \\\n",
    "                                                                                ovca_transcriptomics_processed, \n",
    "                                                                                ovca_proteomics_processed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "common_samples['Ovarian'] = ovca_transcriptomics_processed.shape[1]\n",
    "common_genes['Ovarian'] = ovca_transcriptomics_processed.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Spearman Correlation:  0.4121\n"
     ]
    }
   ],
   "source": [
    "correlation_ovca = standardised_pipeline_utils.correlate_genewise(ovca_transcriptomics_processed, \n",
    "                                                                   ovca_proteomics_processed, 'Ovarian')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Pearson Correlation:  0.4097\n"
     ]
    }
   ],
   "source": [
    "correlation_ovca_pearson = standardised_pipeline_utils.correlate_genewise(ovca_transcriptomics_processed, \n",
    "                                                                   ovca_proteomics_processed, 'Ovarian', method='pearson')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"Colon_Cancer\"></a>\n",
    "### 4. Colon Cancer"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Transcriptomic Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (13482, 107)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>attrib_name</th>\n",
       "      <th>01CO001</th>\n",
       "      <th>01CO005</th>\n",
       "      <th>01CO006</th>\n",
       "      <th>01CO008</th>\n",
       "      <th>01CO013</th>\n",
       "      <th>01CO014</th>\n",
       "      <th>01CO015</th>\n",
       "      <th>01CO019</th>\n",
       "      <th>01CO022</th>\n",
       "      <th>...</th>\n",
       "      <th>20CO001</th>\n",
       "      <th>20CO003</th>\n",
       "      <th>20CO004</th>\n",
       "      <th>20CO007</th>\n",
       "      <th>21CO006</th>\n",
       "      <th>21CO007</th>\n",
       "      <th>22CO004</th>\n",
       "      <th>22CO006</th>\n",
       "      <th>24CO005</th>\n",
       "      <th>27CO004</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>10.1984</td>\n",
       "      <td>10.1586</td>\n",
       "      <td>9.2119</td>\n",
       "      <td>10.7322</td>\n",
       "      <td>9.7160</td>\n",
       "      <td>9.5177</td>\n",
       "      <td>9.6275</td>\n",
       "      <td>9.5736</td>\n",
       "      <td>4.8580</td>\n",
       "      <td>...</td>\n",
       "      <td>8.5078</td>\n",
       "      <td>2.5850</td>\n",
       "      <td>10.1408</td>\n",
       "      <td>9.0688</td>\n",
       "      <td>9.2336</td>\n",
       "      <td>10.0768</td>\n",
       "      <td>10.9293</td>\n",
       "      <td>9.2432</td>\n",
       "      <td>8.6546</td>\n",
       "      <td>9.8025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M</td>\n",
       "      <td>13.8117</td>\n",
       "      <td>12.6434</td>\n",
       "      <td>15.0040</td>\n",
       "      <td>13.9284</td>\n",
       "      <td>12.9963</td>\n",
       "      <td>11.8700</td>\n",
       "      <td>13.0274</td>\n",
       "      <td>12.5637</td>\n",
       "      <td>11.6234</td>\n",
       "      <td>...</td>\n",
       "      <td>14.2033</td>\n",
       "      <td>13.7171</td>\n",
       "      <td>11.3078</td>\n",
       "      <td>12.7292</td>\n",
       "      <td>12.9123</td>\n",
       "      <td>13.4762</td>\n",
       "      <td>12.6131</td>\n",
       "      <td>14.2620</td>\n",
       "      <td>13.0250</td>\n",
       "      <td>13.1139</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 107 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  attrib_name  01CO001  01CO005  01CO006  01CO008  01CO013  01CO014  01CO015  \\\n",
       "0        A1CF  10.1984  10.1586   9.2119  10.7322   9.7160   9.5177   9.6275   \n",
       "1         A2M  13.8117  12.6434  15.0040  13.9284  12.9963  11.8700  13.0274   \n",
       "\n",
       "   01CO019  01CO022  ...  20CO001  20CO003  20CO004  20CO007  21CO006  \\\n",
       "0   9.5736   4.8580  ...   8.5078   2.5850  10.1408   9.0688   9.2336   \n",
       "1  12.5637  11.6234  ...  14.2033  13.7171  11.3078  12.7292  12.9123   \n",
       "\n",
       "   21CO007  22CO004  22CO006  24CO005  27CO004  \n",
       "0  10.0768  10.9293   9.2432   8.6546   9.8025  \n",
       "1  13.4762  12.6131  14.2620  13.0250  13.1139  \n",
       "\n",
       "[2 rows x 107 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "colon_transcriptomics = pd.read_csv(file_colon_transcriptomics, sep='\\t',  low_memory=False)\n",
    "print('Dimensions: ', colon_transcriptomics.shape)\n",
    "colon_transcriptomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (13482, 106)\n",
      "Null values count:  0\n"
     ]
    }
   ],
   "source": [
    "colon_transcriptomics = preprocess(colon_transcriptomics, index_col='attrib_name')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert len(colon_transcriptomics.columns[colon_transcriptomics.columns.duplicated()]) == 0, \"columns contain duplicates\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (13448, 106)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>01CO001</th>\n",
       "      <th>01CO005</th>\n",
       "      <th>01CO006</th>\n",
       "      <th>01CO008</th>\n",
       "      <th>01CO013</th>\n",
       "      <th>01CO014</th>\n",
       "      <th>01CO015</th>\n",
       "      <th>01CO019</th>\n",
       "      <th>01CO022</th>\n",
       "      <th>05CO002</th>\n",
       "      <th>...</th>\n",
       "      <th>20CO001</th>\n",
       "      <th>20CO003</th>\n",
       "      <th>20CO004</th>\n",
       "      <th>20CO007</th>\n",
       "      <th>21CO006</th>\n",
       "      <th>21CO007</th>\n",
       "      <th>22CO004</th>\n",
       "      <th>22CO006</th>\n",
       "      <th>24CO005</th>\n",
       "      <th>27CO004</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>attrib_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>10.1984</td>\n",
       "      <td>10.1586</td>\n",
       "      <td>9.2119</td>\n",
       "      <td>10.7322</td>\n",
       "      <td>9.7160</td>\n",
       "      <td>9.5177</td>\n",
       "      <td>9.6275</td>\n",
       "      <td>9.5736</td>\n",
       "      <td>4.8580</td>\n",
       "      <td>9.2432</td>\n",
       "      <td>...</td>\n",
       "      <td>8.5078</td>\n",
       "      <td>2.5850</td>\n",
       "      <td>10.1408</td>\n",
       "      <td>9.0688</td>\n",
       "      <td>9.2336</td>\n",
       "      <td>10.0768</td>\n",
       "      <td>10.9293</td>\n",
       "      <td>9.2432</td>\n",
       "      <td>8.6546</td>\n",
       "      <td>9.8025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>13.8117</td>\n",
       "      <td>12.6434</td>\n",
       "      <td>15.0040</td>\n",
       "      <td>13.9284</td>\n",
       "      <td>12.9963</td>\n",
       "      <td>11.8700</td>\n",
       "      <td>13.0274</td>\n",
       "      <td>12.5637</td>\n",
       "      <td>11.6234</td>\n",
       "      <td>15.3414</td>\n",
       "      <td>...</td>\n",
       "      <td>14.2033</td>\n",
       "      <td>13.7171</td>\n",
       "      <td>11.3078</td>\n",
       "      <td>12.7292</td>\n",
       "      <td>12.9123</td>\n",
       "      <td>13.4762</td>\n",
       "      <td>12.6131</td>\n",
       "      <td>14.2620</td>\n",
       "      <td>13.0250</td>\n",
       "      <td>13.1139</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 106 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             01CO001  01CO005  01CO006  01CO008  01CO013  01CO014  01CO015  \\\n",
       "attrib_name                                                                  \n",
       "A1CF         10.1984  10.1586   9.2119  10.7322   9.7160   9.5177   9.6275   \n",
       "A2M          13.8117  12.6434  15.0040  13.9284  12.9963  11.8700  13.0274   \n",
       "\n",
       "             01CO019  01CO022  05CO002  ...  20CO001  20CO003  20CO004  \\\n",
       "attrib_name                             ...                              \n",
       "A1CF          9.5736   4.8580   9.2432  ...   8.5078   2.5850  10.1408   \n",
       "A2M          12.5637  11.6234  15.3414  ...  14.2033  13.7171  11.3078   \n",
       "\n",
       "             20CO007  21CO006  21CO007  22CO004  22CO006  24CO005  27CO004  \n",
       "attrib_name                                                                 \n",
       "A1CF          9.0688   9.2336  10.0768  10.9293   9.2432   8.6546   9.8025  \n",
       "A2M          12.7292  12.9123  13.4762  12.6131  14.2620  13.0250  13.1139  \n",
       "\n",
       "[2 rows x 106 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "colon_transcriptomics_processed = standardised_pipeline_utils.process(colon_transcriptomics)\n",
    "colon_transcriptomics_processed[:2]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Proteomic Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (6422, 97)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>attrib_name</th>\n",
       "      <th>01CO005</th>\n",
       "      <th>01CO006</th>\n",
       "      <th>01CO008</th>\n",
       "      <th>01CO013</th>\n",
       "      <th>01CO014</th>\n",
       "      <th>01CO015</th>\n",
       "      <th>01CO019</th>\n",
       "      <th>01CO022</th>\n",
       "      <th>05CO002</th>\n",
       "      <th>...</th>\n",
       "      <th>20CO001</th>\n",
       "      <th>20CO003</th>\n",
       "      <th>20CO004</th>\n",
       "      <th>20CO006</th>\n",
       "      <th>20CO007</th>\n",
       "      <th>21CO006</th>\n",
       "      <th>21CO007</th>\n",
       "      <th>22CO004</th>\n",
       "      <th>22CO006</th>\n",
       "      <th>27CO004</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>-1.672</td>\n",
       "      <td>-2.1500</td>\n",
       "      <td>-1.786</td>\n",
       "      <td>-2.115</td>\n",
       "      <td>-1.793</td>\n",
       "      <td>-2.526</td>\n",
       "      <td>-1.441</td>\n",
       "      <td>1.047</td>\n",
       "      <td>-1.878</td>\n",
       "      <td>...</td>\n",
       "      <td>0.297</td>\n",
       "      <td>0.005</td>\n",
       "      <td>-3.390</td>\n",
       "      <td>-1.6480</td>\n",
       "      <td>-1.72</td>\n",
       "      <td>0.038</td>\n",
       "      <td>-1.952</td>\n",
       "      <td>-1.3342</td>\n",
       "      <td>-0.069</td>\n",
       "      <td>-2.6020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>-0.340</td>\n",
       "      <td>-0.3476</td>\n",
       "      <td>-0.124</td>\n",
       "      <td>-0.197</td>\n",
       "      <td>-0.730</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.103</td>\n",
       "      <td>-1.126</td>\n",
       "      <td>0.507</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.097</td>\n",
       "      <td>-1.693</td>\n",
       "      <td>0.985</td>\n",
       "      <td>0.9395</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.877</td>\n",
       "      <td>0.777</td>\n",
       "      <td>0.1120</td>\n",
       "      <td>-0.519</td>\n",
       "      <td>0.1017</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 97 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  attrib_name  01CO005  01CO006  01CO008  01CO013  01CO014  01CO015  01CO019  \\\n",
       "0        A1BG   -1.672  -2.1500   -1.786   -2.115   -1.793   -2.526   -1.441   \n",
       "1        A1CF   -0.340  -0.3476   -0.124   -0.197   -0.730      NaN   -0.103   \n",
       "\n",
       "   01CO022  05CO002  ...  20CO001  20CO003  20CO004  20CO006  20CO007  \\\n",
       "0    1.047   -1.878  ...    0.297    0.005   -3.390  -1.6480    -1.72   \n",
       "1   -1.126    0.507  ...   -0.097   -1.693    0.985   0.9395      NaN   \n",
       "\n",
       "   21CO006  21CO007  22CO004  22CO006  27CO004  \n",
       "0    0.038   -1.952  -1.3342   -0.069  -2.6020  \n",
       "1   -0.877    0.777   0.1120   -0.519   0.1017  \n",
       "\n",
       "[2 rows x 97 columns]"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "colon_proteomics = pd.read_csv(file_colon_proteomics, sep='\\t')\n",
    "print('Dimensions: ', colon_proteomics.shape)\n",
    "colon_proteomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (6422, 96)\n",
      "Null values count:  64560\n"
     ]
    }
   ],
   "source": [
    "colon_proteomics = preprocess(colon_proteomics, index_col='attrib_name')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert len(colon_proteomics.columns[colon_proteomics.columns.duplicated()]) == 0, \"columns contain duplicates\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (5152, 96)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>01CO005</th>\n",
       "      <th>01CO006</th>\n",
       "      <th>01CO008</th>\n",
       "      <th>01CO013</th>\n",
       "      <th>01CO014</th>\n",
       "      <th>01CO015</th>\n",
       "      <th>01CO019</th>\n",
       "      <th>01CO022</th>\n",
       "      <th>05CO002</th>\n",
       "      <th>05CO003</th>\n",
       "      <th>...</th>\n",
       "      <th>20CO001</th>\n",
       "      <th>20CO003</th>\n",
       "      <th>20CO004</th>\n",
       "      <th>20CO006</th>\n",
       "      <th>20CO007</th>\n",
       "      <th>21CO006</th>\n",
       "      <th>21CO007</th>\n",
       "      <th>22CO004</th>\n",
       "      <th>22CO006</th>\n",
       "      <th>27CO004</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>attrib_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-1.672</td>\n",
       "      <td>-2.1500</td>\n",
       "      <td>-1.786</td>\n",
       "      <td>-2.115</td>\n",
       "      <td>-1.793</td>\n",
       "      <td>-2.526</td>\n",
       "      <td>-1.441</td>\n",
       "      <td>1.047</td>\n",
       "      <td>-1.878</td>\n",
       "      <td>-0.294</td>\n",
       "      <td>...</td>\n",
       "      <td>0.297</td>\n",
       "      <td>0.005</td>\n",
       "      <td>-3.390</td>\n",
       "      <td>-1.6480</td>\n",
       "      <td>-1.72</td>\n",
       "      <td>0.038</td>\n",
       "      <td>-1.952</td>\n",
       "      <td>-1.3342</td>\n",
       "      <td>-0.069</td>\n",
       "      <td>-2.6020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>-0.340</td>\n",
       "      <td>-0.3476</td>\n",
       "      <td>-0.124</td>\n",
       "      <td>-0.197</td>\n",
       "      <td>-0.730</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.103</td>\n",
       "      <td>-1.126</td>\n",
       "      <td>0.507</td>\n",
       "      <td>-1.657</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.097</td>\n",
       "      <td>-1.693</td>\n",
       "      <td>0.985</td>\n",
       "      <td>0.9395</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.877</td>\n",
       "      <td>0.777</td>\n",
       "      <td>0.1120</td>\n",
       "      <td>-0.519</td>\n",
       "      <td>0.1017</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 96 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             01CO005  01CO006  01CO008  01CO013  01CO014  01CO015  01CO019  \\\n",
       "attrib_name                                                                  \n",
       "A1BG          -1.672  -2.1500   -1.786   -2.115   -1.793   -2.526   -1.441   \n",
       "A1CF          -0.340  -0.3476   -0.124   -0.197   -0.730      NaN   -0.103   \n",
       "\n",
       "             01CO022  05CO002  05CO003  ...  20CO001  20CO003  20CO004  \\\n",
       "attrib_name                             ...                              \n",
       "A1BG           1.047   -1.878   -0.294  ...    0.297    0.005   -3.390   \n",
       "A1CF          -1.126    0.507   -1.657  ...   -0.097   -1.693    0.985   \n",
       "\n",
       "             20CO006  20CO007  21CO006  21CO007  22CO004  22CO006  27CO004  \n",
       "attrib_name                                                                 \n",
       "A1BG         -1.6480    -1.72    0.038   -1.952  -1.3342   -0.069  -2.6020  \n",
       "A1CF          0.9395      NaN   -0.877    0.777   0.1120   -0.519   0.1017  \n",
       "\n",
       "[2 rows x 96 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "colon_proteomics_processed = standardised_pipeline_utils.process(colon_proteomics)\n",
    "colon_proteomics_processed[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "transcriptomic_samples['Colon'] = colon_transcriptomics_processed.shape[1]\n",
    "proteomic_samples['Colon'] = colon_proteomics_processed.shape[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of common proteins:  4970\n",
      "Number of common samples:  95\n"
     ]
    }
   ],
   "source": [
    "colon_transcriptomics_processed, colon_proteomics_processed =  standardised_pipeline_utils.match_proteins_samples( \\\n",
    "                                                                                colon_transcriptomics_processed, \n",
    "                                                                                colon_proteomics_processed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "common_samples['Colon'] = colon_transcriptomics_processed.shape[1]\n",
    "common_genes['Colon'] = colon_transcriptomics_processed.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Spearman Correlation:  0.2702\n"
     ]
    }
   ],
   "source": [
    "correlation_colon = standardised_pipeline_utils.correlate_genewise(colon_transcriptomics_processed, \n",
    "                                                                    colon_proteomics_processed, 'Colon')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Pearson Correlation:  0.2791\n"
     ]
    }
   ],
   "source": [
    "correlation_colon_pearson = standardised_pipeline_utils.correlate_genewise(colon_transcriptomics_processed, \n",
    "                                                                    colon_proteomics_processed, 'Colon', method='pearson')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CRC (2014)</th>\n",
       "      <th>BrCa (2016)</th>\n",
       "      <th>Ovarian</th>\n",
       "      <th>Colon</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>0.123133</td>\n",
       "      <td>-0.093805</td>\n",
       "      <td>-0.078889</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>-0.037595</td>\n",
       "      <td>0.118235</td>\n",
       "      <td>0.191865</td>\n",
       "      <td>-0.026288</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      CRC (2014)  BrCa (2016)   Ovarian     Colon\n",
       "A1BG    0.123133    -0.093805 -0.078889       NaN\n",
       "A2M    -0.037595     0.118235  0.191865 -0.026288"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "correlation_combined = pd.concat([correlation_crc, correlation_brca, correlation_ovca, correlation_colon], axis=1)\n",
    "correlation_combined.to_csv(file_tumour_correlation)\n",
    "correlation_combined[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CRC (2014)     0.22\n",
       "BrCa (2016)    0.42\n",
       "Ovarian        0.41\n",
       "Colon          0.28\n",
       "dtype: float64"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "correlation_combined_pearson = pd.concat([correlation_crc_pearson, correlation_brca_pearson, \n",
    "                                          correlation_ovca_pearson, correlation_colon_pearson], axis=1)\n",
    "round(correlation_combined_pearson.median(), 2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># Samples in Trancriptomic Data</th>\n",
       "      <th># Samples in Proteomic Data</th>\n",
       "      <th># Common Samples</th>\n",
       "      <th># Common Proteins</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Data</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CRC (2014)</th>\n",
       "      <td>244</td>\n",
       "      <td>90</td>\n",
       "      <td>77</td>\n",
       "      <td>3238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BrCa (2016)</th>\n",
       "      <td>817</td>\n",
       "      <td>77</td>\n",
       "      <td>59</td>\n",
       "      <td>7283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ovarian</th>\n",
       "      <td>579</td>\n",
       "      <td>174</td>\n",
       "      <td>174</td>\n",
       "      <td>5841</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Colon</th>\n",
       "      <td>106</td>\n",
       "      <td>96</td>\n",
       "      <td>95</td>\n",
       "      <td>4970</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             # Samples in Trancriptomic Data  # Samples in Proteomic Data  \\\n",
       "Data                                                                        \n",
       "CRC (2014)                               244                           90   \n",
       "BrCa (2016)                              817                           77   \n",
       "Ovarian                                  579                          174   \n",
       "Colon                                    106                           96   \n",
       "\n",
       "             # Common Samples  # Common Proteins  \n",
       "Data                                              \n",
       "CRC (2014)                 77               3238  \n",
       "BrCa (2016)                59               7283  \n",
       "Ovarian                   174               5841  \n",
       "Colon                      95               4970  "
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def dataframe_from_dict(*dict_args):\n",
    "    dataframe = pd.DataFrame.from_dict(dict_args[0], orient='index')\n",
    "    for i in range(1, len(dict_args)):\n",
    "        dataframe = pd.concat([dataframe, pd.DataFrame.from_dict(dict_args[i], orient='index')], axis=1)\n",
    "    dataframe.reset_index(inplace=True)\n",
    "    dataframe.columns=['Data', '# Samples in Trancriptomic Data', '# Samples in Proteomic Data', \n",
    "                       '# Common Samples', '# Common Proteins']\n",
    "    return dataframe.set_index('Data')\n",
    "\n",
    "sample_data = dataframe_from_dict(transcriptomic_samples, proteomic_samples, common_samples, common_genes)\n",
    "sample_data.to_csv(file_samples_info)\n",
    "sample_data"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
