{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##  Part 2: Compute mRNA-protein correlation for new tumour studies"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Input:** Transcriptomics and Proteomics data of the tumour studies listed below.\n",
    "1. <a href=https://tinyurl.com/3fv3wdrp>Clear cell renal carcinoma (2019)</a>     \n",
    "2. <a href=https://tinyurl.com/k829xt9r>Endometrial Cancer (2020)</a>         \n",
    "3. <a href=https://tinyurl.com/yfvtp3wp>Lung Adenocarcinoma (2020)</a>         \n",
    "4. <a href=https://tinyurl.com/2zsjx24s>Breast Cancer (2020)</a>           \n",
    "5. <a href=https://tinyurl.com/2mxy6vw7>Head and Neck Squammous Cell Carcinoma (2021)</a>           \n",
    "6. <a href=https://tinyurl.com/3f65rr3x>Glioblastoma (2021)</a>            \n",
    "\n",
    "**Output:** Gene-wise correlation between mRNA and protein abundances "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-info\">\n",
    "    <b>Note:</b> The input data are downloaded from the <a href=https://pypi.org/project/cptac/>CPTAC python API</a>\n",
    "</div>    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Import Packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import cptac\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from cptac import utils as ut\n",
    "\n",
    "%load_ext autoreload\n",
    "%autoreload 1\n",
    "%aimport standardised_pipeline_utils"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "get_local_data_path = lambda folders, fname: os.path.normpath('../local_data/'+'/'.join(folders) +'/'+ fname)\n",
    "\n",
    "# Output File\n",
    "file_tumour_correlation = get_local_data_path(['processed', 'correlation_mRNA_protein'], 'cptac_tumour_studies.csv')\n",
    "file_samples_info = get_local_data_path(['processed', 'correlation_mRNA_protein'], 'samples_info.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_transcriptomics(data, multiIndexed=False):\n",
    "    transcriptomics = data.get_transcriptomics(tissue_type='tumor').transpose()\n",
    "    if(multiIndexed):    \n",
    "        transcriptomics = ut.reduce_multiindex(transcriptomics.transpose(), levels_to_drop='Database_ID', quiet=True)\n",
    "        transcriptomics = transcriptomics.transpose()\n",
    "    print(\"Dimensions: \", transcriptomics.shape)\n",
    "    print(\"Null values count: \", transcriptomics.isnull().sum().sum())\n",
    "    return transcriptomics\n",
    "\n",
    "def get_proteomics(data, multiIndexed=True):\n",
    "    proteomics = data.get_proteomics(tissue_type='tumor')\n",
    "    if(multiIndexed):    \n",
    "        proteomics = ut.reduce_multiindex(proteomics, levels_to_drop='Database_ID', quiet=True)\n",
    "    print(\"Dataframe transposed.\")\n",
    "    proteomics = proteomics.transpose()\n",
    "    print(\"Dimensions: \", proteomics.shape)\n",
    "    print(\"Null values count: \", proteomics.isnull().sum().sum())\n",
    "    return proteomics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Information collected for the Supplemental Table S1B\n",
    "common_samples = {}\n",
    "common_genes = {}\n",
    "transcriptomic_samples = {}\n",
    "proteomic_samples = {}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Download data from CPTAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Description</th>\n",
       "      <th>Data reuse status</th>\n",
       "      <th>Publication link</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Dataset name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Brca</th>\n",
       "      <td>breast cancer</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/33212010/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ccrcc</th>\n",
       "      <td>clear cell renal cell carcinoma (kidney)</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/31675502/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Colon</th>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/31031003/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Endometrial</th>\n",
       "      <td>endometrial carcinoma (uterine)</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/32059776/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gbm</th>\n",
       "      <td>glioblastoma</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/33577785/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hnscc</th>\n",
       "      <td>head and neck squamous cell carcinoma</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/33417831/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lscc</th>\n",
       "      <td>lung squamous cell carcinoma</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/34358469/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Luad</th>\n",
       "      <td>lung adenocarcinoma</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/32649874/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ovarian</th>\n",
       "      <td>high grade serous ovarian cancer</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/27372738/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pdac</th>\n",
       "      <td>pancreatic ductal adenocarcinoma</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/34534465/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UcecConf</th>\n",
       "      <td>endometrial confirmatory carcinoma</td>\n",
       "      <td>password access only</td>\n",
       "      <td>unpublished</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GbmConf</th>\n",
       "      <td>glioblastoma confirmatory</td>\n",
       "      <td>password access only</td>\n",
       "      <td>unpublished</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           Description     Data reuse status  \\\n",
       "Dataset name                                                                   \n",
       "Brca                                     breast cancer       no restrictions   \n",
       "Ccrcc         clear cell renal cell carcinoma (kidney)       no restrictions   \n",
       "Colon                                colorectal cancer       no restrictions   \n",
       "Endometrial            endometrial carcinoma (uterine)       no restrictions   \n",
       "Gbm                                       glioblastoma       no restrictions   \n",
       "Hnscc            head and neck squamous cell carcinoma       no restrictions   \n",
       "Lscc                      lung squamous cell carcinoma       no restrictions   \n",
       "Luad                               lung adenocarcinoma       no restrictions   \n",
       "Ovarian               high grade serous ovarian cancer       no restrictions   \n",
       "Pdac                  pancreatic ductal adenocarcinoma       no restrictions   \n",
       "UcecConf            endometrial confirmatory carcinoma  password access only   \n",
       "GbmConf                      glioblastoma confirmatory  password access only   \n",
       "\n",
       "                                       Publication link  \n",
       "Dataset name                                             \n",
       "Brca          https://pubmed.ncbi.nlm.nih.gov/33212010/  \n",
       "Ccrcc         https://pubmed.ncbi.nlm.nih.gov/31675502/  \n",
       "Colon         https://pubmed.ncbi.nlm.nih.gov/31031003/  \n",
       "Endometrial   https://pubmed.ncbi.nlm.nih.gov/32059776/  \n",
       "Gbm           https://pubmed.ncbi.nlm.nih.gov/33577785/  \n",
       "Hnscc         https://pubmed.ncbi.nlm.nih.gov/33417831/  \n",
       "Lscc          https://pubmed.ncbi.nlm.nih.gov/34358469/  \n",
       "Luad          https://pubmed.ncbi.nlm.nih.gov/32649874/  \n",
       "Ovarian       https://pubmed.ncbi.nlm.nih.gov/27372738/  \n",
       "Pdac          https://pubmed.ncbi.nlm.nih.gov/34534465/  \n",
       "UcecConf                                    unpublished  \n",
       "GbmConf                                     unpublished  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cptac.list_datasets()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    },
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cptac.download('Ccrcc', version='0.1.1')\n",
    "cptac.download('Endometrial', version='2.1.1')\n",
    "cptac.download('Luad', version='3.1.1')\n",
    "cptac.download('Brca', version='5.4')\n",
    "cptac.download('Gbm', version='3.0')\n",
    "cptac.download('Hnscc', version='2.0')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Clear Cell Renal Carcinoma (ccRCC)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                          \r"
     ]
    }
   ],
   "source": [
    "ccrcc = cptac.Ccrcc(version='0.1.1')\n",
    "ccrcc_clinical_info = ccrcc.get_clinical()\n",
    "interested_samples = ccrcc_clinical_info.index[ccrcc_clinical_info['histologic_type']=='Clear cell renal cell carcinoma']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (19275, 110)\n",
      "Null values count:  0\n"
     ]
    }
   ],
   "source": [
    "ccrcc_transcriptomics = get_transcriptomics(ccrcc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (16718, 103)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>C3L-00004</th>\n",
       "      <th>C3L-00010</th>\n",
       "      <th>C3L-00011</th>\n",
       "      <th>C3L-00026</th>\n",
       "      <th>C3L-00079</th>\n",
       "      <th>C3L-00088</th>\n",
       "      <th>C3L-00096</th>\n",
       "      <th>C3L-00097</th>\n",
       "      <th>C3L-00103</th>\n",
       "      <th>C3L-00183</th>\n",
       "      <th>...</th>\n",
       "      <th>C3N-01220</th>\n",
       "      <th>C3N-01261</th>\n",
       "      <th>C3N-01361</th>\n",
       "      <th>C3N-01522</th>\n",
       "      <th>C3N-01524</th>\n",
       "      <th>C3N-01646</th>\n",
       "      <th>C3N-01648</th>\n",
       "      <th>C3N-01649</th>\n",
       "      <th>C3N-01651</th>\n",
       "      <th>C3N-01808</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>0.995336</td>\n",
       "      <td>0.679400</td>\n",
       "      <td>0.354549</td>\n",
       "      <td>2.543775</td>\n",
       "      <td>4.355205</td>\n",
       "      <td>1.114256</td>\n",
       "      <td>1.624697</td>\n",
       "      <td>1.060201</td>\n",
       "      <td>1.294317</td>\n",
       "      <td>1.091372</td>\n",
       "      <td>...</td>\n",
       "      <td>1.400173</td>\n",
       "      <td>1.34950</td>\n",
       "      <td>2.578430</td>\n",
       "      <td>0.664676</td>\n",
       "      <td>4.472127</td>\n",
       "      <td>2.823319</td>\n",
       "      <td>7.008482</td>\n",
       "      <td>2.212953</td>\n",
       "      <td>1.268049</td>\n",
       "      <td>0.903522</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>16.677828</td>\n",
       "      <td>16.682712</td>\n",
       "      <td>0.245606</td>\n",
       "      <td>16.347532</td>\n",
       "      <td>4.858958</td>\n",
       "      <td>13.654469</td>\n",
       "      <td>8.107277</td>\n",
       "      <td>4.541293</td>\n",
       "      <td>1.853419</td>\n",
       "      <td>6.293220</td>\n",
       "      <td>...</td>\n",
       "      <td>0.215648</td>\n",
       "      <td>6.03394</td>\n",
       "      <td>3.416981</td>\n",
       "      <td>9.471037</td>\n",
       "      <td>8.165651</td>\n",
       "      <td>2.720128</td>\n",
       "      <td>0.018267</td>\n",
       "      <td>2.237772</td>\n",
       "      <td>13.311588</td>\n",
       "      <td>12.117981</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 103 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  C3L-00004  C3L-00010  C3L-00011  C3L-00026  C3L-00079  C3L-00088  \\\n",
       "Name                                                                           \n",
       "A1BG         0.995336   0.679400   0.354549   2.543775   4.355205   1.114256   \n",
       "A1CF        16.677828  16.682712   0.245606  16.347532   4.858958  13.654469   \n",
       "\n",
       "Patient_ID  C3L-00096  C3L-00097  C3L-00103  C3L-00183  ...  C3N-01220  \\\n",
       "Name                                                    ...              \n",
       "A1BG         1.624697   1.060201   1.294317   1.091372  ...   1.400173   \n",
       "A1CF         8.107277   4.541293   1.853419   6.293220  ...   0.215648   \n",
       "\n",
       "Patient_ID  C3N-01261  C3N-01361  C3N-01522  C3N-01524  C3N-01646  C3N-01648  \\\n",
       "Name                                                                           \n",
       "A1BG          1.34950   2.578430   0.664676   4.472127   2.823319   7.008482   \n",
       "A1CF          6.03394   3.416981   9.471037   8.165651   2.720128   0.018267   \n",
       "\n",
       "Patient_ID  C3N-01649  C3N-01651  C3N-01808  \n",
       "Name                                         \n",
       "A1BG         2.212953   1.268049   0.903522  \n",
       "A1CF         2.237772  13.311588  12.117981  \n",
       "\n",
       "[2 rows x 103 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ccrcc_transcriptomics = ccrcc_transcriptomics[interested_samples]\n",
    "ccrcc_transcriptomics_processed = standardised_pipeline_utils.process(ccrcc_transcriptomics)\n",
    "ccrcc_transcriptomics_processed[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dataframe transposed.\n",
      "Dimensions:  (11710, 110)\n",
      "Null values count:  284842\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>C3L-00004</th>\n",
       "      <th>C3L-00010</th>\n",
       "      <th>C3L-00011</th>\n",
       "      <th>C3L-00026</th>\n",
       "      <th>C3L-00079</th>\n",
       "      <th>C3L-00088</th>\n",
       "      <th>C3L-00096</th>\n",
       "      <th>C3L-00097</th>\n",
       "      <th>C3L-00103</th>\n",
       "      <th>C3L-00183</th>\n",
       "      <th>...</th>\n",
       "      <th>C3N-01220</th>\n",
       "      <th>C3N-01261</th>\n",
       "      <th>C3N-01361</th>\n",
       "      <th>C3N-01522</th>\n",
       "      <th>C3N-01524</th>\n",
       "      <th>C3N-01646</th>\n",
       "      <th>C3N-01648</th>\n",
       "      <th>C3N-01649</th>\n",
       "      <th>C3N-01651</th>\n",
       "      <th>C3N-01808</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-0.304302</td>\n",
       "      <td>1.195915</td>\n",
       "      <td>-0.286155</td>\n",
       "      <td>0.135730</td>\n",
       "      <td>-0.123959</td>\n",
       "      <td>0.427542</td>\n",
       "      <td>-0.242107</td>\n",
       "      <td>0.506469</td>\n",
       "      <td>0.720836</td>\n",
       "      <td>0.082946</td>\n",
       "      <td>...</td>\n",
       "      <td>0.791576</td>\n",
       "      <td>0.318540</td>\n",
       "      <td>0.093607</td>\n",
       "      <td>-0.504522</td>\n",
       "      <td>0.788178</td>\n",
       "      <td>-0.173487</td>\n",
       "      <td>-0.350081</td>\n",
       "      <td>0.246378</td>\n",
       "      <td>-0.242872</td>\n",
       "      <td>0.171883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>0.641447</td>\n",
       "      <td>0.194620</td>\n",
       "      <td>-0.780455</td>\n",
       "      <td>0.404286</td>\n",
       "      <td>-0.677773</td>\n",
       "      <td>0.310249</td>\n",
       "      <td>-0.128732</td>\n",
       "      <td>-0.513243</td>\n",
       "      <td>-1.135859</td>\n",
       "      <td>-0.128068</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.892166</td>\n",
       "      <td>-0.251923</td>\n",
       "      <td>-0.535844</td>\n",
       "      <td>0.087143</td>\n",
       "      <td>-0.126760</td>\n",
       "      <td>-0.686012</td>\n",
       "      <td>-0.699248</td>\n",
       "      <td>-0.847288</td>\n",
       "      <td>0.486950</td>\n",
       "      <td>0.364511</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 110 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  C3L-00004  C3L-00010  C3L-00011  C3L-00026  C3L-00079  C3L-00088  \\\n",
       "Name                                                                           \n",
       "A1BG        -0.304302   1.195915  -0.286155   0.135730  -0.123959   0.427542   \n",
       "A1CF         0.641447   0.194620  -0.780455   0.404286  -0.677773   0.310249   \n",
       "\n",
       "Patient_ID  C3L-00096  C3L-00097  C3L-00103  C3L-00183  ...  C3N-01220  \\\n",
       "Name                                                    ...              \n",
       "A1BG        -0.242107   0.506469   0.720836   0.082946  ...   0.791576   \n",
       "A1CF        -0.128732  -0.513243  -1.135859  -0.128068  ...  -0.892166   \n",
       "\n",
       "Patient_ID  C3N-01261  C3N-01361  C3N-01522  C3N-01524  C3N-01646  C3N-01648  \\\n",
       "Name                                                                           \n",
       "A1BG         0.318540   0.093607  -0.504522   0.788178  -0.173487  -0.350081   \n",
       "A1CF        -0.251923  -0.535844   0.087143  -0.126760  -0.686012  -0.699248   \n",
       "\n",
       "Patient_ID  C3N-01649  C3N-01651  C3N-01808  \n",
       "Name                                         \n",
       "A1BG         0.246378  -0.242872   0.171883  \n",
       "A1CF        -0.847288   0.486950   0.364511  \n",
       "\n",
       "[2 rows x 110 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ccrcc_proteomics = get_proteomics(ccrcc)\n",
    "ccrcc_proteomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert len(ccrcc_proteomics.columns[ccrcc_proteomics.columns.duplicated()]) == 0, \"columns contain duplicates\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-warning\">\n",
    "<b>Note:</b> Despite dropping one level of the multi-index, there are no columns/samples with duplicates.\n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (7820, 103)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>C3L-00004</th>\n",
       "      <th>C3L-00010</th>\n",
       "      <th>C3L-00011</th>\n",
       "      <th>C3L-00026</th>\n",
       "      <th>C3L-00079</th>\n",
       "      <th>C3L-00088</th>\n",
       "      <th>C3L-00096</th>\n",
       "      <th>C3L-00097</th>\n",
       "      <th>C3L-00103</th>\n",
       "      <th>C3L-00183</th>\n",
       "      <th>...</th>\n",
       "      <th>C3N-01220</th>\n",
       "      <th>C3N-01261</th>\n",
       "      <th>C3N-01361</th>\n",
       "      <th>C3N-01522</th>\n",
       "      <th>C3N-01524</th>\n",
       "      <th>C3N-01646</th>\n",
       "      <th>C3N-01648</th>\n",
       "      <th>C3N-01649</th>\n",
       "      <th>C3N-01651</th>\n",
       "      <th>C3N-01808</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-0.304302</td>\n",
       "      <td>1.195915</td>\n",
       "      <td>-0.286155</td>\n",
       "      <td>0.135730</td>\n",
       "      <td>-0.123959</td>\n",
       "      <td>0.427542</td>\n",
       "      <td>-0.242107</td>\n",
       "      <td>0.506469</td>\n",
       "      <td>0.720836</td>\n",
       "      <td>0.082946</td>\n",
       "      <td>...</td>\n",
       "      <td>0.791576</td>\n",
       "      <td>0.318540</td>\n",
       "      <td>0.093607</td>\n",
       "      <td>-0.504522</td>\n",
       "      <td>0.788178</td>\n",
       "      <td>-0.173487</td>\n",
       "      <td>-0.350081</td>\n",
       "      <td>0.246378</td>\n",
       "      <td>-0.242872</td>\n",
       "      <td>0.171883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>0.641447</td>\n",
       "      <td>0.194620</td>\n",
       "      <td>-0.780455</td>\n",
       "      <td>0.404286</td>\n",
       "      <td>-0.677773</td>\n",
       "      <td>0.310249</td>\n",
       "      <td>-0.128732</td>\n",
       "      <td>-0.513243</td>\n",
       "      <td>-1.135859</td>\n",
       "      <td>-0.128068</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.892166</td>\n",
       "      <td>-0.251923</td>\n",
       "      <td>-0.535844</td>\n",
       "      <td>0.087143</td>\n",
       "      <td>-0.126760</td>\n",
       "      <td>-0.686012</td>\n",
       "      <td>-0.699248</td>\n",
       "      <td>-0.847288</td>\n",
       "      <td>0.486950</td>\n",
       "      <td>0.364511</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 103 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  C3L-00004  C3L-00010  C3L-00011  C3L-00026  C3L-00079  C3L-00088  \\\n",
       "Name                                                                           \n",
       "A1BG        -0.304302   1.195915  -0.286155   0.135730  -0.123959   0.427542   \n",
       "A1CF         0.641447   0.194620  -0.780455   0.404286  -0.677773   0.310249   \n",
       "\n",
       "Patient_ID  C3L-00096  C3L-00097  C3L-00103  C3L-00183  ...  C3N-01220  \\\n",
       "Name                                                    ...              \n",
       "A1BG        -0.242107   0.506469   0.720836   0.082946  ...   0.791576   \n",
       "A1CF        -0.128732  -0.513243  -1.135859  -0.128068  ...  -0.892166   \n",
       "\n",
       "Patient_ID  C3N-01261  C3N-01361  C3N-01522  C3N-01524  C3N-01646  C3N-01648  \\\n",
       "Name                                                                           \n",
       "A1BG         0.318540   0.093607  -0.504522   0.788178  -0.173487  -0.350081   \n",
       "A1CF        -0.251923  -0.535844   0.087143  -0.126760  -0.686012  -0.699248   \n",
       "\n",
       "Patient_ID  C3N-01649  C3N-01651  C3N-01808  \n",
       "Name                                         \n",
       "A1BG         0.246378  -0.242872   0.171883  \n",
       "A1CF        -0.847288   0.486950   0.364511  \n",
       "\n",
       "[2 rows x 103 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ccrcc_proteomics = ccrcc_proteomics[interested_samples]\n",
    "ccrcc_proteomics_processed = standardised_pipeline_utils.process(ccrcc_proteomics)\n",
    "ccrcc_proteomics_processed[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "transcriptomic_samples['ccRCC'] = ccrcc_transcriptomics_processed.shape[1]\n",
    "proteomic_samples['ccRCC'] = ccrcc_proteomics_processed.shape[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of common proteins:  7609\n",
      "Number of common samples:  103\n"
     ]
    }
   ],
   "source": [
    "ccrcc_transcriptomics_processed, ccrcc_proteomics_processed = standardised_pipeline_utils.match_proteins_samples( \\\n",
    "                                                                            ccrcc_transcriptomics_processed, \n",
    "                                                                            ccrcc_proteomics_processed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "common_samples['ccRCC'] = ccrcc_transcriptomics_processed.shape[1]\n",
    "common_genes['ccRCC'] = ccrcc_proteomics_processed.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Spearman Correlation:  0.4103\n"
     ]
    }
   ],
   "source": [
    "correlation_ccrcc = standardised_pipeline_utils.correlate_genewise(ccrcc_transcriptomics_processed, \n",
    "                                                                    ccrcc_proteomics_processed, 'ccRCC')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Pearson Correlation:  0.4233\n"
     ]
    }
   ],
   "source": [
    "correlation_ccrcc_pearson = standardised_pipeline_utils.correlate_genewise(ccrcc_transcriptomics_processed, \n",
    "                                                                    ccrcc_proteomics_processed, 'ccRCC', method='pearson')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=Endometrial_Cancer></a>\n",
    "### Endometrial Cancer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    }
   ],
   "source": [
    "endo = cptac.Endometrial(version='2.1.1')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (28057, 95)\n",
      "Null values count:  0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>C3L-00006</th>\n",
       "      <th>C3L-00008</th>\n",
       "      <th>C3L-00032</th>\n",
       "      <th>C3L-00090</th>\n",
       "      <th>C3L-00098</th>\n",
       "      <th>C3L-00136</th>\n",
       "      <th>C3L-00137</th>\n",
       "      <th>C3L-00139</th>\n",
       "      <th>C3L-00143</th>\n",
       "      <th>C3L-00145</th>\n",
       "      <th>...</th>\n",
       "      <th>C3N-01219</th>\n",
       "      <th>C3N-01267</th>\n",
       "      <th>C3N-01346</th>\n",
       "      <th>C3N-01349</th>\n",
       "      <th>C3N-01510</th>\n",
       "      <th>C3N-01520</th>\n",
       "      <th>C3N-01521</th>\n",
       "      <th>C3N-01537</th>\n",
       "      <th>C3N-01802</th>\n",
       "      <th>C3N-01825</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>4.02</td>\n",
       "      <td>4.81</td>\n",
       "      <td>6.24</td>\n",
       "      <td>5.31</td>\n",
       "      <td>9.84</td>\n",
       "      <td>5.03</td>\n",
       "      <td>3.17</td>\n",
       "      <td>6.03</td>\n",
       "      <td>6.02</td>\n",
       "      <td>5.71</td>\n",
       "      <td>...</td>\n",
       "      <td>5.34</td>\n",
       "      <td>5.22</td>\n",
       "      <td>5.63</td>\n",
       "      <td>7.63</td>\n",
       "      <td>4.85</td>\n",
       "      <td>4.84</td>\n",
       "      <td>4.80</td>\n",
       "      <td>5.95</td>\n",
       "      <td>5.25</td>\n",
       "      <td>6.79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <td>2.16</td>\n",
       "      <td>2.21</td>\n",
       "      <td>6.43</td>\n",
       "      <td>4.87</td>\n",
       "      <td>8.83</td>\n",
       "      <td>5.59</td>\n",
       "      <td>3.56</td>\n",
       "      <td>5.46</td>\n",
       "      <td>5.90</td>\n",
       "      <td>5.43</td>\n",
       "      <td>...</td>\n",
       "      <td>5.77</td>\n",
       "      <td>5.50</td>\n",
       "      <td>6.89</td>\n",
       "      <td>7.54</td>\n",
       "      <td>3.44</td>\n",
       "      <td>4.95</td>\n",
       "      <td>5.02</td>\n",
       "      <td>5.92</td>\n",
       "      <td>5.97</td>\n",
       "      <td>6.72</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 95 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  C3L-00006  C3L-00008  C3L-00032  C3L-00090  C3L-00098  C3L-00136  \\\n",
       "Name                                                                           \n",
       "A1BG             4.02       4.81       6.24       5.31       9.84       5.03   \n",
       "A1BG-AS1         2.16       2.21       6.43       4.87       8.83       5.59   \n",
       "\n",
       "Patient_ID  C3L-00137  C3L-00139  C3L-00143  C3L-00145  ...  C3N-01219  \\\n",
       "Name                                                    ...              \n",
       "A1BG             3.17       6.03       6.02       5.71  ...       5.34   \n",
       "A1BG-AS1         3.56       5.46       5.90       5.43  ...       5.77   \n",
       "\n",
       "Patient_ID  C3N-01267  C3N-01346  C3N-01349  C3N-01510  C3N-01520  C3N-01521  \\\n",
       "Name                                                                           \n",
       "A1BG             5.22       5.63       7.63       4.85       4.84       4.80   \n",
       "A1BG-AS1         5.50       6.89       7.54       3.44       4.95       5.02   \n",
       "\n",
       "Patient_ID  C3N-01537  C3N-01802  C3N-01825  \n",
       "Name                                         \n",
       "A1BG             5.95       5.25       6.79  \n",
       "A1BG-AS1         5.92       5.97       6.72  \n",
       "\n",
       "[2 rows x 95 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "endo_transcriptomics = get_transcriptomics(endo)\n",
    "endo_transcriptomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (20807, 95)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>C3L-00006</th>\n",
       "      <th>C3L-00008</th>\n",
       "      <th>C3L-00032</th>\n",
       "      <th>C3L-00090</th>\n",
       "      <th>C3L-00098</th>\n",
       "      <th>C3L-00136</th>\n",
       "      <th>C3L-00137</th>\n",
       "      <th>C3L-00139</th>\n",
       "      <th>C3L-00143</th>\n",
       "      <th>C3L-00145</th>\n",
       "      <th>...</th>\n",
       "      <th>C3N-01219</th>\n",
       "      <th>C3N-01267</th>\n",
       "      <th>C3N-01346</th>\n",
       "      <th>C3N-01349</th>\n",
       "      <th>C3N-01510</th>\n",
       "      <th>C3N-01520</th>\n",
       "      <th>C3N-01521</th>\n",
       "      <th>C3N-01537</th>\n",
       "      <th>C3N-01802</th>\n",
       "      <th>C3N-01825</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>4.02</td>\n",
       "      <td>4.81</td>\n",
       "      <td>6.24</td>\n",
       "      <td>5.31</td>\n",
       "      <td>9.84</td>\n",
       "      <td>5.03</td>\n",
       "      <td>3.17</td>\n",
       "      <td>6.03</td>\n",
       "      <td>6.02</td>\n",
       "      <td>5.71</td>\n",
       "      <td>...</td>\n",
       "      <td>5.34</td>\n",
       "      <td>5.22</td>\n",
       "      <td>5.63</td>\n",
       "      <td>7.63</td>\n",
       "      <td>4.85</td>\n",
       "      <td>4.84</td>\n",
       "      <td>4.80</td>\n",
       "      <td>5.95</td>\n",
       "      <td>5.25</td>\n",
       "      <td>6.79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <td>2.16</td>\n",
       "      <td>2.21</td>\n",
       "      <td>6.43</td>\n",
       "      <td>4.87</td>\n",
       "      <td>8.83</td>\n",
       "      <td>5.59</td>\n",
       "      <td>3.56</td>\n",
       "      <td>5.46</td>\n",
       "      <td>5.90</td>\n",
       "      <td>5.43</td>\n",
       "      <td>...</td>\n",
       "      <td>5.77</td>\n",
       "      <td>5.50</td>\n",
       "      <td>6.89</td>\n",
       "      <td>7.54</td>\n",
       "      <td>3.44</td>\n",
       "      <td>4.95</td>\n",
       "      <td>5.02</td>\n",
       "      <td>5.92</td>\n",
       "      <td>5.97</td>\n",
       "      <td>6.72</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 95 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  C3L-00006  C3L-00008  C3L-00032  C3L-00090  C3L-00098  C3L-00136  \\\n",
       "Name                                                                           \n",
       "A1BG             4.02       4.81       6.24       5.31       9.84       5.03   \n",
       "A1BG-AS1         2.16       2.21       6.43       4.87       8.83       5.59   \n",
       "\n",
       "Patient_ID  C3L-00137  C3L-00139  C3L-00143  C3L-00145  ...  C3N-01219  \\\n",
       "Name                                                    ...              \n",
       "A1BG             3.17       6.03       6.02       5.71  ...       5.34   \n",
       "A1BG-AS1         3.56       5.46       5.90       5.43  ...       5.77   \n",
       "\n",
       "Patient_ID  C3N-01267  C3N-01346  C3N-01349  C3N-01510  C3N-01520  C3N-01521  \\\n",
       "Name                                                                           \n",
       "A1BG             5.22       5.63       7.63       4.85       4.84       4.80   \n",
       "A1BG-AS1         5.50       6.89       7.54       3.44       4.95       5.02   \n",
       "\n",
       "Patient_ID  C3N-01537  C3N-01802  C3N-01825  \n",
       "Name                                         \n",
       "A1BG             5.95       5.25       6.79  \n",
       "A1BG-AS1         5.92       5.97       6.72  \n",
       "\n",
       "[2 rows x 95 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "endo_transcriptomics_processed = standardised_pipeline_utils.process(endo_transcriptomics)\n",
    "endo_transcriptomics_processed[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dataframe transposed.\n",
      "Dimensions:  (10999, 95)\n",
      "Null values count:  116089\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>C3L-00006</th>\n",
       "      <th>C3L-00008</th>\n",
       "      <th>C3L-00032</th>\n",
       "      <th>C3L-00090</th>\n",
       "      <th>C3L-00098</th>\n",
       "      <th>C3L-00136</th>\n",
       "      <th>C3L-00137</th>\n",
       "      <th>C3L-00139</th>\n",
       "      <th>C3L-00143</th>\n",
       "      <th>C3L-00145</th>\n",
       "      <th>...</th>\n",
       "      <th>C3N-01219</th>\n",
       "      <th>C3N-01267</th>\n",
       "      <th>C3N-01346</th>\n",
       "      <th>C3N-01349</th>\n",
       "      <th>C3N-01510</th>\n",
       "      <th>C3N-01520</th>\n",
       "      <th>C3N-01521</th>\n",
       "      <th>C3N-01537</th>\n",
       "      <th>C3N-01802</th>\n",
       "      <th>C3N-01825</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-1.180</td>\n",
       "      <td>-0.685</td>\n",
       "      <td>-0.528</td>\n",
       "      <td>-1.67</td>\n",
       "      <td>-0.3740</td>\n",
       "      <td>-1.080</td>\n",
       "      <td>-1.320</td>\n",
       "      <td>-0.467</td>\n",
       "      <td>-1.12</td>\n",
       "      <td>-0.716</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.2950</td>\n",
       "      <td>-1.30</td>\n",
       "      <td>-0.67</td>\n",
       "      <td>0.687</td>\n",
       "      <td>-0.269</td>\n",
       "      <td>-1.070</td>\n",
       "      <td>-1.280</td>\n",
       "      <td>-0.29</td>\n",
       "      <td>0.266</td>\n",
       "      <td>0.692</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>-0.863</td>\n",
       "      <td>-1.070</td>\n",
       "      <td>-1.320</td>\n",
       "      <td>-1.19</td>\n",
       "      <td>-0.0206</td>\n",
       "      <td>-0.708</td>\n",
       "      <td>-0.708</td>\n",
       "      <td>0.370</td>\n",
       "      <td>-1.31</td>\n",
       "      <td>-0.885</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0589</td>\n",
       "      <td>-1.29</td>\n",
       "      <td>-1.11</td>\n",
       "      <td>1.440</td>\n",
       "      <td>0.944</td>\n",
       "      <td>-0.712</td>\n",
       "      <td>-0.736</td>\n",
       "      <td>-0.32</td>\n",
       "      <td>1.390</td>\n",
       "      <td>0.589</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 95 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  C3L-00006  C3L-00008  C3L-00032  C3L-00090  C3L-00098  C3L-00136  \\\n",
       "Name                                                                           \n",
       "A1BG           -1.180     -0.685     -0.528      -1.67    -0.3740     -1.080   \n",
       "A2M            -0.863     -1.070     -1.320      -1.19    -0.0206     -0.708   \n",
       "\n",
       "Patient_ID  C3L-00137  C3L-00139  C3L-00143  C3L-00145  ...  C3N-01219  \\\n",
       "Name                                                    ...              \n",
       "A1BG           -1.320     -0.467      -1.12     -0.716  ...    -0.2950   \n",
       "A2M            -0.708      0.370      -1.31     -0.885  ...    -0.0589   \n",
       "\n",
       "Patient_ID  C3N-01267  C3N-01346  C3N-01349  C3N-01510  C3N-01520  C3N-01521  \\\n",
       "Name                                                                           \n",
       "A1BG            -1.30      -0.67      0.687     -0.269     -1.070     -1.280   \n",
       "A2M             -1.29      -1.11      1.440      0.944     -0.712     -0.736   \n",
       "\n",
       "Patient_ID  C3N-01537  C3N-01802  C3N-01825  \n",
       "Name                                         \n",
       "A1BG            -0.29      0.266      0.692  \n",
       "A2M             -0.32      1.390      0.589  \n",
       "\n",
       "[2 rows x 95 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "endo_proteomics = get_proteomics(endo, multiIndexed = False)\n",
    "endo_proteomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (9099, 95)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>C3L-00006</th>\n",
       "      <th>C3L-00008</th>\n",
       "      <th>C3L-00032</th>\n",
       "      <th>C3L-00090</th>\n",
       "      <th>C3L-00098</th>\n",
       "      <th>C3L-00136</th>\n",
       "      <th>C3L-00137</th>\n",
       "      <th>C3L-00139</th>\n",
       "      <th>C3L-00143</th>\n",
       "      <th>C3L-00145</th>\n",
       "      <th>...</th>\n",
       "      <th>C3N-01219</th>\n",
       "      <th>C3N-01267</th>\n",
       "      <th>C3N-01346</th>\n",
       "      <th>C3N-01349</th>\n",
       "      <th>C3N-01510</th>\n",
       "      <th>C3N-01520</th>\n",
       "      <th>C3N-01521</th>\n",
       "      <th>C3N-01537</th>\n",
       "      <th>C3N-01802</th>\n",
       "      <th>C3N-01825</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-1.180</td>\n",
       "      <td>-0.685</td>\n",
       "      <td>-0.528</td>\n",
       "      <td>-1.67</td>\n",
       "      <td>-0.3740</td>\n",
       "      <td>-1.080</td>\n",
       "      <td>-1.320</td>\n",
       "      <td>-0.467</td>\n",
       "      <td>-1.12</td>\n",
       "      <td>-0.716</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.2950</td>\n",
       "      <td>-1.30</td>\n",
       "      <td>-0.67</td>\n",
       "      <td>0.687</td>\n",
       "      <td>-0.269</td>\n",
       "      <td>-1.070</td>\n",
       "      <td>-1.280</td>\n",
       "      <td>-0.29</td>\n",
       "      <td>0.266</td>\n",
       "      <td>0.692</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>-0.863</td>\n",
       "      <td>-1.070</td>\n",
       "      <td>-1.320</td>\n",
       "      <td>-1.19</td>\n",
       "      <td>-0.0206</td>\n",
       "      <td>-0.708</td>\n",
       "      <td>-0.708</td>\n",
       "      <td>0.370</td>\n",
       "      <td>-1.31</td>\n",
       "      <td>-0.885</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0589</td>\n",
       "      <td>-1.29</td>\n",
       "      <td>-1.11</td>\n",
       "      <td>1.440</td>\n",
       "      <td>0.944</td>\n",
       "      <td>-0.712</td>\n",
       "      <td>-0.736</td>\n",
       "      <td>-0.32</td>\n",
       "      <td>1.390</td>\n",
       "      <td>0.589</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 95 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  C3L-00006  C3L-00008  C3L-00032  C3L-00090  C3L-00098  C3L-00136  \\\n",
       "Name                                                                           \n",
       "A1BG           -1.180     -0.685     -0.528      -1.67    -0.3740     -1.080   \n",
       "A2M            -0.863     -1.070     -1.320      -1.19    -0.0206     -0.708   \n",
       "\n",
       "Patient_ID  C3L-00137  C3L-00139  C3L-00143  C3L-00145  ...  C3N-01219  \\\n",
       "Name                                                    ...              \n",
       "A1BG           -1.320     -0.467      -1.12     -0.716  ...    -0.2950   \n",
       "A2M            -0.708      0.370      -1.31     -0.885  ...    -0.0589   \n",
       "\n",
       "Patient_ID  C3N-01267  C3N-01346  C3N-01349  C3N-01510  C3N-01520  C3N-01521  \\\n",
       "Name                                                                           \n",
       "A1BG            -1.30      -0.67      0.687     -0.269     -1.070     -1.280   \n",
       "A2M             -1.29      -1.11      1.440      0.944     -0.712     -0.736   \n",
       "\n",
       "Patient_ID  C3N-01537  C3N-01802  C3N-01825  \n",
       "Name                                         \n",
       "A1BG            -0.29      0.266      0.692  \n",
       "A2M             -0.32      1.390      0.589  \n",
       "\n",
       "[2 rows x 95 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "endo_proteomics_processed = standardised_pipeline_utils.process(endo_proteomics)\n",
    "endo_proteomics_processed[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "transcriptomic_samples['EC'] = endo_transcriptomics_processed.shape[1]\n",
    "proteomic_samples['EC'] = endo_proteomics_processed.shape[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of common proteins:  8998\n",
      "Number of common samples:  95\n"
     ]
    }
   ],
   "source": [
    "endo_transcriptomics_processed, endo_proteomics_processed = standardised_pipeline_utils.match_proteins_samples( \\\n",
    "                                                                            endo_transcriptomics_processed, \n",
    "                                                                            endo_proteomics_processed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "common_samples['EC'] = endo_transcriptomics_processed.shape[1]\n",
    "common_genes['EC'] = endo_transcriptomics_processed.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Spearman Correlation:  0.4839\n"
     ]
    }
   ],
   "source": [
    "correlation_endo = standardised_pipeline_utils.correlate_genewise(endo_transcriptomics_processed, \n",
    "                                                                    endo_proteomics_processed, 'EC')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Pearson Correlation:  0.5116\n"
     ]
    }
   ],
   "source": [
    "correlation_endo_pearson = standardised_pipeline_utils.correlate_genewise(endo_transcriptomics_processed, \n",
    "                                                                    endo_proteomics_processed, 'EC', method='pearson')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"Luad\"></a>\n",
    "### Lung Adenocarcinoma"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         \r"
     ]
    }
   ],
   "source": [
    "lung_adenocarcinoma = cptac.Luad(version='3.1.1')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (18099, 110)\n",
      "Null values count:  64320\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>C3L-00001</th>\n",
       "      <th>C3L-00009</th>\n",
       "      <th>C3L-00080</th>\n",
       "      <th>C3L-00083</th>\n",
       "      <th>C3L-00093</th>\n",
       "      <th>C3L-00094</th>\n",
       "      <th>C3L-00095</th>\n",
       "      <th>C3L-00140</th>\n",
       "      <th>C3L-00144</th>\n",
       "      <th>C3L-00263</th>\n",
       "      <th>...</th>\n",
       "      <th>C3N-02572</th>\n",
       "      <th>C3N-02582</th>\n",
       "      <th>C3N-02586</th>\n",
       "      <th>C3N-02587</th>\n",
       "      <th>C3N-02588</th>\n",
       "      <th>C3N-02729</th>\n",
       "      <th>X11LU013</th>\n",
       "      <th>X11LU016</th>\n",
       "      <th>X11LU022</th>\n",
       "      <th>X11LU035</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>2.2545</td>\n",
       "      <td>1.4770</td>\n",
       "      <td>1.5103</td>\n",
       "      <td>3.0398</td>\n",
       "      <td>1.7528</td>\n",
       "      <td>0.5742</td>\n",
       "      <td>2.2706</td>\n",
       "      <td>0.9331</td>\n",
       "      <td>1.0906</td>\n",
       "      <td>1.1381</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0492</td>\n",
       "      <td>2.3424</td>\n",
       "      <td>1.6608</td>\n",
       "      <td>0.6747</td>\n",
       "      <td>0.7923</td>\n",
       "      <td>-0.1409</td>\n",
       "      <td>0.7896</td>\n",
       "      <td>0.6455</td>\n",
       "      <td>1.6085</td>\n",
       "      <td>0.0421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>-2.7845</td>\n",
       "      <td>-1.9278</td>\n",
       "      <td>-4.9913</td>\n",
       "      <td>-3.5440</td>\n",
       "      <td>-5.2883</td>\n",
       "      <td>-3.6693</td>\n",
       "      <td>-2.2618</td>\n",
       "      <td>-3.7772</td>\n",
       "      <td>-5.1850</td>\n",
       "      <td>-1.0380</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.9021</td>\n",
       "      <td>-5.5138</td>\n",
       "      <td>-4.7460</td>\n",
       "      <td>-3.9404</td>\n",
       "      <td>-4.3875</td>\n",
       "      <td>-6.2414</td>\n",
       "      <td>-5.1965</td>\n",
       "      <td>-5.9792</td>\n",
       "      <td>-3.4240</td>\n",
       "      <td>-5.5014</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 110 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  C3L-00001  C3L-00009  C3L-00080  C3L-00083  C3L-00093  C3L-00094  \\\n",
       "Name                                                                           \n",
       "A1BG           2.2545     1.4770     1.5103     3.0398     1.7528     0.5742   \n",
       "A1CF          -2.7845    -1.9278    -4.9913    -3.5440    -5.2883    -3.6693   \n",
       "\n",
       "Patient_ID  C3L-00095  C3L-00140  C3L-00144  C3L-00263  ...  C3N-02572  \\\n",
       "Name                                                    ...              \n",
       "A1BG           2.2706     0.9331     1.0906     1.1381  ...     1.0492   \n",
       "A1CF          -2.2618    -3.7772    -5.1850    -1.0380  ...    -2.9021   \n",
       "\n",
       "Patient_ID  C3N-02582  C3N-02586  C3N-02587  C3N-02588  C3N-02729  X11LU013  \\\n",
       "Name                                                                          \n",
       "A1BG           2.3424     1.6608     0.6747     0.7923    -0.1409    0.7896   \n",
       "A1CF          -5.5138    -4.7460    -3.9404    -4.3875    -6.2414   -5.1965   \n",
       "\n",
       "Patient_ID  X11LU016  X11LU022  X11LU035  \n",
       "Name                                      \n",
       "A1BG          0.6455    1.6085    0.0421  \n",
       "A1CF         -5.9792   -3.4240   -5.5014  \n",
       "\n",
       "[2 rows x 110 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "luad_transcriptomics = get_transcriptomics(lung_adenocarcinoma)\n",
    "luad_transcriptomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert len(luad_transcriptomics.columns[luad_transcriptomics.columns.duplicated()]) == 0, \"columns contain duplicates\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (17022, 110)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>C3L-00001</th>\n",
       "      <th>C3L-00009</th>\n",
       "      <th>C3L-00080</th>\n",
       "      <th>C3L-00083</th>\n",
       "      <th>C3L-00093</th>\n",
       "      <th>C3L-00094</th>\n",
       "      <th>C3L-00095</th>\n",
       "      <th>C3L-00140</th>\n",
       "      <th>C3L-00144</th>\n",
       "      <th>C3L-00263</th>\n",
       "      <th>...</th>\n",
       "      <th>C3N-02572</th>\n",
       "      <th>C3N-02582</th>\n",
       "      <th>C3N-02586</th>\n",
       "      <th>C3N-02587</th>\n",
       "      <th>C3N-02588</th>\n",
       "      <th>C3N-02729</th>\n",
       "      <th>X11LU013</th>\n",
       "      <th>X11LU016</th>\n",
       "      <th>X11LU022</th>\n",
       "      <th>X11LU035</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>2.2545</td>\n",
       "      <td>1.4770</td>\n",
       "      <td>1.5103</td>\n",
       "      <td>3.0398</td>\n",
       "      <td>1.7528</td>\n",
       "      <td>0.5742</td>\n",
       "      <td>2.2706</td>\n",
       "      <td>0.9331</td>\n",
       "      <td>1.0906</td>\n",
       "      <td>1.1381</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0492</td>\n",
       "      <td>2.3424</td>\n",
       "      <td>1.6608</td>\n",
       "      <td>0.6747</td>\n",
       "      <td>0.7923</td>\n",
       "      <td>-0.1409</td>\n",
       "      <td>0.7896</td>\n",
       "      <td>0.6455</td>\n",
       "      <td>1.6085</td>\n",
       "      <td>0.0421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>-2.7845</td>\n",
       "      <td>-1.9278</td>\n",
       "      <td>-4.9913</td>\n",
       "      <td>-3.5440</td>\n",
       "      <td>-5.2883</td>\n",
       "      <td>-3.6693</td>\n",
       "      <td>-2.2618</td>\n",
       "      <td>-3.7772</td>\n",
       "      <td>-5.1850</td>\n",
       "      <td>-1.0380</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.9021</td>\n",
       "      <td>-5.5138</td>\n",
       "      <td>-4.7460</td>\n",
       "      <td>-3.9404</td>\n",
       "      <td>-4.3875</td>\n",
       "      <td>-6.2414</td>\n",
       "      <td>-5.1965</td>\n",
       "      <td>-5.9792</td>\n",
       "      <td>-3.4240</td>\n",
       "      <td>-5.5014</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 110 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  C3L-00001  C3L-00009  C3L-00080  C3L-00083  C3L-00093  C3L-00094  \\\n",
       "Name                                                                           \n",
       "A1BG           2.2545     1.4770     1.5103     3.0398     1.7528     0.5742   \n",
       "A1CF          -2.7845    -1.9278    -4.9913    -3.5440    -5.2883    -3.6693   \n",
       "\n",
       "Patient_ID  C3L-00095  C3L-00140  C3L-00144  C3L-00263  ...  C3N-02572  \\\n",
       "Name                                                    ...              \n",
       "A1BG           2.2706     0.9331     1.0906     1.1381  ...     1.0492   \n",
       "A1CF          -2.2618    -3.7772    -5.1850    -1.0380  ...    -2.9021   \n",
       "\n",
       "Patient_ID  C3N-02582  C3N-02586  C3N-02587  C3N-02588  C3N-02729  X11LU013  \\\n",
       "Name                                                                          \n",
       "A1BG           2.3424     1.6608     0.6747     0.7923    -0.1409    0.7896   \n",
       "A1CF          -5.5138    -4.7460    -3.9404    -4.3875    -6.2414   -5.1965   \n",
       "\n",
       "Patient_ID  X11LU016  X11LU022  X11LU035  \n",
       "Name                                      \n",
       "A1BG          0.6455    1.6085    0.0421  \n",
       "A1CF         -5.9792   -3.4240   -5.5014  \n",
       "\n",
       "[2 rows x 110 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "luad_transcriptomics_processed = standardised_pipeline_utils.process(luad_transcriptomics)\n",
    "luad_transcriptomics_processed[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dataframe transposed.\n",
      "Dimensions:  (10699, 110)\n",
      "Null values count:  90431\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>C3L-00001</th>\n",
       "      <th>C3L-00009</th>\n",
       "      <th>C3L-00080</th>\n",
       "      <th>C3L-00083</th>\n",
       "      <th>C3L-00093</th>\n",
       "      <th>C3L-00094</th>\n",
       "      <th>C3L-00095</th>\n",
       "      <th>C3L-00140</th>\n",
       "      <th>C3L-00144</th>\n",
       "      <th>C3L-00263</th>\n",
       "      <th>...</th>\n",
       "      <th>C3N-02572</th>\n",
       "      <th>C3N-02582</th>\n",
       "      <th>C3N-02586</th>\n",
       "      <th>C3N-02587</th>\n",
       "      <th>C3N-02588</th>\n",
       "      <th>C3N-02729</th>\n",
       "      <th>X11LU013</th>\n",
       "      <th>X11LU016</th>\n",
       "      <th>X11LU022</th>\n",
       "      <th>X11LU035</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-2.5347</td>\n",
       "      <td>-0.5627</td>\n",
       "      <td>-1.9422</td>\n",
       "      <td>2.1636</td>\n",
       "      <td>-1.0022</td>\n",
       "      <td>-1.5576</td>\n",
       "      <td>-1.0718</td>\n",
       "      <td>-1.0799</td>\n",
       "      <td>-1.9159</td>\n",
       "      <td>-1.1384</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.4006</td>\n",
       "      <td>-3.3718</td>\n",
       "      <td>-1.2578</td>\n",
       "      <td>-1.1300</td>\n",
       "      <td>-1.6323</td>\n",
       "      <td>-0.7255</td>\n",
       "      <td>-1.3882</td>\n",
       "      <td>-1.4884</td>\n",
       "      <td>-0.3318</td>\n",
       "      <td>-0.7338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>-3.4057</td>\n",
       "      <td>-1.7945</td>\n",
       "      <td>-2.3782</td>\n",
       "      <td>3.1227</td>\n",
       "      <td>-0.9632</td>\n",
       "      <td>-3.0225</td>\n",
       "      <td>-3.1204</td>\n",
       "      <td>-0.7682</td>\n",
       "      <td>-3.3380</td>\n",
       "      <td>-2.0141</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.4726</td>\n",
       "      <td>-4.1354</td>\n",
       "      <td>-3.0975</td>\n",
       "      <td>-1.7842</td>\n",
       "      <td>-2.8213</td>\n",
       "      <td>-3.2235</td>\n",
       "      <td>-2.4728</td>\n",
       "      <td>-3.4264</td>\n",
       "      <td>-1.1635</td>\n",
       "      <td>-1.8498</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 110 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  C3L-00001  C3L-00009  C3L-00080  C3L-00083  C3L-00093  C3L-00094  \\\n",
       "Name                                                                           \n",
       "A1BG          -2.5347    -0.5627    -1.9422     2.1636    -1.0022    -1.5576   \n",
       "A2M           -3.4057    -1.7945    -2.3782     3.1227    -0.9632    -3.0225   \n",
       "\n",
       "Patient_ID  C3L-00095  C3L-00140  C3L-00144  C3L-00263  ...  C3N-02572  \\\n",
       "Name                                                    ...              \n",
       "A1BG          -1.0718    -1.0799    -1.9159    -1.1384  ...    -1.4006   \n",
       "A2M           -3.1204    -0.7682    -3.3380    -2.0141  ...    -3.4726   \n",
       "\n",
       "Patient_ID  C3N-02582  C3N-02586  C3N-02587  C3N-02588  C3N-02729  X11LU013  \\\n",
       "Name                                                                          \n",
       "A1BG          -3.3718    -1.2578    -1.1300    -1.6323    -0.7255   -1.3882   \n",
       "A2M           -4.1354    -3.0975    -1.7842    -2.8213    -3.2235   -2.4728   \n",
       "\n",
       "Patient_ID  X11LU016  X11LU022  X11LU035  \n",
       "Name                                      \n",
       "A1BG         -1.4884   -0.3318   -0.7338  \n",
       "A2M          -3.4264   -1.1635   -1.8498  \n",
       "\n",
       "[2 rows x 110 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "luad_proteomics = get_proteomics(lung_adenocarcinoma)\n",
    "luad_proteomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert len(luad_proteomics.columns[luad_proteomics.columns.duplicated()]) == 0, \"columns contain duplicates\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (8758, 110)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>C3L-00001</th>\n",
       "      <th>C3L-00009</th>\n",
       "      <th>C3L-00080</th>\n",
       "      <th>C3L-00083</th>\n",
       "      <th>C3L-00093</th>\n",
       "      <th>C3L-00094</th>\n",
       "      <th>C3L-00095</th>\n",
       "      <th>C3L-00140</th>\n",
       "      <th>C3L-00144</th>\n",
       "      <th>C3L-00263</th>\n",
       "      <th>...</th>\n",
       "      <th>C3N-02572</th>\n",
       "      <th>C3N-02582</th>\n",
       "      <th>C3N-02586</th>\n",
       "      <th>C3N-02587</th>\n",
       "      <th>C3N-02588</th>\n",
       "      <th>C3N-02729</th>\n",
       "      <th>X11LU013</th>\n",
       "      <th>X11LU016</th>\n",
       "      <th>X11LU022</th>\n",
       "      <th>X11LU035</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-2.5347</td>\n",
       "      <td>-0.5627</td>\n",
       "      <td>-1.9422</td>\n",
       "      <td>2.1636</td>\n",
       "      <td>-1.0022</td>\n",
       "      <td>-1.5576</td>\n",
       "      <td>-1.0718</td>\n",
       "      <td>-1.0799</td>\n",
       "      <td>-1.9159</td>\n",
       "      <td>-1.1384</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.4006</td>\n",
       "      <td>-3.3718</td>\n",
       "      <td>-1.2578</td>\n",
       "      <td>-1.1300</td>\n",
       "      <td>-1.6323</td>\n",
       "      <td>-0.7255</td>\n",
       "      <td>-1.3882</td>\n",
       "      <td>-1.4884</td>\n",
       "      <td>-0.3318</td>\n",
       "      <td>-0.7338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>-3.4057</td>\n",
       "      <td>-1.7945</td>\n",
       "      <td>-2.3782</td>\n",
       "      <td>3.1227</td>\n",
       "      <td>-0.9632</td>\n",
       "      <td>-3.0225</td>\n",
       "      <td>-3.1204</td>\n",
       "      <td>-0.7682</td>\n",
       "      <td>-3.3380</td>\n",
       "      <td>-2.0141</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.4726</td>\n",
       "      <td>-4.1354</td>\n",
       "      <td>-3.0975</td>\n",
       "      <td>-1.7842</td>\n",
       "      <td>-2.8213</td>\n",
       "      <td>-3.2235</td>\n",
       "      <td>-2.4728</td>\n",
       "      <td>-3.4264</td>\n",
       "      <td>-1.1635</td>\n",
       "      <td>-1.8498</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 110 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  C3L-00001  C3L-00009  C3L-00080  C3L-00083  C3L-00093  C3L-00094  \\\n",
       "Name                                                                           \n",
       "A1BG          -2.5347    -0.5627    -1.9422     2.1636    -1.0022    -1.5576   \n",
       "A2M           -3.4057    -1.7945    -2.3782     3.1227    -0.9632    -3.0225   \n",
       "\n",
       "Patient_ID  C3L-00095  C3L-00140  C3L-00144  C3L-00263  ...  C3N-02572  \\\n",
       "Name                                                    ...              \n",
       "A1BG          -1.0718    -1.0799    -1.9159    -1.1384  ...    -1.4006   \n",
       "A2M           -3.1204    -0.7682    -3.3380    -2.0141  ...    -3.4726   \n",
       "\n",
       "Patient_ID  C3N-02582  C3N-02586  C3N-02587  C3N-02588  C3N-02729  X11LU013  \\\n",
       "Name                                                                          \n",
       "A1BG          -3.3718    -1.2578    -1.1300    -1.6323    -0.7255   -1.3882   \n",
       "A2M           -4.1354    -3.0975    -1.7842    -2.8213    -3.2235   -2.4728   \n",
       "\n",
       "Patient_ID  X11LU016  X11LU022  X11LU035  \n",
       "Name                                      \n",
       "A1BG         -1.4884   -0.3318   -0.7338  \n",
       "A2M          -3.4264   -1.1635   -1.8498  \n",
       "\n",
       "[2 rows x 110 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "luad_proteomics_processed = standardised_pipeline_utils.process(luad_proteomics)\n",
    "luad_proteomics_processed[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "transcriptomic_samples['LUAD'] = luad_transcriptomics_processed.shape[1]\n",
    "proteomic_samples['LUAD'] = luad_proteomics_processed.shape[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of common proteins:  8507\n",
      "Number of common samples:  110\n"
     ]
    }
   ],
   "source": [
    "luad_transcriptomics_processed, luad_proteomics_processed = standardised_pipeline_utils.match_proteins_samples( \\\n",
    "                                                                            luad_transcriptomics_processed, \n",
    "                                                                            luad_proteomics_processed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "common_samples['LUAD'] = luad_transcriptomics_processed.shape[1]\n",
    "common_genes['LUAD'] = luad_transcriptomics_processed.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Spearman Correlation:  0.5465\n"
     ]
    }
   ],
   "source": [
    "correlation_luad = standardised_pipeline_utils.correlate_genewise(luad_transcriptomics_processed, \n",
    "                                                                   luad_proteomics_processed, 'LUAD')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Pearson Correlation:  0.5618\n"
     ]
    }
   ],
   "source": [
    "correlation_luad_pearson = standardised_pipeline_utils.correlate_genewise(luad_transcriptomics_processed, \n",
    "                                                                   luad_proteomics_processed, 'LUAD', method='pearson')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"Breast_Cancer\"></a>\n",
    "### Breast Cancer (2020)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         \r"
     ]
    }
   ],
   "source": [
    "brca = cptac.Brca(version='5.4')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (23121, 122)\n",
      "Null values count:  608161\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>CPT000814</th>\n",
       "      <th>CPT001846</th>\n",
       "      <th>X01BR001</th>\n",
       "      <th>X01BR008</th>\n",
       "      <th>X01BR009</th>\n",
       "      <th>X01BR010</th>\n",
       "      <th>X01BR015</th>\n",
       "      <th>X01BR017</th>\n",
       "      <th>X01BR018</th>\n",
       "      <th>X01BR020</th>\n",
       "      <th>...</th>\n",
       "      <th>X20BR002</th>\n",
       "      <th>X20BR005</th>\n",
       "      <th>X20BR006</th>\n",
       "      <th>X20BR007</th>\n",
       "      <th>X20BR008</th>\n",
       "      <th>X21BR001</th>\n",
       "      <th>X21BR002</th>\n",
       "      <th>X21BR010</th>\n",
       "      <th>X22BR005</th>\n",
       "      <th>X22BR006</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>1.9265</td>\n",
       "      <td>3.6578</td>\n",
       "      <td>0.9896</td>\n",
       "      <td>0.5535</td>\n",
       "      <td>2.8359</td>\n",
       "      <td>1.5804</td>\n",
       "      <td>1.9006</td>\n",
       "      <td>-0.8184</td>\n",
       "      <td>-0.2645</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>4.7120</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.1736</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3261</td>\n",
       "      <td>-1.2102</td>\n",
       "      <td>0.7403</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <td>2.4267</td>\n",
       "      <td>2.6524</td>\n",
       "      <td>2.6363</td>\n",
       "      <td>2.2119</td>\n",
       "      <td>3.3449</td>\n",
       "      <td>2.1647</td>\n",
       "      <td>2.5487</td>\n",
       "      <td>-0.3528</td>\n",
       "      <td>1.3557</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1.9708</td>\n",
       "      <td>1.7106</td>\n",
       "      <td>0.6634</td>\n",
       "      <td>0.3475</td>\n",
       "      <td>1.3309</td>\n",
       "      <td>2.1405</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0329</td>\n",
       "      <td>0.6457</td>\n",
       "      <td>1.6475</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 122 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  CPT000814  CPT001846  X01BR001  X01BR008  X01BR009  X01BR010  \\\n",
       "Name                                                                       \n",
       "A1BG           1.9265     3.6578    0.9896    0.5535    2.8359    1.5804   \n",
       "A1BG-AS1       2.4267     2.6524    2.6363    2.2119    3.3449    2.1647   \n",
       "\n",
       "Patient_ID  X01BR015  X01BR017  X01BR018  X01BR020  ...  X20BR002  X20BR005  \\\n",
       "Name                                                ...                       \n",
       "A1BG          1.9006   -0.8184   -0.2645       NaN  ...    4.7120       NaN   \n",
       "A1BG-AS1      2.5487   -0.3528    1.3557       NaN  ...    1.9708    1.7106   \n",
       "\n",
       "Patient_ID  X20BR006  X20BR007  X20BR008  X21BR001  X21BR002  X21BR010  \\\n",
       "Name                                                                     \n",
       "A1BG             NaN       NaN       NaN    2.1736       NaN   -0.3261   \n",
       "A1BG-AS1      0.6634    0.3475    1.3309    2.1405       NaN    1.0329   \n",
       "\n",
       "Patient_ID  X22BR005  X22BR006  \n",
       "Name                            \n",
       "A1BG         -1.2102    0.7403  \n",
       "A1BG-AS1      0.6457    1.6475  \n",
       "\n",
       "[2 rows x 122 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca_transcriptomics = get_transcriptomics(brca)\n",
    "brca_transcriptomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (16409, 122)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>CPT000814</th>\n",
       "      <th>CPT001846</th>\n",
       "      <th>X01BR001</th>\n",
       "      <th>X01BR008</th>\n",
       "      <th>X01BR009</th>\n",
       "      <th>X01BR010</th>\n",
       "      <th>X01BR015</th>\n",
       "      <th>X01BR017</th>\n",
       "      <th>X01BR018</th>\n",
       "      <th>X01BR020</th>\n",
       "      <th>...</th>\n",
       "      <th>X20BR002</th>\n",
       "      <th>X20BR005</th>\n",
       "      <th>X20BR006</th>\n",
       "      <th>X20BR007</th>\n",
       "      <th>X20BR008</th>\n",
       "      <th>X21BR001</th>\n",
       "      <th>X21BR002</th>\n",
       "      <th>X21BR010</th>\n",
       "      <th>X22BR005</th>\n",
       "      <th>X22BR006</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>1.9265</td>\n",
       "      <td>3.6578</td>\n",
       "      <td>0.9896</td>\n",
       "      <td>0.5535</td>\n",
       "      <td>2.8359</td>\n",
       "      <td>1.5804</td>\n",
       "      <td>1.9006</td>\n",
       "      <td>-0.8184</td>\n",
       "      <td>-0.2645</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>4.7120</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.1736</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3261</td>\n",
       "      <td>-1.2102</td>\n",
       "      <td>0.7403</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <td>2.4267</td>\n",
       "      <td>2.6524</td>\n",
       "      <td>2.6363</td>\n",
       "      <td>2.2119</td>\n",
       "      <td>3.3449</td>\n",
       "      <td>2.1647</td>\n",
       "      <td>2.5487</td>\n",
       "      <td>-0.3528</td>\n",
       "      <td>1.3557</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1.9708</td>\n",
       "      <td>1.7106</td>\n",
       "      <td>0.6634</td>\n",
       "      <td>0.3475</td>\n",
       "      <td>1.3309</td>\n",
       "      <td>2.1405</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0329</td>\n",
       "      <td>0.6457</td>\n",
       "      <td>1.6475</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 122 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  CPT000814  CPT001846  X01BR001  X01BR008  X01BR009  X01BR010  \\\n",
       "Name                                                                       \n",
       "A1BG           1.9265     3.6578    0.9896    0.5535    2.8359    1.5804   \n",
       "A1BG-AS1       2.4267     2.6524    2.6363    2.2119    3.3449    2.1647   \n",
       "\n",
       "Patient_ID  X01BR015  X01BR017  X01BR018  X01BR020  ...  X20BR002  X20BR005  \\\n",
       "Name                                                ...                       \n",
       "A1BG          1.9006   -0.8184   -0.2645       NaN  ...    4.7120       NaN   \n",
       "A1BG-AS1      2.5487   -0.3528    1.3557       NaN  ...    1.9708    1.7106   \n",
       "\n",
       "Patient_ID  X20BR006  X20BR007  X20BR008  X21BR001  X21BR002  X21BR010  \\\n",
       "Name                                                                     \n",
       "A1BG             NaN       NaN       NaN    2.1736       NaN   -0.3261   \n",
       "A1BG-AS1      0.6634    0.3475    1.3309    2.1405       NaN    1.0329   \n",
       "\n",
       "Patient_ID  X22BR005  X22BR006  \n",
       "Name                            \n",
       "A1BG         -1.2102    0.7403  \n",
       "A1BG-AS1      0.6457    1.6475  \n",
       "\n",
       "[2 rows x 122 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca_transcriptomics_processed = standardised_pipeline_utils.process(brca_transcriptomics)\n",
    "brca_transcriptomics_processed[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dataframe transposed.\n",
      "Dimensions:  (10107, 122)\n",
      "Null values count:  65248\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>CPT000814</th>\n",
       "      <th>CPT001846</th>\n",
       "      <th>X01BR001</th>\n",
       "      <th>X01BR008</th>\n",
       "      <th>X01BR009</th>\n",
       "      <th>X01BR010</th>\n",
       "      <th>X01BR015</th>\n",
       "      <th>X01BR017</th>\n",
       "      <th>X01BR018</th>\n",
       "      <th>X01BR020</th>\n",
       "      <th>...</th>\n",
       "      <th>X20BR002</th>\n",
       "      <th>X20BR005</th>\n",
       "      <th>X20BR006</th>\n",
       "      <th>X20BR007</th>\n",
       "      <th>X20BR008</th>\n",
       "      <th>X21BR001</th>\n",
       "      <th>X21BR002</th>\n",
       "      <th>X21BR010</th>\n",
       "      <th>X22BR005</th>\n",
       "      <th>X22BR006</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-0.6712</td>\n",
       "      <td>1.3964</td>\n",
       "      <td>2.0219</td>\n",
       "      <td>-0.5290</td>\n",
       "      <td>1.2556</td>\n",
       "      <td>-0.3843</td>\n",
       "      <td>1.0394</td>\n",
       "      <td>1.1533</td>\n",
       "      <td>1.9579</td>\n",
       "      <td>-0.1637</td>\n",
       "      <td>...</td>\n",
       "      <td>1.8732</td>\n",
       "      <td>-0.4227</td>\n",
       "      <td>1.5862</td>\n",
       "      <td>-0.2970</td>\n",
       "      <td>1.6767</td>\n",
       "      <td>-0.6610</td>\n",
       "      <td>-1.3735</td>\n",
       "      <td>1.1583</td>\n",
       "      <td>0.4948</td>\n",
       "      <td>0.5049</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>-0.2075</td>\n",
       "      <td>1.3302</td>\n",
       "      <td>1.6269</td>\n",
       "      <td>0.3267</td>\n",
       "      <td>3.4489</td>\n",
       "      <td>-1.0239</td>\n",
       "      <td>-0.1915</td>\n",
       "      <td>2.5655</td>\n",
       "      <td>2.4185</td>\n",
       "      <td>-0.5810</td>\n",
       "      <td>...</td>\n",
       "      <td>1.5261</td>\n",
       "      <td>-1.9110</td>\n",
       "      <td>1.6519</td>\n",
       "      <td>1.3457</td>\n",
       "      <td>1.7907</td>\n",
       "      <td>-0.6402</td>\n",
       "      <td>0.4227</td>\n",
       "      <td>0.3329</td>\n",
       "      <td>-1.0986</td>\n",
       "      <td>-0.6582</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 122 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  CPT000814  CPT001846  X01BR001  X01BR008  X01BR009  X01BR010  \\\n",
       "Name                                                                       \n",
       "A1BG          -0.6712     1.3964    2.0219   -0.5290    1.2556   -0.3843   \n",
       "A2M           -0.2075     1.3302    1.6269    0.3267    3.4489   -1.0239   \n",
       "\n",
       "Patient_ID  X01BR015  X01BR017  X01BR018  X01BR020  ...  X20BR002  X20BR005  \\\n",
       "Name                                                ...                       \n",
       "A1BG          1.0394    1.1533    1.9579   -0.1637  ...    1.8732   -0.4227   \n",
       "A2M          -0.1915    2.5655    2.4185   -0.5810  ...    1.5261   -1.9110   \n",
       "\n",
       "Patient_ID  X20BR006  X20BR007  X20BR008  X21BR001  X21BR002  X21BR010  \\\n",
       "Name                                                                     \n",
       "A1BG          1.5862   -0.2970    1.6767   -0.6610   -1.3735    1.1583   \n",
       "A2M           1.6519    1.3457    1.7907   -0.6402    0.4227    0.3329   \n",
       "\n",
       "Patient_ID  X22BR005  X22BR006  \n",
       "Name                            \n",
       "A1BG          0.4948    0.5049  \n",
       "A2M          -1.0986   -0.6582  \n",
       "\n",
       "[2 rows x 122 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca_proteomics = get_proteomics(brca)\n",
    "brca_proteomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert len(brca_proteomics.columns[brca_proteomics.columns.duplicated()]) == 0, \"columns contain duplicates\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (8785, 122)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>CPT000814</th>\n",
       "      <th>CPT001846</th>\n",
       "      <th>X01BR001</th>\n",
       "      <th>X01BR008</th>\n",
       "      <th>X01BR009</th>\n",
       "      <th>X01BR010</th>\n",
       "      <th>X01BR015</th>\n",
       "      <th>X01BR017</th>\n",
       "      <th>X01BR018</th>\n",
       "      <th>X01BR020</th>\n",
       "      <th>...</th>\n",
       "      <th>X20BR002</th>\n",
       "      <th>X20BR005</th>\n",
       "      <th>X20BR006</th>\n",
       "      <th>X20BR007</th>\n",
       "      <th>X20BR008</th>\n",
       "      <th>X21BR001</th>\n",
       "      <th>X21BR002</th>\n",
       "      <th>X21BR010</th>\n",
       "      <th>X22BR005</th>\n",
       "      <th>X22BR006</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-0.6712</td>\n",
       "      <td>1.3964</td>\n",
       "      <td>2.0219</td>\n",
       "      <td>-0.5290</td>\n",
       "      <td>1.2556</td>\n",
       "      <td>-0.3843</td>\n",
       "      <td>1.0394</td>\n",
       "      <td>1.1533</td>\n",
       "      <td>1.9579</td>\n",
       "      <td>-0.1637</td>\n",
       "      <td>...</td>\n",
       "      <td>1.8732</td>\n",
       "      <td>-0.4227</td>\n",
       "      <td>1.5862</td>\n",
       "      <td>-0.2970</td>\n",
       "      <td>1.6767</td>\n",
       "      <td>-0.6610</td>\n",
       "      <td>-1.3735</td>\n",
       "      <td>1.1583</td>\n",
       "      <td>0.4948</td>\n",
       "      <td>0.5049</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>-0.2075</td>\n",
       "      <td>1.3302</td>\n",
       "      <td>1.6269</td>\n",
       "      <td>0.3267</td>\n",
       "      <td>3.4489</td>\n",
       "      <td>-1.0239</td>\n",
       "      <td>-0.1915</td>\n",
       "      <td>2.5655</td>\n",
       "      <td>2.4185</td>\n",
       "      <td>-0.5810</td>\n",
       "      <td>...</td>\n",
       "      <td>1.5261</td>\n",
       "      <td>-1.9110</td>\n",
       "      <td>1.6519</td>\n",
       "      <td>1.3457</td>\n",
       "      <td>1.7907</td>\n",
       "      <td>-0.6402</td>\n",
       "      <td>0.4227</td>\n",
       "      <td>0.3329</td>\n",
       "      <td>-1.0986</td>\n",
       "      <td>-0.6582</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 122 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  CPT000814  CPT001846  X01BR001  X01BR008  X01BR009  X01BR010  \\\n",
       "Name                                                                       \n",
       "A1BG          -0.6712     1.3964    2.0219   -0.5290    1.2556   -0.3843   \n",
       "A2M           -0.2075     1.3302    1.6269    0.3267    3.4489   -1.0239   \n",
       "\n",
       "Patient_ID  X01BR015  X01BR017  X01BR018  X01BR020  ...  X20BR002  X20BR005  \\\n",
       "Name                                                ...                       \n",
       "A1BG          1.0394    1.1533    1.9579   -0.1637  ...    1.8732   -0.4227   \n",
       "A2M          -0.1915    2.5655    2.4185   -0.5810  ...    1.5261   -1.9110   \n",
       "\n",
       "Patient_ID  X20BR006  X20BR007  X20BR008  X21BR001  X21BR002  X21BR010  \\\n",
       "Name                                                                     \n",
       "A1BG          1.5862   -0.2970    1.6767   -0.6610   -1.3735    1.1583   \n",
       "A2M           1.6519    1.3457    1.7907   -0.6402    0.4227    0.3329   \n",
       "\n",
       "Patient_ID  X22BR005  X22BR006  \n",
       "Name                            \n",
       "A1BG          0.4948    0.5049  \n",
       "A2M          -1.0986   -0.6582  \n",
       "\n",
       "[2 rows x 122 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca_proteomics_processed = standardised_pipeline_utils.process(brca_proteomics)\n",
    "brca_proteomics_processed[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "transcriptomic_samples['BrCa (2020)'] = brca_transcriptomics_processed.shape[1]\n",
    "proteomic_samples['BrCa (2020)'] = brca_proteomics_processed.shape[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of common proteins:  8243\n",
      "Number of common samples:  122\n"
     ]
    }
   ],
   "source": [
    "brca_transcriptomics_processed, brca_proteomics_processed = standardised_pipeline_utils.match_proteins_samples( \\\n",
    "                                                                            brca_transcriptomics_processed, \n",
    "                                                                            brca_proteomics_processed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "common_samples['BrCa (2020)'] = brca_transcriptomics_processed.shape[1]\n",
    "common_genes['BrCa (2020)'] = brca_transcriptomics_processed.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Spearman Correlation:  0.4348\n"
     ]
    }
   ],
   "source": [
    "correlation_brca = standardised_pipeline_utils.correlate_genewise(brca_transcriptomics_processed, \n",
    "                                                                  brca_proteomics_processed, 'BrCa (2020)')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Pearson Correlation:  0.425\n"
     ]
    }
   ],
   "source": [
    "correlation_brca_pearson = standardised_pipeline_utils.correlate_genewise(brca_transcriptomics_processed, \n",
    "                                                                  brca_proteomics_processed, 'BrCa (2020)', method='pearson')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"Hnscc_Cancer\"></a>\n",
    "### HNSCC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                          \r"
     ]
    }
   ],
   "source": [
    "hnscc = cptac.Hnscc(version='2.0')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (38456, 109)\n",
      "Null values count:  0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>C3L-00977</th>\n",
       "      <th>C3L-00987</th>\n",
       "      <th>C3L-00994</th>\n",
       "      <th>C3L-00995</th>\n",
       "      <th>C3L-00997</th>\n",
       "      <th>C3L-00999</th>\n",
       "      <th>C3L-01138</th>\n",
       "      <th>C3L-01237</th>\n",
       "      <th>C3L-02617</th>\n",
       "      <th>C3L-02621</th>\n",
       "      <th>...</th>\n",
       "      <th>C3N-03933</th>\n",
       "      <th>C3N-04152</th>\n",
       "      <th>C3N-04273</th>\n",
       "      <th>C3N-04275</th>\n",
       "      <th>C3N-04276</th>\n",
       "      <th>C3N-04277</th>\n",
       "      <th>C3N-04278</th>\n",
       "      <th>C3N-04279</th>\n",
       "      <th>C3N-04280</th>\n",
       "      <th>C3N-04611</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>6.20</td>\n",
       "      <td>5.44</td>\n",
       "      <td>5.14</td>\n",
       "      <td>5.69</td>\n",
       "      <td>4.54</td>\n",
       "      <td>4.89</td>\n",
       "      <td>6.59</td>\n",
       "      <td>5.42</td>\n",
       "      <td>4.26</td>\n",
       "      <td>6.21</td>\n",
       "      <td>...</td>\n",
       "      <td>5.14</td>\n",
       "      <td>3.77</td>\n",
       "      <td>4.26</td>\n",
       "      <td>5.22</td>\n",
       "      <td>6.64</td>\n",
       "      <td>6.30</td>\n",
       "      <td>5.67</td>\n",
       "      <td>5.76</td>\n",
       "      <td>5.79</td>\n",
       "      <td>4.77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <td>6.79</td>\n",
       "      <td>6.63</td>\n",
       "      <td>6.31</td>\n",
       "      <td>6.06</td>\n",
       "      <td>5.14</td>\n",
       "      <td>5.76</td>\n",
       "      <td>6.74</td>\n",
       "      <td>5.38</td>\n",
       "      <td>5.42</td>\n",
       "      <td>5.90</td>\n",
       "      <td>...</td>\n",
       "      <td>6.31</td>\n",
       "      <td>4.75</td>\n",
       "      <td>5.41</td>\n",
       "      <td>5.26</td>\n",
       "      <td>6.80</td>\n",
       "      <td>7.03</td>\n",
       "      <td>7.02</td>\n",
       "      <td>6.14</td>\n",
       "      <td>6.23</td>\n",
       "      <td>5.92</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 109 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  C3L-00977  C3L-00987  C3L-00994  C3L-00995  C3L-00997  C3L-00999  \\\n",
       "Name                                                                           \n",
       "A1BG             6.20       5.44       5.14       5.69       4.54       4.89   \n",
       "A1BG-AS1         6.79       6.63       6.31       6.06       5.14       5.76   \n",
       "\n",
       "Patient_ID  C3L-01138  C3L-01237  C3L-02617  C3L-02621  ...  C3N-03933  \\\n",
       "Name                                                    ...              \n",
       "A1BG             6.59       5.42       4.26       6.21  ...       5.14   \n",
       "A1BG-AS1         6.74       5.38       5.42       5.90  ...       6.31   \n",
       "\n",
       "Patient_ID  C3N-04152  C3N-04273  C3N-04275  C3N-04276  C3N-04277  C3N-04278  \\\n",
       "Name                                                                           \n",
       "A1BG             3.77       4.26       5.22       6.64       6.30       5.67   \n",
       "A1BG-AS1         4.75       5.41       5.26       6.80       7.03       7.02   \n",
       "\n",
       "Patient_ID  C3N-04279  C3N-04280  C3N-04611  \n",
       "Name                                         \n",
       "A1BG             5.76       5.79       4.77  \n",
       "A1BG-AS1         6.14       6.23       5.92  \n",
       "\n",
       "[2 rows x 109 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hnscc_transcriptomics = get_transcriptomics(hnscc)\n",
    "hnscc_transcriptomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Samples removed due to (i) QC failed (ii) HPV positive as specified in Table S1 of HNSCC Cancer Cell Paper\n",
    "# https://www.cell.com/cancer-cell/fulltext/S1535-6108(20)30655-3#supplementaryMaterial\n",
    "removed_samples = ['C3N-01643', 'C3N-02693']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "hnscc_transcriptomics.drop(columns = [col for col in hnscc_transcriptomics.columns if col in removed_samples], \n",
    "                           inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (26134, 108)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>C3L-00977</th>\n",
       "      <th>C3L-00987</th>\n",
       "      <th>C3L-00994</th>\n",
       "      <th>C3L-00995</th>\n",
       "      <th>C3L-00997</th>\n",
       "      <th>C3L-00999</th>\n",
       "      <th>C3L-01138</th>\n",
       "      <th>C3L-01237</th>\n",
       "      <th>C3L-02617</th>\n",
       "      <th>C3L-02621</th>\n",
       "      <th>...</th>\n",
       "      <th>C3N-03933</th>\n",
       "      <th>C3N-04152</th>\n",
       "      <th>C3N-04273</th>\n",
       "      <th>C3N-04275</th>\n",
       "      <th>C3N-04276</th>\n",
       "      <th>C3N-04277</th>\n",
       "      <th>C3N-04278</th>\n",
       "      <th>C3N-04279</th>\n",
       "      <th>C3N-04280</th>\n",
       "      <th>C3N-04611</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>6.20</td>\n",
       "      <td>5.44</td>\n",
       "      <td>5.14</td>\n",
       "      <td>5.69</td>\n",
       "      <td>4.54</td>\n",
       "      <td>4.89</td>\n",
       "      <td>6.59</td>\n",
       "      <td>5.42</td>\n",
       "      <td>4.26</td>\n",
       "      <td>6.21</td>\n",
       "      <td>...</td>\n",
       "      <td>5.14</td>\n",
       "      <td>3.77</td>\n",
       "      <td>4.26</td>\n",
       "      <td>5.22</td>\n",
       "      <td>6.64</td>\n",
       "      <td>6.30</td>\n",
       "      <td>5.67</td>\n",
       "      <td>5.76</td>\n",
       "      <td>5.79</td>\n",
       "      <td>4.77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <td>6.79</td>\n",
       "      <td>6.63</td>\n",
       "      <td>6.31</td>\n",
       "      <td>6.06</td>\n",
       "      <td>5.14</td>\n",
       "      <td>5.76</td>\n",
       "      <td>6.74</td>\n",
       "      <td>5.38</td>\n",
       "      <td>5.42</td>\n",
       "      <td>5.90</td>\n",
       "      <td>...</td>\n",
       "      <td>6.31</td>\n",
       "      <td>4.75</td>\n",
       "      <td>5.41</td>\n",
       "      <td>5.26</td>\n",
       "      <td>6.80</td>\n",
       "      <td>7.03</td>\n",
       "      <td>7.02</td>\n",
       "      <td>6.14</td>\n",
       "      <td>6.23</td>\n",
       "      <td>5.92</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 108 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  C3L-00977  C3L-00987  C3L-00994  C3L-00995  C3L-00997  C3L-00999  \\\n",
       "Name                                                                           \n",
       "A1BG             6.20       5.44       5.14       5.69       4.54       4.89   \n",
       "A1BG-AS1         6.79       6.63       6.31       6.06       5.14       5.76   \n",
       "\n",
       "Patient_ID  C3L-01138  C3L-01237  C3L-02617  C3L-02621  ...  C3N-03933  \\\n",
       "Name                                                    ...              \n",
       "A1BG             6.59       5.42       4.26       6.21  ...       5.14   \n",
       "A1BG-AS1         6.74       5.38       5.42       5.90  ...       6.31   \n",
       "\n",
       "Patient_ID  C3N-04152  C3N-04273  C3N-04275  C3N-04276  C3N-04277  C3N-04278  \\\n",
       "Name                                                                           \n",
       "A1BG             3.77       4.26       5.22       6.64       6.30       5.67   \n",
       "A1BG-AS1         4.75       5.41       5.26       6.80       7.03       7.02   \n",
       "\n",
       "Patient_ID  C3N-04279  C3N-04280  C3N-04611  \n",
       "Name                                         \n",
       "A1BG             5.76       5.79       4.77  \n",
       "A1BG-AS1         6.14       6.23       5.92  \n",
       "\n",
       "[2 rows x 108 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hnscc_transcriptomics_processed = standardised_pipeline_utils.process(hnscc_transcriptomics)\n",
    "hnscc_transcriptomics_processed[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dataframe transposed.\n",
      "Dimensions:  (11744, 115)\n",
      "Null values count:  238975\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>C3L-00977</th>\n",
       "      <th>C3L-00987</th>\n",
       "      <th>C3L-00994</th>\n",
       "      <th>C3L-00995</th>\n",
       "      <th>C3L-00997</th>\n",
       "      <th>C3L-00999</th>\n",
       "      <th>C3L-01138</th>\n",
       "      <th>C3L-01237</th>\n",
       "      <th>C3L-02617</th>\n",
       "      <th>C3L-02621</th>\n",
       "      <th>...</th>\n",
       "      <th>C3N-04278</th>\n",
       "      <th>C3N-04279</th>\n",
       "      <th>C3N-04280</th>\n",
       "      <th>C3N-04611</th>\n",
       "      <th>C3L-00994.C</th>\n",
       "      <th>C3L-02617.C</th>\n",
       "      <th>C3L-04350.C</th>\n",
       "      <th>C3L-05257.C</th>\n",
       "      <th>C3N-01757.C</th>\n",
       "      <th>C3N-03042.C</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>27.725342</td>\n",
       "      <td>28.152905</td>\n",
       "      <td>28.348186</td>\n",
       "      <td>28.004445</td>\n",
       "      <td>27.735214</td>\n",
       "      <td>27.949122</td>\n",
       "      <td>28.293267</td>\n",
       "      <td>28.216073</td>\n",
       "      <td>27.452281</td>\n",
       "      <td>27.70104</td>\n",
       "      <td>...</td>\n",
       "      <td>27.475926</td>\n",
       "      <td>27.349798</td>\n",
       "      <td>27.786746</td>\n",
       "      <td>27.881307</td>\n",
       "      <td>27.865217</td>\n",
       "      <td>28.26514</td>\n",
       "      <td>28.828969</td>\n",
       "      <td>28.14128</td>\n",
       "      <td>28.141688</td>\n",
       "      <td>28.680969</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>19.056377</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18.058554</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19.320859</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 115 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  C3L-00977  C3L-00987  C3L-00994  C3L-00995  C3L-00997  C3L-00999  \\\n",
       "Name                                                                           \n",
       "A1BG        27.725342  28.152905  28.348186  28.004445  27.735214  27.949122   \n",
       "A1CF        19.056377        NaN  18.058554        NaN        NaN        NaN   \n",
       "\n",
       "Patient_ID  C3L-01138  C3L-01237  C3L-02617  C3L-02621  ...  C3N-04278  \\\n",
       "Name                                                    ...              \n",
       "A1BG        28.293267  28.216073  27.452281   27.70104  ...  27.475926   \n",
       "A1CF              NaN        NaN  19.320859        NaN  ...        NaN   \n",
       "\n",
       "Patient_ID  C3N-04279  C3N-04280  C3N-04611  C3L-00994.C  C3L-02617.C  \\\n",
       "Name                                                                    \n",
       "A1BG        27.349798  27.786746  27.881307    27.865217     28.26514   \n",
       "A1CF              NaN        NaN        NaN          NaN          NaN   \n",
       "\n",
       "Patient_ID  C3L-04350.C  C3L-05257.C  C3N-01757.C  C3N-03042.C  \n",
       "Name                                                            \n",
       "A1BG          28.828969     28.14128    28.141688    28.680969  \n",
       "A1CF                NaN          NaN          NaN          NaN  \n",
       "\n",
       "[2 rows x 115 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hnscc_proteomics = get_proteomics(hnscc, multiIndexed=False)\n",
    "hnscc_proteomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Aggregate the replicate samples as they have > 0.9 spearman correlation \n",
    "hnscc_proteomics.rename(columns = lambda x: str(x).replace('.C', ''), inplace=True) \n",
    "hnscc_proteomics = hnscc_proteomics.groupby(hnscc_proteomics.columns, axis=1).mean()\n",
    "# Drop removed samples from our analysis\n",
    "hnscc_proteomics.drop(columns = [x for x in hnscc_proteomics.columns if x in removed_samples], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (8696, 110)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>C3L-00977</th>\n",
       "      <th>C3L-00987</th>\n",
       "      <th>C3L-00994</th>\n",
       "      <th>C3L-00995</th>\n",
       "      <th>C3L-00997</th>\n",
       "      <th>C3L-00999</th>\n",
       "      <th>C3L-01138</th>\n",
       "      <th>C3L-01237</th>\n",
       "      <th>C3L-02617</th>\n",
       "      <th>C3L-02621</th>\n",
       "      <th>...</th>\n",
       "      <th>C3N-03933</th>\n",
       "      <th>C3N-04152</th>\n",
       "      <th>C3N-04273</th>\n",
       "      <th>C3N-04275</th>\n",
       "      <th>C3N-04276</th>\n",
       "      <th>C3N-04277</th>\n",
       "      <th>C3N-04278</th>\n",
       "      <th>C3N-04279</th>\n",
       "      <th>C3N-04280</th>\n",
       "      <th>C3N-04611</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>27.725342</td>\n",
       "      <td>28.152905</td>\n",
       "      <td>28.106702</td>\n",
       "      <td>28.004445</td>\n",
       "      <td>27.735214</td>\n",
       "      <td>27.949122</td>\n",
       "      <td>28.293267</td>\n",
       "      <td>28.216073</td>\n",
       "      <td>27.858710</td>\n",
       "      <td>27.701040</td>\n",
       "      <td>...</td>\n",
       "      <td>27.853538</td>\n",
       "      <td>27.473217</td>\n",
       "      <td>27.351122</td>\n",
       "      <td>27.520569</td>\n",
       "      <td>28.709391</td>\n",
       "      <td>27.736346</td>\n",
       "      <td>27.475926</td>\n",
       "      <td>27.349798</td>\n",
       "      <td>27.786746</td>\n",
       "      <td>27.881307</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>28.565472</td>\n",
       "      <td>29.374443</td>\n",
       "      <td>30.067640</td>\n",
       "      <td>29.267877</td>\n",
       "      <td>28.724642</td>\n",
       "      <td>29.352481</td>\n",
       "      <td>29.229332</td>\n",
       "      <td>29.100412</td>\n",
       "      <td>29.014712</td>\n",
       "      <td>28.806067</td>\n",
       "      <td>...</td>\n",
       "      <td>28.972230</td>\n",
       "      <td>28.476319</td>\n",
       "      <td>28.415198</td>\n",
       "      <td>29.133823</td>\n",
       "      <td>29.190307</td>\n",
       "      <td>29.491318</td>\n",
       "      <td>29.052760</td>\n",
       "      <td>28.465475</td>\n",
       "      <td>28.505049</td>\n",
       "      <td>28.674746</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 110 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  C3L-00977  C3L-00987  C3L-00994  C3L-00995  C3L-00997  C3L-00999  \\\n",
       "Name                                                                           \n",
       "A1BG        27.725342  28.152905  28.106702  28.004445  27.735214  27.949122   \n",
       "A2M         28.565472  29.374443  30.067640  29.267877  28.724642  29.352481   \n",
       "\n",
       "Patient_ID  C3L-01138  C3L-01237  C3L-02617  C3L-02621  ...  C3N-03933  \\\n",
       "Name                                                    ...              \n",
       "A1BG        28.293267  28.216073  27.858710  27.701040  ...  27.853538   \n",
       "A2M         29.229332  29.100412  29.014712  28.806067  ...  28.972230   \n",
       "\n",
       "Patient_ID  C3N-04152  C3N-04273  C3N-04275  C3N-04276  C3N-04277  C3N-04278  \\\n",
       "Name                                                                           \n",
       "A1BG        27.473217  27.351122  27.520569  28.709391  27.736346  27.475926   \n",
       "A2M         28.476319  28.415198  29.133823  29.190307  29.491318  29.052760   \n",
       "\n",
       "Patient_ID  C3N-04279  C3N-04280  C3N-04611  \n",
       "Name                                         \n",
       "A1BG        27.349798  27.786746  27.881307  \n",
       "A2M         28.465475  28.505049  28.674746  \n",
       "\n",
       "[2 rows x 110 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hnscc_proteomics_processed = standardised_pipeline_utils.process(hnscc_proteomics)\n",
    "hnscc_proteomics_processed[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "transcriptomic_samples['HNSCC'] = hnscc_transcriptomics_processed.shape[1]\n",
    "proteomic_samples['HNSCC'] = hnscc_proteomics_processed.shape[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of common proteins:  8583\n",
      "Number of common samples:  108\n"
     ]
    }
   ],
   "source": [
    "hnscc_transcriptomics_processed, hnscc_proteomics_processed = standardised_pipeline_utils.match_proteins_samples( \\\n",
    "                                                                            hnscc_transcriptomics_processed, \n",
    "                                                                            hnscc_proteomics_processed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "common_samples['HNSCC'] = hnscc_transcriptomics_processed.shape[1]\n",
    "common_genes['HNSCC'] = hnscc_transcriptomics_processed.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Spearman Correlation:  0.5296\n"
     ]
    }
   ],
   "source": [
    "correlation_hnscc = standardised_pipeline_utils.correlate_genewise(hnscc_transcriptomics_processed, \n",
    "                                                                   hnscc_proteomics_processed, 'HNSCC')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Pearson Correlation:  0.5603\n"
     ]
    }
   ],
   "source": [
    "correlation_hnscc_pearson = standardised_pipeline_utils.correlate_genewise(hnscc_transcriptomics_processed, \n",
    "                                                                   hnscc_proteomics_processed, 'HNSCC', method='pearson')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"GBM_Cancer\"></a>\n",
    "### GBM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                        \r"
     ]
    }
   ],
   "source": [
    "gbm = cptac.Gbm(version='3.0')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (60483, 99)\n",
      "Null values count:  0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>C3L-00104</th>\n",
       "      <th>C3L-00365</th>\n",
       "      <th>C3L-00674</th>\n",
       "      <th>C3L-00677</th>\n",
       "      <th>C3L-01040</th>\n",
       "      <th>C3L-01043</th>\n",
       "      <th>C3L-01045</th>\n",
       "      <th>C3L-01046</th>\n",
       "      <th>C3L-01048</th>\n",
       "      <th>C3L-01049</th>\n",
       "      <th>...</th>\n",
       "      <th>C3N-02788</th>\n",
       "      <th>C3N-03070</th>\n",
       "      <th>C3N-03088</th>\n",
       "      <th>C3N-03180</th>\n",
       "      <th>C3N-03182</th>\n",
       "      <th>C3N-03183</th>\n",
       "      <th>C3N-03184</th>\n",
       "      <th>C3N-03186</th>\n",
       "      <th>C3N-03188</th>\n",
       "      <th>C3N-03473</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5S_rRNA</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5S_rRNA</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 99 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  C3L-00104  C3L-00365  C3L-00674  C3L-00677  C3L-01040  C3L-01043  \\\n",
       "Name                                                                           \n",
       "5S_rRNA           0.0        0.0        0.0        0.0        0.0        0.0   \n",
       "5S_rRNA           0.0        0.0        0.0        0.0        0.0        0.0   \n",
       "\n",
       "Patient_ID  C3L-01045  C3L-01046  C3L-01048  C3L-01049  ...  C3N-02788  \\\n",
       "Name                                                    ...              \n",
       "5S_rRNA           0.0        0.0        0.0        0.0  ...        0.0   \n",
       "5S_rRNA           0.0        0.0        0.0        0.0  ...        0.0   \n",
       "\n",
       "Patient_ID  C3N-03070  C3N-03088  C3N-03180  C3N-03182  C3N-03183  C3N-03184  \\\n",
       "Name                                                                           \n",
       "5S_rRNA           0.0        0.0        0.0        0.0        0.0        0.0   \n",
       "5S_rRNA           0.0        0.0        0.0        0.0        0.0        0.0   \n",
       "\n",
       "Patient_ID  C3N-03186  C3N-03188  C3N-03473  \n",
       "Name                                         \n",
       "5S_rRNA           0.0        0.0        0.0  \n",
       "5S_rRNA           0.0        0.0        0.0  \n",
       "\n",
       "[2 rows x 99 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gbm_transcriptomics = get_transcriptomics(gbm, multiIndexed=True)\n",
    "gbm_transcriptomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert len(gbm_transcriptomics.columns[gbm_transcriptomics.columns.duplicated()]) == 0, \"columns contain duplicates\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (29628, 99)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>C3L-00104</th>\n",
       "      <th>C3L-00365</th>\n",
       "      <th>C3L-00674</th>\n",
       "      <th>C3L-00677</th>\n",
       "      <th>C3L-01040</th>\n",
       "      <th>C3L-01043</th>\n",
       "      <th>C3L-01045</th>\n",
       "      <th>C3L-01046</th>\n",
       "      <th>C3L-01048</th>\n",
       "      <th>C3L-01049</th>\n",
       "      <th>...</th>\n",
       "      <th>C3N-02788</th>\n",
       "      <th>C3N-03070</th>\n",
       "      <th>C3N-03088</th>\n",
       "      <th>C3N-03180</th>\n",
       "      <th>C3N-03182</th>\n",
       "      <th>C3N-03183</th>\n",
       "      <th>C3N-03184</th>\n",
       "      <th>C3N-03186</th>\n",
       "      <th>C3N-03188</th>\n",
       "      <th>C3N-03473</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5S_rRNA</th>\n",
       "      <td>38886.167971</td>\n",
       "      <td>26532.401969</td>\n",
       "      <td>43273.830168</td>\n",
       "      <td>34050.208702</td>\n",
       "      <td>42224.309970</td>\n",
       "      <td>44217.304860</td>\n",
       "      <td>47885.637617</td>\n",
       "      <td>45879.768789</td>\n",
       "      <td>49169.733975</td>\n",
       "      <td>48205.474974</td>\n",
       "      <td>...</td>\n",
       "      <td>28434.976673</td>\n",
       "      <td>62476.163734</td>\n",
       "      <td>23732.935866</td>\n",
       "      <td>25153.158215</td>\n",
       "      <td>40051.285619</td>\n",
       "      <td>36967.844139</td>\n",
       "      <td>80850.137537</td>\n",
       "      <td>37860.609540</td>\n",
       "      <td>34691.447141</td>\n",
       "      <td>61714.912891</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7SK</th>\n",
       "      <td>354300.002325</td>\n",
       "      <td>270272.124551</td>\n",
       "      <td>89523.626156</td>\n",
       "      <td>121644.956366</td>\n",
       "      <td>93259.597405</td>\n",
       "      <td>130696.137435</td>\n",
       "      <td>116847.685886</td>\n",
       "      <td>106623.883707</td>\n",
       "      <td>105016.376686</td>\n",
       "      <td>81733.377534</td>\n",
       "      <td>...</td>\n",
       "      <td>125895.148430</td>\n",
       "      <td>147671.301692</td>\n",
       "      <td>64687.124482</td>\n",
       "      <td>172464.282283</td>\n",
       "      <td>186020.044424</td>\n",
       "      <td>181443.805445</td>\n",
       "      <td>271901.473898</td>\n",
       "      <td>216507.270389</td>\n",
       "      <td>73552.366041</td>\n",
       "      <td>134589.007570</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 99 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID      C3L-00104      C3L-00365     C3L-00674      C3L-00677  \\\n",
       "Name                                                                    \n",
       "5S_rRNA      38886.167971   26532.401969  43273.830168   34050.208702   \n",
       "7SK         354300.002325  270272.124551  89523.626156  121644.956366   \n",
       "\n",
       "Patient_ID     C3L-01040      C3L-01043      C3L-01045      C3L-01046  \\\n",
       "Name                                                                    \n",
       "5S_rRNA     42224.309970   44217.304860   47885.637617   45879.768789   \n",
       "7SK         93259.597405  130696.137435  116847.685886  106623.883707   \n",
       "\n",
       "Patient_ID      C3L-01048     C3L-01049  ...      C3N-02788      C3N-03070  \\\n",
       "Name                                     ...                                 \n",
       "5S_rRNA      49169.733975  48205.474974  ...   28434.976673   62476.163734   \n",
       "7SK         105016.376686  81733.377534  ...  125895.148430  147671.301692   \n",
       "\n",
       "Patient_ID     C3N-03088      C3N-03180      C3N-03182      C3N-03183  \\\n",
       "Name                                                                    \n",
       "5S_rRNA     23732.935866   25153.158215   40051.285619   36967.844139   \n",
       "7SK         64687.124482  172464.282283  186020.044424  181443.805445   \n",
       "\n",
       "Patient_ID      C3N-03184      C3N-03186     C3N-03188      C3N-03473  \n",
       "Name                                                                   \n",
       "5S_rRNA      80850.137537   37860.609540  34691.447141   61714.912891  \n",
       "7SK         271901.473898  216507.270389  73552.366041  134589.007570  \n",
       "\n",
       "[2 rows x 99 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gbm_transcriptomics_processed = standardised_pipeline_utils.process(gbm_transcriptomics)\n",
    "gbm_transcriptomics_processed[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dataframe transposed.\n",
      "Dimensions:  (11141, 99)\n",
      "Null values count:  85305\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>C3L-00104</th>\n",
       "      <th>C3L-00365</th>\n",
       "      <th>C3L-00674</th>\n",
       "      <th>C3L-00677</th>\n",
       "      <th>C3L-01040</th>\n",
       "      <th>C3L-01043</th>\n",
       "      <th>C3L-01045</th>\n",
       "      <th>C3L-01046</th>\n",
       "      <th>C3L-01048</th>\n",
       "      <th>C3L-01049</th>\n",
       "      <th>...</th>\n",
       "      <th>C3N-02788</th>\n",
       "      <th>C3N-03070</th>\n",
       "      <th>C3N-03088</th>\n",
       "      <th>C3N-03180</th>\n",
       "      <th>C3N-03182</th>\n",
       "      <th>C3N-03183</th>\n",
       "      <th>C3N-03184</th>\n",
       "      <th>C3N-03186</th>\n",
       "      <th>C3N-03188</th>\n",
       "      <th>C3N-03473</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>0.077630</td>\n",
       "      <td>-0.145975</td>\n",
       "      <td>0.821991</td>\n",
       "      <td>-0.064567</td>\n",
       "      <td>-0.763691</td>\n",
       "      <td>1.094879</td>\n",
       "      <td>-0.027903</td>\n",
       "      <td>-0.375754</td>\n",
       "      <td>-0.394736</td>\n",
       "      <td>-0.025968</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.327487</td>\n",
       "      <td>1.942106</td>\n",
       "      <td>0.27851</td>\n",
       "      <td>1.045880</td>\n",
       "      <td>-0.424647</td>\n",
       "      <td>0.230843</td>\n",
       "      <td>-0.635316</td>\n",
       "      <td>0.616640</td>\n",
       "      <td>-0.059547</td>\n",
       "      <td>-0.899255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>0.487228</td>\n",
       "      <td>0.798796</td>\n",
       "      <td>1.096470</td>\n",
       "      <td>0.129385</td>\n",
       "      <td>-1.031834</td>\n",
       "      <td>0.769231</td>\n",
       "      <td>-0.735991</td>\n",
       "      <td>-0.037553</td>\n",
       "      <td>-0.485108</td>\n",
       "      <td>-0.310086</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.340301</td>\n",
       "      <td>1.657565</td>\n",
       "      <td>0.83660</td>\n",
       "      <td>1.151704</td>\n",
       "      <td>-0.733923</td>\n",
       "      <td>0.426624</td>\n",
       "      <td>-0.478657</td>\n",
       "      <td>0.767029</td>\n",
       "      <td>-0.526563</td>\n",
       "      <td>-0.333312</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 99 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  C3L-00104  C3L-00365  C3L-00674  C3L-00677  C3L-01040  C3L-01043  \\\n",
       "Name                                                                           \n",
       "A1BG         0.077630  -0.145975   0.821991  -0.064567  -0.763691   1.094879   \n",
       "A2M          0.487228   0.798796   1.096470   0.129385  -1.031834   0.769231   \n",
       "\n",
       "Patient_ID  C3L-01045  C3L-01046  C3L-01048  C3L-01049  ...  C3N-02788  \\\n",
       "Name                                                    ...              \n",
       "A1BG        -0.027903  -0.375754  -0.394736  -0.025968  ...  -0.327487   \n",
       "A2M         -0.735991  -0.037553  -0.485108  -0.310086  ...  -0.340301   \n",
       "\n",
       "Patient_ID  C3N-03070  C3N-03088  C3N-03180  C3N-03182  C3N-03183  C3N-03184  \\\n",
       "Name                                                                           \n",
       "A1BG         1.942106    0.27851   1.045880  -0.424647   0.230843  -0.635316   \n",
       "A2M          1.657565    0.83660   1.151704  -0.733923   0.426624  -0.478657   \n",
       "\n",
       "Patient_ID  C3N-03186  C3N-03188  C3N-03473  \n",
       "Name                                         \n",
       "A1BG         0.616640  -0.059547  -0.899255  \n",
       "A2M          0.767029  -0.526563  -0.333312  \n",
       "\n",
       "[2 rows x 99 columns]"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gbm_proteomics = get_proteomics(gbm, multiIndexed=False)\n",
    "gbm_proteomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (9786, 99)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Patient_ID</th>\n",
       "      <th>C3L-00104</th>\n",
       "      <th>C3L-00365</th>\n",
       "      <th>C3L-00674</th>\n",
       "      <th>C3L-00677</th>\n",
       "      <th>C3L-01040</th>\n",
       "      <th>C3L-01043</th>\n",
       "      <th>C3L-01045</th>\n",
       "      <th>C3L-01046</th>\n",
       "      <th>C3L-01048</th>\n",
       "      <th>C3L-01049</th>\n",
       "      <th>...</th>\n",
       "      <th>C3N-02788</th>\n",
       "      <th>C3N-03070</th>\n",
       "      <th>C3N-03088</th>\n",
       "      <th>C3N-03180</th>\n",
       "      <th>C3N-03182</th>\n",
       "      <th>C3N-03183</th>\n",
       "      <th>C3N-03184</th>\n",
       "      <th>C3N-03186</th>\n",
       "      <th>C3N-03188</th>\n",
       "      <th>C3N-03473</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>0.077630</td>\n",
       "      <td>-0.145975</td>\n",
       "      <td>0.821991</td>\n",
       "      <td>-0.064567</td>\n",
       "      <td>-0.763691</td>\n",
       "      <td>1.094879</td>\n",
       "      <td>-0.027903</td>\n",
       "      <td>-0.375754</td>\n",
       "      <td>-0.394736</td>\n",
       "      <td>-0.025968</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.327487</td>\n",
       "      <td>1.942106</td>\n",
       "      <td>0.27851</td>\n",
       "      <td>1.045880</td>\n",
       "      <td>-0.424647</td>\n",
       "      <td>0.230843</td>\n",
       "      <td>-0.635316</td>\n",
       "      <td>0.616640</td>\n",
       "      <td>-0.059547</td>\n",
       "      <td>-0.899255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>0.487228</td>\n",
       "      <td>0.798796</td>\n",
       "      <td>1.096470</td>\n",
       "      <td>0.129385</td>\n",
       "      <td>-1.031834</td>\n",
       "      <td>0.769231</td>\n",
       "      <td>-0.735991</td>\n",
       "      <td>-0.037553</td>\n",
       "      <td>-0.485108</td>\n",
       "      <td>-0.310086</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.340301</td>\n",
       "      <td>1.657565</td>\n",
       "      <td>0.83660</td>\n",
       "      <td>1.151704</td>\n",
       "      <td>-0.733923</td>\n",
       "      <td>0.426624</td>\n",
       "      <td>-0.478657</td>\n",
       "      <td>0.767029</td>\n",
       "      <td>-0.526563</td>\n",
       "      <td>-0.333312</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 99 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Patient_ID  C3L-00104  C3L-00365  C3L-00674  C3L-00677  C3L-01040  C3L-01043  \\\n",
       "Name                                                                           \n",
       "A1BG         0.077630  -0.145975   0.821991  -0.064567  -0.763691   1.094879   \n",
       "A2M          0.487228   0.798796   1.096470   0.129385  -1.031834   0.769231   \n",
       "\n",
       "Patient_ID  C3L-01045  C3L-01046  C3L-01048  C3L-01049  ...  C3N-02788  \\\n",
       "Name                                                    ...              \n",
       "A1BG        -0.027903  -0.375754  -0.394736  -0.025968  ...  -0.327487   \n",
       "A2M         -0.735991  -0.037553  -0.485108  -0.310086  ...  -0.340301   \n",
       "\n",
       "Patient_ID  C3N-03070  C3N-03088  C3N-03180  C3N-03182  C3N-03183  C3N-03184  \\\n",
       "Name                                                                           \n",
       "A1BG         1.942106    0.27851   1.045880  -0.424647   0.230843  -0.635316   \n",
       "A2M          1.657565    0.83660   1.151704  -0.733923   0.426624  -0.478657   \n",
       "\n",
       "Patient_ID  C3N-03186  C3N-03188  C3N-03473  \n",
       "Name                                         \n",
       "A1BG         0.616640  -0.059547  -0.899255  \n",
       "A2M          0.767029  -0.526563  -0.333312  \n",
       "\n",
       "[2 rows x 99 columns]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gbm_proteomics_processed = standardised_pipeline_utils.process(gbm_proteomics)\n",
    "gbm_proteomics_processed[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "transcriptomic_samples['GBM'] = gbm_transcriptomics_processed.shape[1]\n",
    "proteomic_samples['GBM'] = gbm_proteomics_processed.shape[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of common proteins:  9348\n",
      "Number of common samples:  99\n"
     ]
    }
   ],
   "source": [
    "gbm_transcriptomics_processed, gbm_proteomics_processed = standardised_pipeline_utils.match_proteins_samples( \\\n",
    "                                                                            gbm_transcriptomics_processed, \n",
    "                                                                            gbm_proteomics_processed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "common_samples['GBM'] = gbm_transcriptomics_processed.shape[1]\n",
    "common_genes['GBM'] = gbm_proteomics_processed.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Spearman Correlation:  0.5014\n"
     ]
    }
   ],
   "source": [
    "correlation_gbm = standardised_pipeline_utils.correlate_genewise(gbm_transcriptomics_processed, \n",
    "                                                                   gbm_proteomics_processed, 'GBM')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Pearson Correlation:  0.5061\n"
     ]
    }
   ],
   "source": [
    "correlation_gbm_pearson = standardised_pipeline_utils.correlate_genewise(gbm_transcriptomics_processed, \n",
    "                                                                   gbm_proteomics_processed, 'GBM', method='pearson')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># Samples in Trancriptomic Data</th>\n",
       "      <th># Samples in Proteomic Data</th>\n",
       "      <th># Common Samples</th>\n",
       "      <th># Common Proteins</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Data</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ccRCC</th>\n",
       "      <td>103</td>\n",
       "      <td>103</td>\n",
       "      <td>103</td>\n",
       "      <td>7609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EC</th>\n",
       "      <td>95</td>\n",
       "      <td>95</td>\n",
       "      <td>95</td>\n",
       "      <td>8998</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LUAD</th>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "      <td>8507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BrCa (2020)</th>\n",
       "      <td>122</td>\n",
       "      <td>122</td>\n",
       "      <td>122</td>\n",
       "      <td>8243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HNSCC</th>\n",
       "      <td>108</td>\n",
       "      <td>110</td>\n",
       "      <td>108</td>\n",
       "      <td>8583</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GBM</th>\n",
       "      <td>99</td>\n",
       "      <td>99</td>\n",
       "      <td>99</td>\n",
       "      <td>9348</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             # Samples in Trancriptomic Data  # Samples in Proteomic Data  \\\n",
       "Data                                                                        \n",
       "ccRCC                                    103                          103   \n",
       "EC                                        95                           95   \n",
       "LUAD                                     110                          110   \n",
       "BrCa (2020)                              122                          122   \n",
       "HNSCC                                    108                          110   \n",
       "GBM                                       99                           99   \n",
       "\n",
       "             # Common Samples  # Common Proteins  \n",
       "Data                                              \n",
       "ccRCC                     103               7609  \n",
       "EC                         95               8998  \n",
       "LUAD                      110               8507  \n",
       "BrCa (2020)               122               8243  \n",
       "HNSCC                     108               8583  \n",
       "GBM                        99               9348  "
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def dataframe_from_dict(*dict_args):\n",
    "    dataframe = pd.DataFrame.from_dict(dict_args[0], orient='index')\n",
    "    for i in range(1, len(dict_args)):\n",
    "        dataframe = pd.concat([dataframe, pd.DataFrame.from_dict(dict_args[i], orient='index')], axis=1)\n",
    "    dataframe.reset_index(inplace=True)\n",
    "    dataframe.columns=['Data', '# Samples in Trancriptomic Data', '# Samples in Proteomic Data', \n",
    "                       '# Common Samples', '# Common Proteins']\n",
    "    return dataframe.set_index('Data')\n",
    "\n",
    "sample_data = dataframe_from_dict(transcriptomic_samples, proteomic_samples, common_samples, common_genes)\n",
    "sample_data.to_csv(file_samples_info, header = False, mode='a')\n",
    "sample_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ccRCC</th>\n",
       "      <th>EC</th>\n",
       "      <th>LUAD</th>\n",
       "      <th>BrCa (2020)</th>\n",
       "      <th>HNSCC</th>\n",
       "      <th>GBM</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>0.035573</td>\n",
       "      <td>0.324974</td>\n",
       "      <td>-0.119663</td>\n",
       "      <td>0.077242</td>\n",
       "      <td>0.065872</td>\n",
       "      <td>-0.192863</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>0.908613</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         ccRCC        EC      LUAD  BrCa (2020)     HNSCC       GBM\n",
       "Name                                                               \n",
       "A1BG  0.035573  0.324974 -0.119663     0.077242  0.065872 -0.192863\n",
       "A1CF  0.908613       NaN       NaN          NaN       NaN       NaN"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "correlation_combined = pd.concat([correlation_ccrcc, correlation_endo, correlation_luad, correlation_brca, \n",
    "                                  correlation_hnscc, correlation_gbm], axis=1)\n",
    "correlation_combined.to_csv(file_tumour_correlation)\n",
    "correlation_combined[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ccRCC          0.42\n",
       "EC             0.51\n",
       "LUAD           0.56\n",
       "BrCa (2020)    0.43\n",
       "HNSCC          0.56\n",
       "GBM            0.51\n",
       "dtype: float64"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "correlation_combined_pearson = pd.concat([correlation_ccrcc_pearson, correlation_endo_pearson, \n",
    "                                          correlation_luad_pearson, correlation_brca_pearson, \n",
    "                                          correlation_hnscc_pearson, correlation_gbm_pearson], axis=1)\n",
    "round(correlation_combined_pearson.median(), 2)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
