{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##  Part 3: Compute mRNA-protein correlation for non-tumour studies"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Input:** Transcriptomics and Proteomics data of the non tumour studies listed below.\n",
    "1. <a href=https://tinyurl.com/54af2rvy>Cancer cell lines  encyclopedia (2020)</a>           \n",
    "2. <a href=https://tinyurl.com/rsv3efp5>GTEx 32 healthy tissues (2020)</a>          \n",
    "3. <a href=https://tinyurl.com/56439rpr>NCI-60 cancer cellines (2019)</a>\n",
    "\n",
    "**Output:** Gene-wise correlation between mRNA and protein abundances "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Import Packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os, re\n",
    "import numpy as np\n",
    "import pandas as pd \n",
    "\n",
    "import standardised_pipeline_utils"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "get_data_path = lambda folders, fname: os.path.normpath(os.environ['DATA_PATH']+'/'+'/'.join(folders) +'/'+ fname)\n",
    "get_local_data_path = lambda folders, fname: os.path.normpath('../local_data/'+'/'.join(folders) +'/'+ fname)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "file_ccle_transcriptomics = get_data_path(['depmap','20Q4'], 'CCLE_expression.csv')\n",
    "file_ccle_sample_info = get_data_path(['depmap','20Q4'], 'sample_info.csv')\n",
    "file_ccle_proteomics = get_data_path(['nusinow_2020'],'Table_S2.xlsx')\n",
    "\n",
    "file_gtex_transcriptomics = get_data_path(['jiang_2020'], 'Table_S3.xlsx')\n",
    "file_gtex_proteomics = get_data_path(['jiang_2020'], 'Table_S2.xlsx')\n",
    "\n",
    "file_nci60_transcriptomics =  get_data_path(['guo_2019'], 'Table_S6.xlsx')\n",
    "file_nci60_proteomics =  get_data_path(['guo_2019'], 'Table_S1.xlsx')\n",
    "\n",
    "file_non_tumour_studies = get_local_data_path(['processed', 'correlation_mRNA_protein'], 'non_tumour_studies.csv')\n",
    "file_samples_info = get_local_data_path(['processed', 'correlation_mRNA_protein'], 'samples_info.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Information collected for the Supplemental Table S1B\n",
    "common_samples = {}\n",
    "common_genes = {}\n",
    "transcriptomic_samples = {}\n",
    "proteomic_samples = {}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### CCLE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Transcriptomics Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (1812, 26)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DepMap_ID</th>\n",
       "      <th>cell_line_name</th>\n",
       "      <th>stripped_cell_line_name</th>\n",
       "      <th>CCLE_Name</th>\n",
       "      <th>alias</th>\n",
       "      <th>COSMICID</th>\n",
       "      <th>sex</th>\n",
       "      <th>source</th>\n",
       "      <th>Achilles_n_replicates</th>\n",
       "      <th>cell_line_NNMD</th>\n",
       "      <th>...</th>\n",
       "      <th>primary_or_metastasis</th>\n",
       "      <th>primary_disease</th>\n",
       "      <th>Subtype</th>\n",
       "      <th>age</th>\n",
       "      <th>Sanger_Model_ID</th>\n",
       "      <th>depmap_public_comments</th>\n",
       "      <th>lineage</th>\n",
       "      <th>lineage_subtype</th>\n",
       "      <th>lineage_sub_subtype</th>\n",
       "      <th>lineage_molecular_subtype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000001</td>\n",
       "      <td>NIH:OVCAR-3</td>\n",
       "      <td>NIHOVCAR3</td>\n",
       "      <td>NIHOVCAR3_OVARY</td>\n",
       "      <td>OVCAR3</td>\n",
       "      <td>905933.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>ATCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>Ovarian Cancer</td>\n",
       "      <td>Adenocarcinoma, high grade serous</td>\n",
       "      <td>60.0</td>\n",
       "      <td>SIDM00105</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ovary</td>\n",
       "      <td>ovary_adenocarcinoma</td>\n",
       "      <td>high_grade_serous</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000002</td>\n",
       "      <td>HL-60</td>\n",
       "      <td>HL60</td>\n",
       "      <td>HL60_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>905938.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>ATCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Leukemia</td>\n",
       "      <td>Acute Myelogenous Leukemia (AML), M3 (Promyelo...</td>\n",
       "      <td>35.0</td>\n",
       "      <td>SIDM00829</td>\n",
       "      <td>NaN</td>\n",
       "      <td>blood</td>\n",
       "      <td>AML</td>\n",
       "      <td>M3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    DepMap_ID cell_line_name stripped_cell_line_name  \\\n",
       "0  ACH-000001    NIH:OVCAR-3               NIHOVCAR3   \n",
       "1  ACH-000002          HL-60                    HL60   \n",
       "\n",
       "                                 CCLE_Name   alias  COSMICID     sex source  \\\n",
       "0                          NIHOVCAR3_OVARY  OVCAR3  905933.0  Female   ATCC   \n",
       "1  HL60_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE     NaN  905938.0  Female   ATCC   \n",
       "\n",
       "   Achilles_n_replicates  cell_line_NNMD  ... primary_or_metastasis  \\\n",
       "0                    NaN             NaN  ...            Metastasis   \n",
       "1                    NaN             NaN  ...               Primary   \n",
       "\n",
       "  primary_disease                                            Subtype   age  \\\n",
       "0  Ovarian Cancer                  Adenocarcinoma, high grade serous  60.0   \n",
       "1        Leukemia  Acute Myelogenous Leukemia (AML), M3 (Promyelo...  35.0   \n",
       "\n",
       "   Sanger_Model_ID depmap_public_comments lineage       lineage_subtype  \\\n",
       "0        SIDM00105                    NaN   ovary  ovary_adenocarcinoma   \n",
       "1        SIDM00829                    NaN   blood                   AML   \n",
       "\n",
       "  lineage_sub_subtype  lineage_molecular_subtype  \n",
       "0   high_grade_serous                        NaN  \n",
       "1                  M3                        NaN  \n",
       "\n",
       "[2 rows x 26 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ccle_sample_info = pd.read_csv(file_ccle_sample_info)\n",
    "print(\"Dimensions: \", ccle_sample_info.shape)\n",
    "ccle_sample_info[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (1376, 19182)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TSPAN6 (7105)</th>\n",
       "      <th>TNMD (64102)</th>\n",
       "      <th>DPM1 (8813)</th>\n",
       "      <th>SCYL3 (57147)</th>\n",
       "      <th>C1orf112 (55732)</th>\n",
       "      <th>FGR (2268)</th>\n",
       "      <th>CFH (3075)</th>\n",
       "      <th>FUCA2 (2519)</th>\n",
       "      <th>GCLC (2729)</th>\n",
       "      <th>NFYA (4800)</th>\n",
       "      <th>...</th>\n",
       "      <th>AC090517.4 (54816)</th>\n",
       "      <th>AL160269.1 (11046)</th>\n",
       "      <th>ABCF2-H2BE1 (114483834)</th>\n",
       "      <th>FO393400.1 (149951)</th>\n",
       "      <th>POLR2J3 (548644)</th>\n",
       "      <th>H2BE1 (114483833)</th>\n",
       "      <th>AL445238.1 (647264)</th>\n",
       "      <th>GET1-SH3BGR (106865373)</th>\n",
       "      <th>AC024598.1 (22891)</th>\n",
       "      <th>AC113348.1 (102724657)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ACH-001113</th>\n",
       "      <td>4.990501</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>7.273702</td>\n",
       "      <td>2.765535</td>\n",
       "      <td>4.480265</td>\n",
       "      <td>0.028569</td>\n",
       "      <td>1.269033</td>\n",
       "      <td>3.058316</td>\n",
       "      <td>6.483171</td>\n",
       "      <td>5.05398</td>\n",
       "      <td>...</td>\n",
       "      <td>0.111031</td>\n",
       "      <td>0.15056</td>\n",
       "      <td>1.427606</td>\n",
       "      <td>0.042644</td>\n",
       "      <td>5.781884</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.799087</td>\n",
       "      <td>0.014355</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACH-001289</th>\n",
       "      <td>5.209843</td>\n",
       "      <td>0.545968</td>\n",
       "      <td>7.070604</td>\n",
       "      <td>2.538538</td>\n",
       "      <td>3.510962</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.176323</td>\n",
       "      <td>3.836934</td>\n",
       "      <td>4.200850</td>\n",
       "      <td>3.83289</td>\n",
       "      <td>...</td>\n",
       "      <td>0.310340</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.807355</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.704319</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.464668</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.070389</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 19182 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            TSPAN6 (7105)  TNMD (64102)  DPM1 (8813)  SCYL3 (57147)  \\\n",
       "ACH-001113       4.990501      0.000000     7.273702       2.765535   \n",
       "ACH-001289       5.209843      0.545968     7.070604       2.538538   \n",
       "\n",
       "            C1orf112 (55732)  FGR (2268)  CFH (3075)  FUCA2 (2519)  \\\n",
       "ACH-001113          4.480265    0.028569    1.269033      3.058316   \n",
       "ACH-001289          3.510962    0.000000    0.176323      3.836934   \n",
       "\n",
       "            GCLC (2729)  NFYA (4800)  ...  AC090517.4 (54816)  \\\n",
       "ACH-001113     6.483171      5.05398  ...            0.111031   \n",
       "ACH-001289     4.200850      3.83289  ...            0.310340   \n",
       "\n",
       "            AL160269.1 (11046)  ABCF2-H2BE1 (114483834)  FO393400.1 (149951)  \\\n",
       "ACH-001113             0.15056                 1.427606             0.042644   \n",
       "ACH-001289             0.00000                 0.807355             0.000000   \n",
       "\n",
       "            POLR2J3 (548644)  H2BE1 (114483833)  AL445238.1 (647264)  \\\n",
       "ACH-001113          5.781884                0.0                  0.0   \n",
       "ACH-001289          4.704319                0.0                  0.0   \n",
       "\n",
       "            GET1-SH3BGR (106865373)  AC024598.1 (22891)  \\\n",
       "ACH-001113                 0.799087            0.014355   \n",
       "ACH-001289                 0.464668            0.000000   \n",
       "\n",
       "            AC113348.1 (102724657)  \n",
       "ACH-001113                0.000000  \n",
       "ACH-001289                0.070389  \n",
       "\n",
       "[2 rows x 19182 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ccle_transcriptomics = pd.read_csv(file_ccle_transcriptomics, index_col=0)\n",
    "print(\"Dimensions: \", ccle_transcriptomics.shape)\n",
    "ccle_transcriptomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>LC1SQSF_LUNG</th>\n",
       "      <th>COGAR359_SOFT_TISSUE</th>\n",
       "      <th>COLO794_SKIN</th>\n",
       "      <th>KKU213_BILIARY_TRACT</th>\n",
       "      <th>RT4_URINARY_TRACT</th>\n",
       "      <th>SNU283_LARGE_INTESTINE</th>\n",
       "      <th>NCIH1395_LUNG</th>\n",
       "      <th>DEL_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>SNU1196_BILIARY_TRACT</th>\n",
       "      <th>LC1F_LUNG</th>\n",
       "      <th>...</th>\n",
       "      <th>BL70_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>U343_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>639V_URINARY_TRACT</th>\n",
       "      <th>MON_SOFT_TISSUE</th>\n",
       "      <th>LOXIMVI_SKIN</th>\n",
       "      <th>TOLEDO_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>SSP25_BILIARY_TRACT</th>\n",
       "      <th>ECC2_BILIARY_TRACT</th>\n",
       "      <th>A673_BONE</th>\n",
       "      <th>TT_OESOPHAGUS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TSPAN6</th>\n",
       "      <td>4.990501</td>\n",
       "      <td>5.209843</td>\n",
       "      <td>3.77926</td>\n",
       "      <td>5.726831</td>\n",
       "      <td>7.465648</td>\n",
       "      <td>4.914086</td>\n",
       "      <td>4.032982</td>\n",
       "      <td>0.097611</td>\n",
       "      <td>4.712596</td>\n",
       "      <td>5.101398</td>\n",
       "      <td>...</td>\n",
       "      <td>0.070389</td>\n",
       "      <td>4.69265</td>\n",
       "      <td>5.0268</td>\n",
       "      <td>6.699052</td>\n",
       "      <td>4.173127</td>\n",
       "      <td>0.097611</td>\n",
       "      <td>5.045268</td>\n",
       "      <td>5.805292</td>\n",
       "      <td>4.870858</td>\n",
       "      <td>5.117695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TNMD</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.545968</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.176323</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 1376 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        LC1SQSF_LUNG  COGAR359_SOFT_TISSUE  COLO794_SKIN  \\\n",
       "TSPAN6      4.990501              5.209843       3.77926   \n",
       "TNMD        0.000000              0.545968       0.00000   \n",
       "\n",
       "        KKU213_BILIARY_TRACT  RT4_URINARY_TRACT  SNU283_LARGE_INTESTINE  \\\n",
       "TSPAN6              5.726831           7.465648                4.914086   \n",
       "TNMD                0.000000           0.000000                0.176323   \n",
       "\n",
       "        NCIH1395_LUNG  DEL_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "TSPAN6       4.032982                                0.097611   \n",
       "TNMD         0.000000                                0.000000   \n",
       "\n",
       "        SNU1196_BILIARY_TRACT  LC1F_LUNG  ...  \\\n",
       "TSPAN6               4.712596   5.101398  ...   \n",
       "TNMD                 0.000000   0.000000  ...   \n",
       "\n",
       "        BL70_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  U343_CENTRAL_NERVOUS_SYSTEM  \\\n",
       "TSPAN6                                 0.070389                      4.69265   \n",
       "TNMD                                   0.000000                      0.00000   \n",
       "\n",
       "        639V_URINARY_TRACT  MON_SOFT_TISSUE  LOXIMVI_SKIN  \\\n",
       "TSPAN6              5.0268         6.699052      4.173127   \n",
       "TNMD                0.0000         0.000000      0.000000   \n",
       "\n",
       "        TOLEDO_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  SSP25_BILIARY_TRACT  \\\n",
       "TSPAN6                                   0.097611             5.045268   \n",
       "TNMD                                     0.000000             0.000000   \n",
       "\n",
       "        ECC2_BILIARY_TRACT  A673_BONE  TT_OESOPHAGUS  \n",
       "TSPAN6            5.805292   4.870858       5.117695  \n",
       "TNMD              0.000000   0.000000       0.000000  \n",
       "\n",
       "[2 rows x 1376 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Rename the cell lines given as DepMap ID to CCLE_Name and columns to just contain the hgnc_symbol\n",
    "ccle_transcriptomics = ccle_transcriptomics.rename(index = dict(zip(ccle_sample_info['DepMap_ID'], \n",
    "                                                                    ccle_sample_info['CCLE_Name'])),\n",
    "                                                   columns = lambda x : str(x).split(' ')[0])\n",
    "# Transpose to obtain a matrix similar to other studies\n",
    "ccle_transcriptomics = ccle_transcriptomics.transpose()\n",
    "ccle_transcriptomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert len(ccle_transcriptomics.columns[ccle_transcriptomics.columns.duplicated()]) == 0, \"columns contain duplicates\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (19180, 1376)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>LC1SQSF_LUNG</th>\n",
       "      <th>COGAR359_SOFT_TISSUE</th>\n",
       "      <th>COLO794_SKIN</th>\n",
       "      <th>KKU213_BILIARY_TRACT</th>\n",
       "      <th>RT4_URINARY_TRACT</th>\n",
       "      <th>SNU283_LARGE_INTESTINE</th>\n",
       "      <th>NCIH1395_LUNG</th>\n",
       "      <th>DEL_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>SNU1196_BILIARY_TRACT</th>\n",
       "      <th>LC1F_LUNG</th>\n",
       "      <th>...</th>\n",
       "      <th>BL70_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>U343_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>639V_URINARY_TRACT</th>\n",
       "      <th>MON_SOFT_TISSUE</th>\n",
       "      <th>LOXIMVI_SKIN</th>\n",
       "      <th>TOLEDO_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>SSP25_BILIARY_TRACT</th>\n",
       "      <th>ECC2_BILIARY_TRACT</th>\n",
       "      <th>A673_BONE</th>\n",
       "      <th>TT_OESOPHAGUS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>3.976364</td>\n",
       "      <td>2.134221</td>\n",
       "      <td>2.650765</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.111031</td>\n",
       "      <td>0.226509</td>\n",
       "      <td>0.097611</td>\n",
       "      <td>5.276869</td>\n",
       "      <td>0.111031</td>\n",
       "      <td>2.871844</td>\n",
       "      <td>...</td>\n",
       "      <td>3.854993</td>\n",
       "      <td>2.570463</td>\n",
       "      <td>4.990955</td>\n",
       "      <td>4.296457</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.874797</td>\n",
       "      <td>0.189034</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.878235</td>\n",
       "      <td>2.650765</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>0.014355</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.028569</td>\n",
       "      <td>1.673556</td>\n",
       "      <td>0.757023</td>\n",
       "      <td>0.028569</td>\n",
       "      <td>0.056584</td>\n",
       "      <td>0.014355</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.097611</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.028569</td>\n",
       "      <td>0.014355</td>\n",
       "      <td>0.056584</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.042644</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 1376 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      LC1SQSF_LUNG  COGAR359_SOFT_TISSUE  COLO794_SKIN  KKU213_BILIARY_TRACT  \\\n",
       "A1BG      3.976364              2.134221      2.650765                   0.0   \n",
       "A1CF      0.014355              0.000000      0.000000                   0.0   \n",
       "\n",
       "      RT4_URINARY_TRACT  SNU283_LARGE_INTESTINE  NCIH1395_LUNG  \\\n",
       "A1BG           0.111031                0.226509       0.097611   \n",
       "A1CF           0.028569                1.673556       0.757023   \n",
       "\n",
       "      DEL_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  SNU1196_BILIARY_TRACT  \\\n",
       "A1BG                                5.276869               0.111031   \n",
       "A1CF                                0.028569               0.056584   \n",
       "\n",
       "      LC1F_LUNG  ...  BL70_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "A1BG   2.871844  ...                                 3.854993   \n",
       "A1CF   0.014355  ...                                 0.000000   \n",
       "\n",
       "      U343_CENTRAL_NERVOUS_SYSTEM  639V_URINARY_TRACT  MON_SOFT_TISSUE  \\\n",
       "A1BG                     2.570463            4.990955         4.296457   \n",
       "A1CF                     0.097611            0.000000         0.000000   \n",
       "\n",
       "      LOXIMVI_SKIN  TOLEDO_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "A1BG      0.000000                                   4.874797   \n",
       "A1CF      0.028569                                   0.014355   \n",
       "\n",
       "      SSP25_BILIARY_TRACT  ECC2_BILIARY_TRACT  A673_BONE  TT_OESOPHAGUS  \n",
       "A1BG             0.189034                 0.0   4.878235       2.650765  \n",
       "A1CF             0.056584                 0.0   0.000000       0.042644  \n",
       "\n",
       "[2 rows x 1376 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ccle_transcriptomics_processed = standardised_pipeline_utils.process(ccle_transcriptomics)\n",
    "ccle_transcriptomics_processed[:2]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Proteomic Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (12755, 16384)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Protein_Id</th>\n",
       "      <th>Gene_Symbol</th>\n",
       "      <th>Description</th>\n",
       "      <th>Group_ID</th>\n",
       "      <th>Uniprot</th>\n",
       "      <th>Uniprot_Acc</th>\n",
       "      <th>TenPx01_Peptides</th>\n",
       "      <th>TenPx02_Peptides</th>\n",
       "      <th>TenPx03_Peptides</th>\n",
       "      <th>TenPx04_Peptides</th>\n",
       "      <th>...</th>\n",
       "      <th>Column15949</th>\n",
       "      <th>Column15950</th>\n",
       "      <th>Column15951</th>\n",
       "      <th>Column15952</th>\n",
       "      <th>Column15953</th>\n",
       "      <th>Column15954</th>\n",
       "      <th>Column15955</th>\n",
       "      <th>Column15956</th>\n",
       "      <th>Column15957</th>\n",
       "      <th>Column15958</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>sp|P55011|S12A2_HUMAN</td>\n",
       "      <td>SLC12A2</td>\n",
       "      <td>S12A2_HUMAN Solute carrier family 12 member 2</td>\n",
       "      <td>0</td>\n",
       "      <td>S12A2_HUMAN</td>\n",
       "      <td>P55011</td>\n",
       "      <td>41</td>\n",
       "      <td>16</td>\n",
       "      <td>23</td>\n",
       "      <td>59</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>sp|P35453|HXD13_HUMAN</td>\n",
       "      <td>HOXD13</td>\n",
       "      <td>HXD13_HUMAN Homeobox protein Hox-D13</td>\n",
       "      <td>1</td>\n",
       "      <td>HXD13_HUMAN</td>\n",
       "      <td>P35453</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 16384 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Protein_Id Gene_Symbol  \\\n",
       "0  sp|P55011|S12A2_HUMAN     SLC12A2   \n",
       "1  sp|P35453|HXD13_HUMAN      HOXD13   \n",
       "\n",
       "                                     Description  Group_ID      Uniprot  \\\n",
       "0  S12A2_HUMAN Solute carrier family 12 member 2         0  S12A2_HUMAN   \n",
       "1           HXD13_HUMAN Homeobox protein Hox-D13         1  HXD13_HUMAN   \n",
       "\n",
       "  Uniprot_Acc  TenPx01_Peptides  TenPx02_Peptides  TenPx03_Peptides  \\\n",
       "0      P55011                41                16                23   \n",
       "1      P35453                 0                 0                 0   \n",
       "\n",
       "   TenPx04_Peptides  ...  Column15949  Column15950  Column15951  Column15952  \\\n",
       "0                59  ...          NaN          NaN          NaN          NaN   \n",
       "1                 0  ...          NaN          NaN          NaN          NaN   \n",
       "\n",
       "   Column15953  Column15954  Column15955  Column15956  Column15957  \\\n",
       "0          NaN          NaN          NaN          NaN          NaN   \n",
       "1          NaN          NaN          NaN          NaN          NaN   \n",
       "\n",
       "   Column15958  \n",
       "0          NaN  \n",
       "1          NaN  \n",
       "\n",
       "[2 rows x 16384 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ccle_proteomics = pd.read_excel(io=file_ccle_proteomics, sheet_name=\"Normalized Protein Expression\", engine='openpyxl')\n",
    "print(\"Dimensions: \", ccle_proteomics.shape)\n",
    "ccle_proteomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "ccle_proteomics.set_index('Gene_Symbol', inplace=True)\n",
    "ccle_proteomics = ccle_proteomics.loc[:, ccle_proteomics.columns.str.contains('_TenPx')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CAL120_BREAST_TenPx02', 'SW948_LARGE_INTESTINE_TenPx11',\n",
       "       'SW948_LARGE_INTESTINE_TenPx20', 'CAL120_BREAST_TenPx28',\n",
       "       'HCT15_LARGE_INTESTINE_TenPx30', 'HCT15_LARGE_INTESTINE_TenPx18'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Checking for cell lines repeated in >1 Ten-plexes\n",
    "ccle_proteomics.filter(regex='SW948_LARGE_INTESTINE|CAL120_BREAST|HCT15_LARGE_INTESTINE').columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-warning\">\n",
    "<b>Note:</b> The above 3 cell lines were repeated in different Ten-plexes. However, there is not much correlation between the replicates of different Ten-plexes. So, as mentioned in <a href=https://tinyurl.com/93acy42y>A Guide to the Quantitative Proteomic Profiles of the Cancer Cell Line Encyclopedia</a>, choose to retain the cell lines that only correlate well with the transcriptomics data. \n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Eliminating the cell lines that do not correlate well with transcriptomics data as mentioned in the paper \n",
    "ccle_proteomics.drop(columns=['SW948_LARGE_INTESTINE_TenPx11', 'CAL120_BREAST_TenPx02', 'HCT15_LARGE_INTESTINE_TenPx30'], \n",
    "                    inplace=True)\n",
    "ccle_proteomics = ccle_proteomics.rename(columns = lambda x : str(x).split('_TenPx')[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert len(ccle_proteomics.columns[ccle_proteomics.columns.duplicated()]) == 0, \"columns contain duplicates\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (7372, 375)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MDAMB468_BREAST</th>\n",
       "      <th>SH4_SKIN</th>\n",
       "      <th>AU565_BREAST</th>\n",
       "      <th>KMRC1_KIDNEY</th>\n",
       "      <th>CAL51_BREAST</th>\n",
       "      <th>RPMI7951_SKIN</th>\n",
       "      <th>RERFLCMS_LUNG</th>\n",
       "      <th>IGR37_SKIN</th>\n",
       "      <th>VMRCRCW_KIDNEY</th>\n",
       "      <th>HEP3B217_LIVER</th>\n",
       "      <th>...</th>\n",
       "      <th>NCIH2030_LUNG</th>\n",
       "      <th>22RV1_PROSTATE</th>\n",
       "      <th>A172_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>BT20_BREAST</th>\n",
       "      <th>CALU6_LUNG</th>\n",
       "      <th>FADU_UPPER_AERODIGESTIVE_TRACT</th>\n",
       "      <th>KP4_PANCREAS</th>\n",
       "      <th>MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>OVCAR8_OVARY</th>\n",
       "      <th>THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene_Symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>-0.345562</td>\n",
       "      <td>1.642186</td>\n",
       "      <td>-1.129114</td>\n",
       "      <td>1.493333</td>\n",
       "      <td>0.274652</td>\n",
       "      <td>-0.615667</td>\n",
       "      <td>-0.957249</td>\n",
       "      <td>-0.174494</td>\n",
       "      <td>-0.987932</td>\n",
       "      <td>1.513325</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.981657</td>\n",
       "      <td>-0.921538</td>\n",
       "      <td>3.905544</td>\n",
       "      <td>-0.649348</td>\n",
       "      <td>-1.055916</td>\n",
       "      <td>-1.116991</td>\n",
       "      <td>1.922976</td>\n",
       "      <td>-1.106344</td>\n",
       "      <td>-0.770372</td>\n",
       "      <td>-1.342713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>-0.109405</td>\n",
       "      <td>-0.283335</td>\n",
       "      <td>0.024663</td>\n",
       "      <td>-0.588886</td>\n",
       "      <td>0.673324</td>\n",
       "      <td>0.010466</td>\n",
       "      <td>0.474615</td>\n",
       "      <td>0.037159</td>\n",
       "      <td>-0.362833</td>\n",
       "      <td>-0.225824</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.093176</td>\n",
       "      <td>-0.518686</td>\n",
       "      <td>-0.642067</td>\n",
       "      <td>-0.188224</td>\n",
       "      <td>-0.259101</td>\n",
       "      <td>-0.779655</td>\n",
       "      <td>-0.628770</td>\n",
       "      <td>1.205109</td>\n",
       "      <td>-0.584588</td>\n",
       "      <td>1.622359</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 375 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             MDAMB468_BREAST  SH4_SKIN  AU565_BREAST  KMRC1_KIDNEY  \\\n",
       "Gene_Symbol                                                          \n",
       "A2M                -0.345562  1.642186     -1.129114      1.493333   \n",
       "AAAS               -0.109405 -0.283335      0.024663     -0.588886   \n",
       "\n",
       "             CAL51_BREAST  RPMI7951_SKIN  RERFLCMS_LUNG  IGR37_SKIN  \\\n",
       "Gene_Symbol                                                           \n",
       "A2M              0.274652      -0.615667      -0.957249   -0.174494   \n",
       "AAAS             0.673324       0.010466       0.474615    0.037159   \n",
       "\n",
       "             VMRCRCW_KIDNEY  HEP3B217_LIVER  ...  NCIH2030_LUNG  \\\n",
       "Gene_Symbol                                  ...                  \n",
       "A2M               -0.987932        1.513325  ...      -0.981657   \n",
       "AAAS              -0.362833       -0.225824  ...      -0.093176   \n",
       "\n",
       "             22RV1_PROSTATE  A172_CENTRAL_NERVOUS_SYSTEM  BT20_BREAST  \\\n",
       "Gene_Symbol                                                             \n",
       "A2M               -0.921538                     3.905544    -0.649348   \n",
       "AAAS              -0.518686                    -0.642067    -0.188224   \n",
       "\n",
       "             CALU6_LUNG  FADU_UPPER_AERODIGESTIVE_TRACT  KP4_PANCREAS  \\\n",
       "Gene_Symbol                                                             \n",
       "A2M           -1.055916                       -1.116991      1.922976   \n",
       "AAAS          -0.259101                       -0.779655     -0.628770   \n",
       "\n",
       "             MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  OVCAR8_OVARY  \\\n",
       "Gene_Symbol                                                              \n",
       "A2M                                            -1.106344     -0.770372   \n",
       "AAAS                                            1.205109     -0.584588   \n",
       "\n",
       "             THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \n",
       "Gene_Symbol                                           \n",
       "A2M                                        -1.342713  \n",
       "AAAS                                        1.622359  \n",
       "\n",
       "[2 rows x 375 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ccle_proteomics_processed = standardised_pipeline_utils.process(ccle_proteomics)\n",
    "ccle_proteomics_processed[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "transcriptomic_samples['CCLE'] = ccle_transcriptomics_processed.shape[1]\n",
    "proteomic_samples['CCLE'] = ccle_proteomics_processed.shape[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of common proteins:  6987\n",
      "Number of common samples:  369\n"
     ]
    }
   ],
   "source": [
    "ccle_transcriptomics_processed, ccle_proteomics_processed = standardised_pipeline_utils.match_proteins_samples( \\\n",
    "                                                                            ccle_transcriptomics_processed, \n",
    "                                                                            ccle_proteomics_processed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "common_samples['CCLE'] = ccle_transcriptomics_processed.shape[1]\n",
    "common_genes['CCLE'] = ccle_transcriptomics_processed.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Spearman Correlation:  0.4566\n"
     ]
    }
   ],
   "source": [
    "correlation_ccle = standardised_pipeline_utils.correlate_genewise(ccle_transcriptomics_processed, \n",
    "                                                                   ccle_proteomics_processed, 'CCLE')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Pearson Correlation:  0.4776\n"
     ]
    }
   ],
   "source": [
    "correlation_ccle_pearson = standardised_pipeline_utils.correlate_genewise(ccle_transcriptomics_processed, \n",
    "                                                                   ccle_proteomics_processed, 'CCLE', method='pearson')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### GTEX"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Transcriptomic Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (12245, 36)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl_id</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>hgnc_name</th>\n",
       "      <th>hgnc_symbol</th>\n",
       "      <th>Adrenal Gland</th>\n",
       "      <th>Artery Aorta</th>\n",
       "      <th>Artery Coronary</th>\n",
       "      <th>Artery Tibial</th>\n",
       "      <th>Brain Cerebellum</th>\n",
       "      <th>Brain Cortex</th>\n",
       "      <th>...</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin Unexpo</th>\n",
       "      <th>Skin SunExpo</th>\n",
       "      <th>Small Intestine</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>7105.0</td>\n",
       "      <td>tetraspanin 6</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>0.28</td>\n",
       "      <td>-0.68</td>\n",
       "      <td>-0.48</td>\n",
       "      <td>-0.88</td>\n",
       "      <td>-1.64</td>\n",
       "      <td>-0.15</td>\n",
       "      <td>...</td>\n",
       "      <td>0.79</td>\n",
       "      <td>-1.26</td>\n",
       "      <td>-0.46</td>\n",
       "      <td>0.46</td>\n",
       "      <td>-0.42</td>\n",
       "      <td>0.57</td>\n",
       "      <td>2.54</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.54</td>\n",
       "      <td>0.74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>8813.0</td>\n",
       "      <td>dolichyl-phosphate mannosyltransferase subunit...</td>\n",
       "      <td>DPM1</td>\n",
       "      <td>1.46</td>\n",
       "      <td>0.8</td>\n",
       "      <td>-0.35</td>\n",
       "      <td>0.34</td>\n",
       "      <td>-2.32</td>\n",
       "      <td>-1.9</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.95</td>\n",
       "      <td>0.24</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>-0.8</td>\n",
       "      <td>0.42</td>\n",
       "      <td>-0.4</td>\n",
       "      <td>-0.07</td>\n",
       "      <td>0.18</td>\n",
       "      <td>-0.74</td>\n",
       "      <td>-0.84</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        ensembl_id  entrez_id  \\\n",
       "0  ENSG00000000003     7105.0   \n",
       "1  ENSG00000000419     8813.0   \n",
       "\n",
       "                                           hgnc_name hgnc_symbol  \\\n",
       "0                                      tetraspanin 6      TSPAN6   \n",
       "1  dolichyl-phosphate mannosyltransferase subunit...        DPM1   \n",
       "\n",
       "  Adrenal Gland Artery Aorta Artery Coronary Artery Tibial Brain Cerebellum  \\\n",
       "0          0.28        -0.68           -0.48         -0.88            -1.64   \n",
       "1          1.46          0.8           -0.35          0.34            -2.32   \n",
       "\n",
       "  Brain Cortex  ... Prostate Skin Unexpo Skin SunExpo Small Intestine Spleen  \\\n",
       "0        -0.15  ...     0.79       -1.26        -0.46            0.46  -0.42   \n",
       "1         -1.9  ...    -0.95        0.24         -0.2            -0.8   0.42   \n",
       "\n",
       "  Stomach Testis Thyroid Uterus Vagina  \n",
       "0    0.57   2.54    0.35   0.54   0.74  \n",
       "1    -0.4  -0.07    0.18  -0.74  -0.84  \n",
       "\n",
       "[2 rows x 36 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtex_transcriptomics = pd.read_excel(io=file_gtex_transcriptomics, sheet_name='D RNA TS score', skiprows=3)\n",
    "print(\"Dimensions: \", gtex_transcriptomics.shape)\n",
    "gtex_transcriptomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "gtex_transcriptomics.set_index('hgnc_symbol', inplace=True)\n",
    "gtex_transcriptomics.drop(columns=['ensembl_id', 'entrez_id', 'hgnc_name'], inplace=True)\n",
    "\n",
    "# Replace the different kinds of null value representation into a recognizable form using np.Nan\n",
    "gtex_transcriptomics.replace('NA;all_tissues_tpm_less_1', np.nan, inplace=True)\n",
    "gtex_transcriptomics.replace('^.*NA_raw_tpm_less_1 ;.*$', np.nan, regex=True, inplace=True)\n",
    "gtex_transcriptomics = gtex_transcriptomics.apply(pd.to_numeric)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (12099, 32)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Adrenal Gland</th>\n",
       "      <th>Artery Aorta</th>\n",
       "      <th>Artery Coronary</th>\n",
       "      <th>Artery Tibial</th>\n",
       "      <th>Brain Cerebellum</th>\n",
       "      <th>Brain Cortex</th>\n",
       "      <th>Breast</th>\n",
       "      <th>Colon Sigmoid</th>\n",
       "      <th>Colon Transverse</th>\n",
       "      <th>GE junction</th>\n",
       "      <th>...</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin Unexpo</th>\n",
       "      <th>Skin SunExpo</th>\n",
       "      <th>Small Intestine</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hgnc_symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-1.00</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.86</td>\n",
       "      <td>0.46</td>\n",
       "      <td>0.87</td>\n",
       "      <td>0.64</td>\n",
       "      <td>0.67</td>\n",
       "      <td>-0.22</td>\n",
       "      <td>0.47</td>\n",
       "      <td>-0.27</td>\n",
       "      <td>...</td>\n",
       "      <td>0.43</td>\n",
       "      <td>-1.76</td>\n",
       "      <td>-0.50</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.54</td>\n",
       "      <td>-0.11</td>\n",
       "      <td>-0.77</td>\n",
       "      <td>0.64</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>0.06</td>\n",
       "      <td>-0.62</td>\n",
       "      <td>0.27</td>\n",
       "      <td>-0.26</td>\n",
       "      <td>0.48</td>\n",
       "      <td>0.32</td>\n",
       "      <td>-0.31</td>\n",
       "      <td>-0.16</td>\n",
       "      <td>3.99</td>\n",
       "      <td>-0.31</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.03</td>\n",
       "      <td>0.74</td>\n",
       "      <td>0.03</td>\n",
       "      <td>3.65</td>\n",
       "      <td>-0.77</td>\n",
       "      <td>2.39</td>\n",
       "      <td>0.29</td>\n",
       "      <td>-0.48</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Adrenal Gland  Artery Aorta  Artery Coronary  Artery Tibial  \\\n",
       "hgnc_symbol                                                                \n",
       "A1BG                 -1.00          0.13             0.86           0.46   \n",
       "A1CF                  0.06         -0.62             0.27          -0.26   \n",
       "\n",
       "             Brain Cerebellum  Brain Cortex  Breast  Colon Sigmoid  \\\n",
       "hgnc_symbol                                                          \n",
       "A1BG                     0.87          0.64    0.67          -0.22   \n",
       "A1CF                     0.48          0.32   -0.31          -0.16   \n",
       "\n",
       "             Colon Transverse  GE junction  ...  Prostate  Skin Unexpo  \\\n",
       "hgnc_symbol                                 ...                          \n",
       "A1BG                     0.47        -0.27  ...      0.43        -1.76   \n",
       "A1CF                     3.99        -0.31  ...     -0.03         0.74   \n",
       "\n",
       "             Skin SunExpo  Small Intestine  Spleen  Stomach  Testis  Thyroid  \\\n",
       "hgnc_symbol                                                                    \n",
       "A1BG                -0.50             0.07    0.54    -0.11   -0.77     0.64   \n",
       "A1CF                 0.03             3.65   -0.77     2.39    0.29    -0.48   \n",
       "\n",
       "             Uterus  Vagina  \n",
       "hgnc_symbol                  \n",
       "A1BG            1.0    0.78  \n",
       "A1CF           -0.5    0.14  \n",
       "\n",
       "[2 rows x 32 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtex_transcriptomics_processed = standardised_pipeline_utils.process(gtex_transcriptomics)\n",
    "gtex_transcriptomics_processed[:2]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Proteomic Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (12627, 36)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl_id</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>hgnc_name</th>\n",
       "      <th>hgnc_symbol</th>\n",
       "      <th>Adrenal Gland</th>\n",
       "      <th>Artery Aorta</th>\n",
       "      <th>Artery Coronary</th>\n",
       "      <th>Artery Tibial</th>\n",
       "      <th>Brain Cerebellum</th>\n",
       "      <th>Brain Cortex</th>\n",
       "      <th>...</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin Unexpo</th>\n",
       "      <th>Skin SunExpo</th>\n",
       "      <th>Small Intestine</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>7105.0</td>\n",
       "      <td>tetraspanin 6</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.81</td>\n",
       "      <td>0.02;NA_one_rep_in_raw</td>\n",
       "      <td>0.26</td>\n",
       "      <td>-1.32</td>\n",
       "      <td>-0.11;NA_one_rep_in_raw</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.07</td>\n",
       "      <td>0.36</td>\n",
       "      <td>0.52</td>\n",
       "      <td>0.23</td>\n",
       "      <td>-0.78</td>\n",
       "      <td>1.18</td>\n",
       "      <td>0.66</td>\n",
       "      <td>-0.03</td>\n",
       "      <td>-0.09</td>\n",
       "      <td>-0.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>8813.0</td>\n",
       "      <td>dolichyl-phosphate mannosyltransferase subunit...</td>\n",
       "      <td>DPM1</td>\n",
       "      <td>1.2</td>\n",
       "      <td>-1.3</td>\n",
       "      <td>-0.8</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>-2.15</td>\n",
       "      <td>-0.46</td>\n",
       "      <td>...</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.3</td>\n",
       "      <td>-0.22</td>\n",
       "      <td>-0.57</td>\n",
       "      <td>-0.08</td>\n",
       "      <td>1.36</td>\n",
       "      <td>1.04</td>\n",
       "      <td>1.41</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        ensembl_id  entrez_id  \\\n",
       "0  ENSG00000000003     7105.0   \n",
       "1  ENSG00000000419     8813.0   \n",
       "\n",
       "                                           hgnc_name hgnc_symbol  \\\n",
       "0                                      tetraspanin 6      TSPAN6   \n",
       "1  dolichyl-phosphate mannosyltransferase subunit...        DPM1   \n",
       "\n",
       "  Adrenal Gland Artery Aorta         Artery Coronary Artery Tibial  \\\n",
       "0          0.26         0.81  0.02;NA_one_rep_in_raw          0.26   \n",
       "1           1.2         -1.3                    -0.8          -0.1   \n",
       "\n",
       "  Brain Cerebellum             Brain Cortex  ... Prostate Skin Unexpo  \\\n",
       "0            -1.32  -0.11;NA_one_rep_in_raw  ...    -0.07        0.36   \n",
       "1            -2.15                    -0.46  ...     0.14         0.3   \n",
       "\n",
       "  Skin SunExpo Small Intestine Spleen Stomach Testis Thyroid Uterus Vagina  \n",
       "0         0.52            0.23  -0.78    1.18   0.66   -0.03  -0.09  -0.28  \n",
       "1        -0.22           -0.57  -0.08    1.36   1.04    1.41    0.1      0  \n",
       "\n",
       "[2 rows x 36 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtex_proteomics = pd.read_excel(io=file_gtex_proteomics, sheet_name='G protein TS score', skiprows=2, engine='openpyxl')\n",
    "print(\"Dimensions: \", gtex_proteomics.shape)\n",
    "gtex_proteomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "gtex_proteomics.set_index('hgnc_symbol', inplace=True)\n",
    "gtex_proteomics.drop(columns=['ensembl_id', 'entrez_id', 'hgnc_name'], inplace=True)\n",
    "gtex_proteomics.replace('^.*NA_one_rep_in_raw.*$', np.nan, regex=True, inplace=True)\n",
    "gtex_proteomics = gtex_proteomics.apply(pd.to_numeric)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (7661, 32)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Adrenal Gland</th>\n",
       "      <th>Artery Aorta</th>\n",
       "      <th>Artery Coronary</th>\n",
       "      <th>Artery Tibial</th>\n",
       "      <th>Brain Cerebellum</th>\n",
       "      <th>Brain Cortex</th>\n",
       "      <th>Breast</th>\n",
       "      <th>Colon Sigmoid</th>\n",
       "      <th>Colon Transverse</th>\n",
       "      <th>GE junction</th>\n",
       "      <th>...</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin Unexpo</th>\n",
       "      <th>Skin SunExpo</th>\n",
       "      <th>Small Intestine</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hgnc_symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-1.39</td>\n",
       "      <td>0.35</td>\n",
       "      <td>1.03</td>\n",
       "      <td>-0.45</td>\n",
       "      <td>-2.67</td>\n",
       "      <td>-2.14</td>\n",
       "      <td>1.7</td>\n",
       "      <td>0.1</td>\n",
       "      <td>-0.42</td>\n",
       "      <td>0.37</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.33</td>\n",
       "      <td>0.75</td>\n",
       "      <td>0.15</td>\n",
       "      <td>-1.24</td>\n",
       "      <td>-1.65</td>\n",
       "      <td>-0.59</td>\n",
       "      <td>-0.55</td>\n",
       "      <td>-0.25</td>\n",
       "      <td>0.72</td>\n",
       "      <td>0.97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>-10.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-10.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Adrenal Gland  Artery Aorta  Artery Coronary  Artery Tibial  \\\n",
       "hgnc_symbol                                                                \n",
       "A1BG                 -1.39          0.35             1.03          -0.45   \n",
       "A1CF                -10.00           NaN             0.00          -1.00   \n",
       "\n",
       "             Brain Cerebellum  Brain Cortex  Breast  Colon Sigmoid  \\\n",
       "hgnc_symbol                                                          \n",
       "A1BG                    -2.67         -2.14     1.7            0.1   \n",
       "A1CF                      NaN          0.00     0.0            0.0   \n",
       "\n",
       "             Colon Transverse  GE junction  ...  Prostate  Skin Unexpo  \\\n",
       "hgnc_symbol                                 ...                          \n",
       "A1BG                    -0.42         0.37  ...     -0.33         0.75   \n",
       "A1CF                     0.00         0.00  ...      0.00         0.00   \n",
       "\n",
       "             Skin SunExpo  Small Intestine  Spleen  Stomach  Testis  Thyroid  \\\n",
       "hgnc_symbol                                                                    \n",
       "A1BG                 0.15            -1.24   -1.65    -0.59   -0.55    -0.25   \n",
       "A1CF                  NaN             0.00    0.00     0.00  -10.00     0.00   \n",
       "\n",
       "             Uterus  Vagina  \n",
       "hgnc_symbol                  \n",
       "A1BG           0.72    0.97  \n",
       "A1CF           0.00     NaN  \n",
       "\n",
       "[2 rows x 32 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtex_proteomics_processed = standardised_pipeline_utils.process(gtex_proteomics)\n",
    "gtex_proteomics_processed[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "transcriptomic_samples['GTEx'] = gtex_transcriptomics_processed.shape[1]\n",
    "proteomic_samples['GTEx'] = gtex_proteomics_processed.shape[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of common proteins:  7471\n",
      "Number of common samples:  32\n"
     ]
    }
   ],
   "source": [
    "gtex_transcriptomics_processed, gtex_proteomics_processed = standardised_pipeline_utils.match_proteins_samples( \\\n",
    "                                                                            gtex_transcriptomics_processed, \n",
    "                                                                            gtex_proteomics_processed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "common_samples['GTEx'] = gtex_transcriptomics_processed.shape[1]\n",
    "common_genes['GTEx'] = gtex_proteomics_processed.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Spearman Correlation:  0.5087\n"
     ]
    }
   ],
   "source": [
    "correlation_gtex = standardised_pipeline_utils.correlate_genewise(gtex_transcriptomics_processed,\n",
    "                                                                   gtex_proteomics_processed, 'GTEx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Pearson Correlation:  0.5911\n"
     ]
    }
   ],
   "source": [
    "correlation_gtex_pearson = standardised_pipeline_utils.correlate_genewise(gtex_transcriptomics_processed, \n",
    "                                                                   gtex_proteomics_processed, 'GTEx', method='pearson')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### NCI-60 cancer cell lines"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Transcriptomic Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (23059, 61)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>BR:MCF7</th>\n",
       "      <th>BR:MDA-MB-231</th>\n",
       "      <th>BR:HS 578T</th>\n",
       "      <th>BR:BT-549</th>\n",
       "      <th>BR:T-47D</th>\n",
       "      <th>CNS:SF-268</th>\n",
       "      <th>CNS:SF-295</th>\n",
       "      <th>CNS:SF-539</th>\n",
       "      <th>CNS:SNB-19</th>\n",
       "      <th>...</th>\n",
       "      <th>PR:PC-3</th>\n",
       "      <th>PR:DU-145</th>\n",
       "      <th>RE:786-0</th>\n",
       "      <th>RE:A498</th>\n",
       "      <th>RE:ACHN</th>\n",
       "      <th>RE:CAKI-1</th>\n",
       "      <th>RE:RXF 393</th>\n",
       "      <th>RE:SN12C</th>\n",
       "      <th>RE:TK-10</th>\n",
       "      <th>RE:UO-31</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LOC729737</td>\n",
       "      <td>9.412</td>\n",
       "      <td>9.313</td>\n",
       "      <td>8.476</td>\n",
       "      <td>8.175</td>\n",
       "      <td>9.621</td>\n",
       "      <td>8.712</td>\n",
       "      <td>8.422</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.472</td>\n",
       "      <td>...</td>\n",
       "      <td>8.250</td>\n",
       "      <td>9.371</td>\n",
       "      <td>8.459</td>\n",
       "      <td>9.132</td>\n",
       "      <td>8.946</td>\n",
       "      <td>8.376</td>\n",
       "      <td>8.876</td>\n",
       "      <td>9.173</td>\n",
       "      <td>8.377</td>\n",
       "      <td>8.594</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CICP3</td>\n",
       "      <td>6.152</td>\n",
       "      <td>6.484</td>\n",
       "      <td>6.477</td>\n",
       "      <td>6.223</td>\n",
       "      <td>6.434</td>\n",
       "      <td>6.282</td>\n",
       "      <td>6.075</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.453</td>\n",
       "      <td>...</td>\n",
       "      <td>5.966</td>\n",
       "      <td>6.227</td>\n",
       "      <td>6.472</td>\n",
       "      <td>6.322</td>\n",
       "      <td>6.477</td>\n",
       "      <td>6.464</td>\n",
       "      <td>6.623</td>\n",
       "      <td>6.313</td>\n",
       "      <td>6.076</td>\n",
       "      <td>5.933</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 61 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        GENE  BR:MCF7  BR:MDA-MB-231  BR:HS 578T  BR:BT-549  BR:T-47D  \\\n",
       "0  LOC729737    9.412          9.313       8.476      8.175     9.621   \n",
       "1      CICP3    6.152          6.484       6.477      6.223     6.434   \n",
       "\n",
       "   CNS:SF-268  CNS:SF-295  CNS:SF-539  CNS:SNB-19  ...  PR:PC-3  PR:DU-145  \\\n",
       "0       8.712       8.422         NaN       8.472  ...    8.250      9.371   \n",
       "1       6.282       6.075         NaN       6.453  ...    5.966      6.227   \n",
       "\n",
       "   RE:786-0  RE:A498  RE:ACHN  RE:CAKI-1  RE:RXF 393  RE:SN12C  RE:TK-10  \\\n",
       "0     8.459    9.132    8.946      8.376       8.876     9.173     8.377   \n",
       "1     6.472    6.322    6.477      6.464       6.623     6.313     6.076   \n",
       "\n",
       "   RE:UO-31  \n",
       "0     8.594  \n",
       "1     5.933  \n",
       "\n",
       "[2 rows x 61 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nci60_transcriptomics = pd.read_excel(io=file_nci60_transcriptomics, sheet_name='B')\n",
    "print(\"Dimensions: \", nci60_transcriptomics.shape)\n",
    "nci60_transcriptomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "nci60_transcriptomics = nci60_transcriptomics.dropna(subset=['GENE']).set_index('GENE')\n",
    "nci60_transcriptomics = nci60_transcriptomics.rename(columns= lambda x: re.sub(\"[-\\s]\", \"\", str(x)))  \\\n",
    "                                             .rename(columns= lambda x: re.sub(\"\\([^)]*\\)\", \"\", str(x))) \\\n",
    "                                             .rename(columns= lambda x: str(x).replace(':', '_')) \\\n",
    "                                             .rename(columns = {'LC_A549/ATCC': 'LC_A549', \n",
    "                                                                'OV_NCI/ADRRES': 'OV_NCIADRRES'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (23059, 60)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BR_MCF7</th>\n",
       "      <th>BR_MDAMB231</th>\n",
       "      <th>BR_HS578T</th>\n",
       "      <th>BR_BT549</th>\n",
       "      <th>BR_T47D</th>\n",
       "      <th>CNS_SF268</th>\n",
       "      <th>CNS_SF295</th>\n",
       "      <th>CNS_SF539</th>\n",
       "      <th>CNS_SNB19</th>\n",
       "      <th>CNS_SNB75</th>\n",
       "      <th>...</th>\n",
       "      <th>PR_PC3</th>\n",
       "      <th>PR_DU145</th>\n",
       "      <th>RE_7860</th>\n",
       "      <th>RE_A498</th>\n",
       "      <th>RE_ACHN</th>\n",
       "      <th>RE_CAKI1</th>\n",
       "      <th>RE_RXF393</th>\n",
       "      <th>RE_SN12C</th>\n",
       "      <th>RE_TK10</th>\n",
       "      <th>RE_UO31</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GENE</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>6.963</td>\n",
       "      <td>6.418</td>\n",
       "      <td>7.593</td>\n",
       "      <td>7.444</td>\n",
       "      <td>6.669</td>\n",
       "      <td>6.610</td>\n",
       "      <td>6.432</td>\n",
       "      <td>4.795</td>\n",
       "      <td>6.219</td>\n",
       "      <td>5.694</td>\n",
       "      <td>...</td>\n",
       "      <td>5.653</td>\n",
       "      <td>6.229</td>\n",
       "      <td>5.681</td>\n",
       "      <td>5.684</td>\n",
       "      <td>5.725</td>\n",
       "      <td>6.371</td>\n",
       "      <td>6.698</td>\n",
       "      <td>6.973</td>\n",
       "      <td>5.858</td>\n",
       "      <td>6.024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <td>6.230</td>\n",
       "      <td>5.271</td>\n",
       "      <td>5.994</td>\n",
       "      <td>6.434</td>\n",
       "      <td>6.129</td>\n",
       "      <td>5.536</td>\n",
       "      <td>5.803</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.230</td>\n",
       "      <td>4.972</td>\n",
       "      <td>...</td>\n",
       "      <td>4.654</td>\n",
       "      <td>5.542</td>\n",
       "      <td>4.819</td>\n",
       "      <td>5.014</td>\n",
       "      <td>5.058</td>\n",
       "      <td>5.554</td>\n",
       "      <td>5.373</td>\n",
       "      <td>5.091</td>\n",
       "      <td>4.886</td>\n",
       "      <td>5.202</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 60 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          BR_MCF7  BR_MDAMB231  BR_HS578T  BR_BT549  BR_T47D  CNS_SF268  \\\n",
       "GENE                                                                      \n",
       "A1BG        6.963        6.418      7.593     7.444    6.669      6.610   \n",
       "A1BG-AS1    6.230        5.271      5.994     6.434    6.129      5.536   \n",
       "\n",
       "          CNS_SF295  CNS_SF539  CNS_SNB19  CNS_SNB75  ...  PR_PC3  PR_DU145  \\\n",
       "GENE                                                  ...                     \n",
       "A1BG          6.432      4.795      6.219      5.694  ...   5.653     6.229   \n",
       "A1BG-AS1      5.803        NaN      5.230      4.972  ...   4.654     5.542   \n",
       "\n",
       "          RE_7860  RE_A498  RE_ACHN  RE_CAKI1  RE_RXF393  RE_SN12C  RE_TK10  \\\n",
       "GENE                                                                          \n",
       "A1BG        5.681    5.684    5.725     6.371      6.698     6.973    5.858   \n",
       "A1BG-AS1    4.819    5.014    5.058     5.554      5.373     5.091    4.886   \n",
       "\n",
       "          RE_UO31  \n",
       "GENE               \n",
       "A1BG        6.024  \n",
       "A1BG-AS1    5.202  \n",
       "\n",
       "[2 rows x 60 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nci60_transcriptomics_processed = standardised_pipeline_utils.process(nci60_transcriptomics)\n",
    "nci60_transcriptomics_processed[:2]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Proteomic Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (3171, 68)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein accession number</th>\n",
       "      <th>gene symbol</th>\n",
       "      <th>numPep</th>\n",
       "      <th>R2</th>\n",
       "      <th>best_mscore</th>\n",
       "      <th>numNA</th>\n",
       "      <th>transition_group_id</th>\n",
       "      <th>average protein intensity</th>\n",
       "      <th>BR_BT549</th>\n",
       "      <th>BR_HS578T</th>\n",
       "      <th>...</th>\n",
       "      <th>PR_DU145</th>\n",
       "      <th>PR_PC3</th>\n",
       "      <th>RE_7860</th>\n",
       "      <th>RE_A498</th>\n",
       "      <th>RE_ACHN</th>\n",
       "      <th>RE_CAKI1</th>\n",
       "      <th>RE_RXF393</th>\n",
       "      <th>RE_SN12C</th>\n",
       "      <th>RE_TK10</th>\n",
       "      <th>RE_UO31</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0AV96</td>\n",
       "      <td>RBM47</td>\n",
       "      <td>2</td>\n",
       "      <td>0.461295</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>89197_VPEGVAGAPNEAALLALMER_3</td>\n",
       "      <td>321415.0725</td>\n",
       "      <td>252295.5</td>\n",
       "      <td>320967.0</td>\n",
       "      <td>...</td>\n",
       "      <td>231154.8</td>\n",
       "      <td>240414.8</td>\n",
       "      <td>260905.1</td>\n",
       "      <td>342500.2</td>\n",
       "      <td>376886.1</td>\n",
       "      <td>289172.1</td>\n",
       "      <td>306021.4</td>\n",
       "      <td>276934.6</td>\n",
       "      <td>378189.4</td>\n",
       "      <td>313858.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0AVT1</td>\n",
       "      <td>UBA6</td>\n",
       "      <td>9</td>\n",
       "      <td>0.684149</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>22</td>\n",
       "      <td>48306_LETGQFLTFR_2</td>\n",
       "      <td>325395.9267</td>\n",
       "      <td>325644.3</td>\n",
       "      <td>47653.2</td>\n",
       "      <td>...</td>\n",
       "      <td>47653.2</td>\n",
       "      <td>313214.0</td>\n",
       "      <td>301022.1</td>\n",
       "      <td>196917.8</td>\n",
       "      <td>47653.2</td>\n",
       "      <td>378384.5</td>\n",
       "      <td>913480.0</td>\n",
       "      <td>333520.5</td>\n",
       "      <td>262599.8</td>\n",
       "      <td>268524.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  protein accession number gene symbol  numPep        R2  best_mscore  numNA  \\\n",
       "0                   A0AV96       RBM47       2  0.461295     0.000013      0   \n",
       "1                   A0AVT1        UBA6       9  0.684149     0.000038     22   \n",
       "\n",
       "            transition_group_id  average protein intensity  BR_BT549  \\\n",
       "0  89197_VPEGVAGAPNEAALLALMER_3                321415.0725  252295.5   \n",
       "1            48306_LETGQFLTFR_2                325395.9267  325644.3   \n",
       "\n",
       "   BR_HS578T  ...  PR_DU145    PR_PC3   RE_7860   RE_A498   RE_ACHN  RE_CAKI1  \\\n",
       "0   320967.0  ...  231154.8  240414.8  260905.1  342500.2  376886.1  289172.1   \n",
       "1    47653.2  ...   47653.2  313214.0  301022.1  196917.8   47653.2  378384.5   \n",
       "\n",
       "   RE_RXF393  RE_SN12C   RE_TK10   RE_UO31  \n",
       "0   306021.4  276934.6  378189.4  313858.5  \n",
       "1   913480.0  333520.5  262599.8  268524.1  \n",
       "\n",
       "[2 rows x 68 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nci60_proteomics = pd.read_excel(io=file_nci60_proteomics, sheet_name='E')\n",
    "print(\"Dimensions: \", nci60_proteomics.shape)\n",
    "nci60_proteomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BR_BT549</th>\n",
       "      <th>BR_HS578T</th>\n",
       "      <th>BR_MCF7</th>\n",
       "      <th>BR_MDAMB231</th>\n",
       "      <th>BR_MDAMB468</th>\n",
       "      <th>BR_T47D</th>\n",
       "      <th>CNS_SF268</th>\n",
       "      <th>CNS_SF295</th>\n",
       "      <th>CNS_SF539</th>\n",
       "      <th>CNS_SNB19</th>\n",
       "      <th>...</th>\n",
       "      <th>PR_DU145</th>\n",
       "      <th>PR_PC3</th>\n",
       "      <th>RE_7860</th>\n",
       "      <th>RE_A498</th>\n",
       "      <th>RE_ACHN</th>\n",
       "      <th>RE_CAKI1</th>\n",
       "      <th>RE_RXF393</th>\n",
       "      <th>RE_SN12C</th>\n",
       "      <th>RE_TK10</th>\n",
       "      <th>RE_UO31</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>RBM47</th>\n",
       "      <td>252295.5</td>\n",
       "      <td>320967.0</td>\n",
       "      <td>386714.5</td>\n",
       "      <td>308863.3</td>\n",
       "      <td>291428.0</td>\n",
       "      <td>296138.2</td>\n",
       "      <td>161876.4</td>\n",
       "      <td>328064.5</td>\n",
       "      <td>259639.0</td>\n",
       "      <td>240479.0</td>\n",
       "      <td>...</td>\n",
       "      <td>231154.8</td>\n",
       "      <td>240414.8</td>\n",
       "      <td>260905.1</td>\n",
       "      <td>342500.2</td>\n",
       "      <td>376886.1</td>\n",
       "      <td>289172.1</td>\n",
       "      <td>306021.4</td>\n",
       "      <td>276934.6</td>\n",
       "      <td>378189.4</td>\n",
       "      <td>313858.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UBA6</th>\n",
       "      <td>325644.3</td>\n",
       "      <td>47653.2</td>\n",
       "      <td>394142.5</td>\n",
       "      <td>47653.2</td>\n",
       "      <td>47653.2</td>\n",
       "      <td>188434.5</td>\n",
       "      <td>409606.1</td>\n",
       "      <td>275464.7</td>\n",
       "      <td>188715.0</td>\n",
       "      <td>47653.2</td>\n",
       "      <td>...</td>\n",
       "      <td>47653.2</td>\n",
       "      <td>313214.0</td>\n",
       "      <td>301022.1</td>\n",
       "      <td>196917.8</td>\n",
       "      <td>47653.2</td>\n",
       "      <td>378384.5</td>\n",
       "      <td>913480.0</td>\n",
       "      <td>333520.5</td>\n",
       "      <td>262599.8</td>\n",
       "      <td>268524.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 60 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             BR_BT549  BR_HS578T   BR_MCF7  BR_MDAMB231  BR_MDAMB468  \\\n",
       "gene symbol                                                            \n",
       "RBM47        252295.5   320967.0  386714.5     308863.3     291428.0   \n",
       "UBA6         325644.3    47653.2  394142.5      47653.2      47653.2   \n",
       "\n",
       "              BR_T47D  CNS_SF268  CNS_SF295  CNS_SF539  CNS_SNB19  ...  \\\n",
       "gene symbol                                                        ...   \n",
       "RBM47        296138.2   161876.4   328064.5   259639.0   240479.0  ...   \n",
       "UBA6         188434.5   409606.1   275464.7   188715.0    47653.2  ...   \n",
       "\n",
       "             PR_DU145    PR_PC3   RE_7860   RE_A498   RE_ACHN  RE_CAKI1  \\\n",
       "gene symbol                                                               \n",
       "RBM47        231154.8  240414.8  260905.1  342500.2  376886.1  289172.1   \n",
       "UBA6          47653.2  313214.0  301022.1  196917.8   47653.2  378384.5   \n",
       "\n",
       "             RE_RXF393  RE_SN12C   RE_TK10   RE_UO31  \n",
       "gene symbol                                           \n",
       "RBM47         306021.4  276934.6  378189.4  313858.5  \n",
       "UBA6          913480.0  333520.5  262599.8  268524.1  \n",
       "\n",
       "[2 rows x 60 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nci60_proteomics.drop(columns = ['protein accession number', 'numPep', 'R2', 'best_mscore', \n",
    "                                 'numNA', 'transition_group_id', 'average protein intensity'], inplace=True)\n",
    "nci60_proteomics = nci60_proteomics.dropna(subset=['gene symbol']).set_index('gene symbol')\n",
    "nci60_proteomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (3157, 60)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BR_BT549</th>\n",
       "      <th>BR_HS578T</th>\n",
       "      <th>BR_MCF7</th>\n",
       "      <th>BR_MDAMB231</th>\n",
       "      <th>BR_MDAMB468</th>\n",
       "      <th>BR_T47D</th>\n",
       "      <th>CNS_SF268</th>\n",
       "      <th>CNS_SF295</th>\n",
       "      <th>CNS_SF539</th>\n",
       "      <th>CNS_SNB19</th>\n",
       "      <th>...</th>\n",
       "      <th>PR_DU145</th>\n",
       "      <th>PR_PC3</th>\n",
       "      <th>RE_7860</th>\n",
       "      <th>RE_A498</th>\n",
       "      <th>RE_ACHN</th>\n",
       "      <th>RE_CAKI1</th>\n",
       "      <th>RE_RXF393</th>\n",
       "      <th>RE_SN12C</th>\n",
       "      <th>RE_TK10</th>\n",
       "      <th>RE_UO31</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>8331.8</td>\n",
       "      <td>64198.0</td>\n",
       "      <td>62750.7</td>\n",
       "      <td>63921.5</td>\n",
       "      <td>77924.6</td>\n",
       "      <td>73633.1</td>\n",
       "      <td>87421.6</td>\n",
       "      <td>76357.9</td>\n",
       "      <td>84516.4</td>\n",
       "      <td>68816.6</td>\n",
       "      <td>...</td>\n",
       "      <td>8331.8</td>\n",
       "      <td>118359.2</td>\n",
       "      <td>54281.8</td>\n",
       "      <td>50204.8</td>\n",
       "      <td>8331.8</td>\n",
       "      <td>70821.3</td>\n",
       "      <td>59718.6</td>\n",
       "      <td>87341.6</td>\n",
       "      <td>92722.8</td>\n",
       "      <td>63334.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAMDC</th>\n",
       "      <td>86447.4</td>\n",
       "      <td>54617.7</td>\n",
       "      <td>109416.6</td>\n",
       "      <td>116873.8</td>\n",
       "      <td>155572.5</td>\n",
       "      <td>98071.3</td>\n",
       "      <td>70818.7</td>\n",
       "      <td>83636.2</td>\n",
       "      <td>80560.1</td>\n",
       "      <td>55873.1</td>\n",
       "      <td>...</td>\n",
       "      <td>72638.6</td>\n",
       "      <td>89337.1</td>\n",
       "      <td>109282.4</td>\n",
       "      <td>85325.6</td>\n",
       "      <td>99538.3</td>\n",
       "      <td>81063.0</td>\n",
       "      <td>57490.7</td>\n",
       "      <td>69688.8</td>\n",
       "      <td>78092.4</td>\n",
       "      <td>98849.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 60 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             BR_BT549  BR_HS578T   BR_MCF7  BR_MDAMB231  BR_MDAMB468  BR_T47D  \\\n",
       "gene symbol                                                                     \n",
       "AAAS           8331.8    64198.0   62750.7      63921.5      77924.6  73633.1   \n",
       "AAMDC         86447.4    54617.7  109416.6     116873.8     155572.5  98071.3   \n",
       "\n",
       "             CNS_SF268  CNS_SF295  CNS_SF539  CNS_SNB19  ...  PR_DU145  \\\n",
       "gene symbol                                              ...             \n",
       "AAAS           87421.6    76357.9    84516.4    68816.6  ...    8331.8   \n",
       "AAMDC          70818.7    83636.2    80560.1    55873.1  ...   72638.6   \n",
       "\n",
       "               PR_PC3   RE_7860  RE_A498  RE_ACHN  RE_CAKI1  RE_RXF393  \\\n",
       "gene symbol                                                              \n",
       "AAAS         118359.2   54281.8  50204.8   8331.8   70821.3    59718.6   \n",
       "AAMDC         89337.1  109282.4  85325.6  99538.3   81063.0    57490.7   \n",
       "\n",
       "             RE_SN12C  RE_TK10  RE_UO31  \n",
       "gene symbol                              \n",
       "AAAS          87341.6  92722.8  63334.2  \n",
       "AAMDC         69688.8  78092.4  98849.5  \n",
       "\n",
       "[2 rows x 60 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nci60_proteomics_processed = standardised_pipeline_utils.process(nci60_proteomics)\n",
    "nci60_proteomics_processed[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "transcriptomic_samples['NCI60'] = nci60_transcriptomics_processed.shape[1]\n",
    "proteomic_samples['NCI60'] = nci60_proteomics_processed.shape[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of common proteins:  3022\n",
      "Number of common samples:  59\n"
     ]
    }
   ],
   "source": [
    "nci60_transcriptomics_processed, nci60_proteomics_processed = standardised_pipeline_utils.match_proteins_samples( \\\n",
    "                                                                            nci60_transcriptomics_processed, \n",
    "                                                                            nci60_proteomics_processed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "common_samples['NCI60'] = nci60_transcriptomics_processed.shape[1]\n",
    "common_genes['NCI60'] = nci60_proteomics_processed.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Spearman Correlation:  0.3658\n"
     ]
    }
   ],
   "source": [
    "correlation_nci60 = standardised_pipeline_utils.correlate_genewise(nci60_transcriptomics_processed,\n",
    "                                                                   nci60_proteomics_processed, 'NCI60')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Pearson Correlation:  0.3981\n"
     ]
    }
   ],
   "source": [
    "correlation_nci60_pearson = standardised_pipeline_utils.correlate_genewise(nci60_transcriptomics_processed,\n",
    "                                                                   nci60_proteomics_processed, 'NCI60', method='pearson')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># Samples in Trancriptomic Data</th>\n",
       "      <th># Samples in Proteomic Data</th>\n",
       "      <th># Common Samples</th>\n",
       "      <th># Common Proteins</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Data</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CCLE</th>\n",
       "      <td>1376</td>\n",
       "      <td>375</td>\n",
       "      <td>369</td>\n",
       "      <td>6987</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GTEx</th>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>7471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCI60</th>\n",
       "      <td>60</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "      <td>3022</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       # Samples in Trancriptomic Data  # Samples in Proteomic Data  \\\n",
       "Data                                                                  \n",
       "CCLE                              1376                          375   \n",
       "GTEx                                32                           32   \n",
       "NCI60                               60                           60   \n",
       "\n",
       "       # Common Samples  # Common Proteins  \n",
       "Data                                        \n",
       "CCLE                369               6987  \n",
       "GTEx                 32               7471  \n",
       "NCI60                59               3022  "
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def dataframe_from_dict(*dict_args):\n",
    "    dataframe = pd.DataFrame.from_dict(dict_args[0], orient='index')\n",
    "    for i in range(1, len(dict_args)):\n",
    "        dataframe = pd.concat([dataframe, pd.DataFrame.from_dict(dict_args[i], orient='index')], axis=1)\n",
    "    dataframe.reset_index(inplace=True)\n",
    "    dataframe.columns=['Data', '# Samples in Trancriptomic Data', '# Samples in Proteomic Data', \n",
    "                       '# Common Samples', '# Common Proteins']\n",
    "    return dataframe.set_index('Data')\n",
    "\n",
    "sample_data = dataframe_from_dict(transcriptomic_samples, proteomic_samples, common_samples, common_genes)\n",
    "sample_data.to_csv(file_samples_info, header = False, mode='a')\n",
    "sample_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCI60</th>\n",
       "      <th>CCLE</th>\n",
       "      <th>GTEx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>0.115409</td>\n",
       "      <td>0.054040</td>\n",
       "      <td>0.070586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAMDC</th>\n",
       "      <td>0.180570</td>\n",
       "      <td>0.293006</td>\n",
       "      <td>0.779245</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          NCI60      CCLE      GTEx\n",
       "AAAS   0.115409  0.054040  0.070586\n",
       "AAMDC  0.180570  0.293006  0.779245"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "correlation_combined = pd.concat([correlation_nci60, correlation_ccle, correlation_gtex], axis=1)\n",
    "correlation_combined.to_csv(file_non_tumour_studies)\n",
    "correlation_combined[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "NCI60    0.40\n",
       "CCLE     0.48\n",
       "GTEx     0.59\n",
       "dtype: float64"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "correlation_combined_pearson = pd.concat([correlation_nci60_pearson, \n",
    "                                          correlation_ccle_pearson, correlation_gtex_pearson], axis=1)\n",
    "round(correlation_combined_pearson.median(), 2)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
