{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Process protein properties"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Input:** Protein Abundance (GTEx), Protein Unique Peptides(GTEx), Protein half-lives and other biochemical properties\n",
    "\n",
    "**Output:** Proteins and their properties"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-info\">\n",
    "<b>Note:</b>      \n",
    "Protein abundance is computed as the mean of the protein abundances across the 32 healthy tissues. Standard pipeline is applied before computing the mean abundances. \n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import math\n",
    "import datetime\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "\n",
    "import standardised_pipeline_utils"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "get_data_path = lambda folders, fname: os.path.normpath(os.environ['DATA_PATH'] + '/' +'/'.join(folders) +'/'+ fname)\n",
    "get_local_data_path = lambda folders, fname: os.path.normpath('../local_data/' + '/'.join(folders) +'/'+ fname)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Input Files\n",
    "file_gtex_abundance =  get_data_path(['jiang_2020'], 'Table_S3.xlsx')\n",
    "file_gtex_experiment_info = get_data_path(['jiang_2020'], 'Table_S5.xlsx')\n",
    "file_protein_properties = get_data_path(['protein_properties', 'zecha_2018'], 'Table_S3.xlsx')\n",
    "\n",
    "# Output File\n",
    "file_processed_properties = get_local_data_path(['processed'], 'protein_properties.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### GTEx Mean Abundance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (12627, 533)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene.id.full</th>\n",
       "      <th>reference</th>\n",
       "      <th>GTEX-12WSD-0526-SM-9KNJH</th>\n",
       "      <th>GTEX-11DXX-1226-SM-9KI3K</th>\n",
       "      <th>GTEX-12WSD-1326-SM-9KMWZ</th>\n",
       "      <th>GTEX-13D11-0226-SM-9KI5R</th>\n",
       "      <th>GTEX-1211K-0226-SM-9KMWV</th>\n",
       "      <th>GTEX-YEC4-1426-SM-9KI3B</th>\n",
       "      <th>GTEX-12WSD-1426-SM-9KMX4</th>\n",
       "      <th>GTEX-12WSD-2626-SM-9KMX7</th>\n",
       "      <th>...</th>\n",
       "      <th>reference.110</th>\n",
       "      <th>GTEX-13D11-2626-SM-9KNJO.1</th>\n",
       "      <th>GTEX-11GSP-0226-SM-9KI3T.1</th>\n",
       "      <th>GTEX-1211K-1726-SM-9KI5K.1</th>\n",
       "      <th>GTEX-13FTW-0526-SM-9KNJC.1</th>\n",
       "      <th>GTEX-11GSP-2126-SM-9KI3Y.1</th>\n",
       "      <th>GTEX-13FTW-1526-SM-9KNJB.1</th>\n",
       "      <th>GTEX-11GSP-1126-SM-9KMWC.2</th>\n",
       "      <th>GTEX-13OW8-0226-SM-9KI4E.1</th>\n",
       "      <th>reference.111</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene.id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000000003</th>\n",
       "      <td>ENSG00000000003.10</td>\n",
       "      <td>15.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>15.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000419</th>\n",
       "      <td>ENSG00000000419.8</td>\n",
       "      <td>15.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>15.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 533 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       gene.id.full  reference  GTEX-12WSD-0526-SM-9KNJH  \\\n",
       "gene.id                                                                    \n",
       "ENSG00000000003  ENSG00000000003.10       15.0                       NaN   \n",
       "ENSG00000000419   ENSG00000000419.8       15.0                       NaN   \n",
       "\n",
       "                 GTEX-11DXX-1226-SM-9KI3K  GTEX-12WSD-1326-SM-9KMWZ  \\\n",
       "gene.id                                                               \n",
       "ENSG00000000003                       NaN                       NaN   \n",
       "ENSG00000000419                       NaN                       NaN   \n",
       "\n",
       "                 GTEX-13D11-0226-SM-9KI5R  GTEX-1211K-0226-SM-9KMWV  \\\n",
       "gene.id                                                               \n",
       "ENSG00000000003                       NaN                       NaN   \n",
       "ENSG00000000419                       NaN                       NaN   \n",
       "\n",
       "                 GTEX-YEC4-1426-SM-9KI3B  GTEX-12WSD-1426-SM-9KMX4  \\\n",
       "gene.id                                                              \n",
       "ENSG00000000003                      NaN                       NaN   \n",
       "ENSG00000000419                      NaN                       NaN   \n",
       "\n",
       "                 GTEX-12WSD-2626-SM-9KMX7  ...  reference.110  \\\n",
       "gene.id                                    ...                  \n",
       "ENSG00000000003                       NaN  ...           15.0   \n",
       "ENSG00000000419                       NaN  ...           15.0   \n",
       "\n",
       "                 GTEX-13D11-2626-SM-9KNJO.1  GTEX-11GSP-0226-SM-9KI3T.1  \\\n",
       "gene.id                                                                   \n",
       "ENSG00000000003                         NaN                         NaN   \n",
       "ENSG00000000419                         NaN                         NaN   \n",
       "\n",
       "                 GTEX-1211K-1726-SM-9KI5K.1  GTEX-13FTW-0526-SM-9KNJC.1  \\\n",
       "gene.id                                                                   \n",
       "ENSG00000000003                         NaN                         NaN   \n",
       "ENSG00000000419                         NaN                         NaN   \n",
       "\n",
       "                 GTEX-11GSP-2126-SM-9KI3Y.1  GTEX-13FTW-1526-SM-9KNJB.1  \\\n",
       "gene.id                                                                   \n",
       "ENSG00000000003                         NaN                         NaN   \n",
       "ENSG00000000419                         NaN                         NaN   \n",
       "\n",
       "                 GTEX-11GSP-1126-SM-9KMWC.2  GTEX-13OW8-0226-SM-9KI4E.1  \\\n",
       "gene.id                                                                   \n",
       "ENSG00000000003                         NaN                         NaN   \n",
       "ENSG00000000419                         NaN                         NaN   \n",
       "\n",
       "                 reference.111  \n",
       "gene.id                         \n",
       "ENSG00000000003           15.0  \n",
       "ENSG00000000419           15.0  \n",
       "\n",
       "[2 rows x 533 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtex_abundance = pd.read_excel(io=file_gtex_abundance, sheet_name=\"C protein normalized abundance\", \n",
    "                               skiprows=3, index_col='gene.id', engine='openpyxl')\n",
    "print(\"Dimensions: \", gtex_abundance.shape)\n",
    "gtex_abundance[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "gtex_id_mapper = pd.read_excel(io=file_gtex_abundance, sheet_name='G protein TS score', \n",
    "                               skiprows=2, engine='openpyxl', converters={'hgnc_symbol': str})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "gtex_abundance.drop(list(gtex_abundance.filter(regex = 'reference')), axis = 1, inplace = True)\n",
    "gtex_abundance.drop(columns=['gene.id.full'], inplace=True)\n",
    "gtex_abundance.index = gtex_abundance.index.map(gtex_id_mapper.set_index('ensembl_id')['hgnc_symbol'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (4725, 420)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GTEx_Mean_Abundance</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene.id</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>1357.680477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>10540.179562</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         GTEx_Mean_Abundance\n",
       "gene.id                     \n",
       "A1BG             1357.680477\n",
       "A2M             10540.179562"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtex_abundance_processed = standardised_pipeline_utils.process(gtex_abundance)\n",
    "gtex_abundance_mean = pd.DataFrame(gtex_abundance_processed.mean(axis=1), \n",
    "                                                         columns=['GTEx_Mean_Abundance'])\n",
    "gtex_abundance_mean[:2]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### GTEx Unique Peptides"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (12627, 62)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl_id</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>hgnc_name</th>\n",
       "      <th>prt_total_pp_cnt_from_56_runs</th>\n",
       "      <th>prt_unique_pp_cnt</th>\n",
       "      <th>prt_confident_indicator_from_full_search</th>\n",
       "      <th>prt_sample_num_out_of_201_after_imputation</th>\n",
       "      <th>prt_tissue_num_out_of_32_after_imputation</th>\n",
       "      <th>prt_sample_median</th>\n",
       "      <th>prt_sample_MAD</th>\n",
       "      <th>...</th>\n",
       "      <th>HPA:Reliability..IH.</th>\n",
       "      <th>HPA:Reliability..Mouse.Brain.</th>\n",
       "      <th>HPA:Reliability..IF.</th>\n",
       "      <th>HPA:Subcellular.location</th>\n",
       "      <th>HPA:Prognostic.p.value</th>\n",
       "      <th>HPA:RNA.cancer.category</th>\n",
       "      <th>HPA:RNA.tissue.category</th>\n",
       "      <th>HPA:RNA.TS</th>\n",
       "      <th>HPA:RNA.TS.TPM</th>\n",
       "      <th>HPA:TPM.max.in.non.specific</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hgnc_symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SCYL3</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>57147.0</td>\n",
       "      <td>SCY1 like pseudokinase 3</td>\n",
       "      <td>8</td>\n",
       "      <td>3</td>\n",
       "      <td>high.conf</td>\n",
       "      <td>55</td>\n",
       "      <td>26</td>\n",
       "      <td>-0.06</td>\n",
       "      <td>0.65</td>\n",
       "      <td>...</td>\n",
       "      <td>Approved</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Uncertain</td>\n",
       "      <td>Nuclear bodies&lt;br&gt;Microtubules</td>\n",
       "      <td>Urothelial cancer:8.85e-4 (favourable)</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>parathyroid gland: 44.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CFH</th>\n",
       "      <td>ENSG00000000971</td>\n",
       "      <td>3075.0</td>\n",
       "      <td>complement factor H</td>\n",
       "      <td>2521</td>\n",
       "      <td>76</td>\n",
       "      <td>high.conf</td>\n",
       "      <td>201</td>\n",
       "      <td>32</td>\n",
       "      <td>-0.68</td>\n",
       "      <td>1.45</td>\n",
       "      <td>...</td>\n",
       "      <td>Supported</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Approved</td>\n",
       "      <td>Vesicles</td>\n",
       "      <td>Renal cancer:1.92e-6 (unfavourable)</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>Tissue enhanced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver: 838.9</td>\n",
       "      <td>gallbladder: 206.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 62 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  ensembl_id  entrez_id                 hgnc_name  \\\n",
       "hgnc_symbol                                                         \n",
       "SCYL3        ENSG00000000457    57147.0  SCY1 like pseudokinase 3   \n",
       "CFH          ENSG00000000971     3075.0       complement factor H   \n",
       "\n",
       "             prt_total_pp_cnt_from_56_runs  prt_unique_pp_cnt  \\\n",
       "hgnc_symbol                                                     \n",
       "SCYL3                                    8                  3   \n",
       "CFH                                   2521                 76   \n",
       "\n",
       "            prt_confident_indicator_from_full_search  \\\n",
       "hgnc_symbol                                            \n",
       "SCYL3                                      high.conf   \n",
       "CFH                                        high.conf   \n",
       "\n",
       "             prt_sample_num_out_of_201_after_imputation  \\\n",
       "hgnc_symbol                                               \n",
       "SCYL3                                                55   \n",
       "CFH                                                 201   \n",
       "\n",
       "             prt_tissue_num_out_of_32_after_imputation  prt_sample_median  \\\n",
       "hgnc_symbol                                                                 \n",
       "SCYL3                                               26              -0.06   \n",
       "CFH                                                 32              -0.68   \n",
       "\n",
       "             prt_sample_MAD  ...  HPA:Reliability..IH.  \\\n",
       "hgnc_symbol                  ...                         \n",
       "SCYL3                  0.65  ...              Approved   \n",
       "CFH                    1.45  ...             Supported   \n",
       "\n",
       "            HPA:Reliability..Mouse.Brain.  HPA:Reliability..IF.  \\\n",
       "hgnc_symbol                                                       \n",
       "SCYL3                                 NaN             Uncertain   \n",
       "CFH                                   NaN              Approved   \n",
       "\n",
       "                   HPA:Subcellular.location  \\\n",
       "hgnc_symbol                                   \n",
       "SCYL3        Nuclear bodies<br>Microtubules   \n",
       "CFH                                Vesicles   \n",
       "\n",
       "                             HPA:Prognostic.p.value HPA:RNA.cancer.category  \\\n",
       "hgnc_symbol                                                                   \n",
       "SCYL3        Urothelial cancer:8.85e-4 (favourable)        Expressed in all   \n",
       "CFH             Renal cancer:1.92e-6 (unfavourable)         Tissue enriched   \n",
       "\n",
       "             HPA:RNA.tissue.category HPA:RNA.TS HPA:RNA.TS.TPM  \\\n",
       "hgnc_symbol                                                      \n",
       "SCYL3               Expressed in all        NaN            NaN   \n",
       "CFH                  Tissue enhanced        NaN   liver: 838.9   \n",
       "\n",
       "            HPA:TPM.max.in.non.specific  \n",
       "hgnc_symbol                              \n",
       "SCYL3           parathyroid gland: 44.0  \n",
       "CFH                  gallbladder: 206.0  \n",
       "\n",
       "[2 rows x 62 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtex_experiment_info = pd.read_excel(io=file_gtex_experiment_info, sheet_name=\"A enrichment comparison\", \n",
    "                                     index_col='hgnc_symbol', skiprows=2)\n",
    "print(\"Dimensions: \", gtex_experiment_info.shape)\n",
    "gtex_experiment_info[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# of duplicated values: 6\n"
     ]
    }
   ],
   "source": [
    "# Select unique peptides count as per the data \n",
    "gtex_unique_peptides = gtex_experiment_info['prt_unique_pp_cnt']\n",
    "print(\"# of duplicated values:\", len(gtex_unique_peptides.index[gtex_unique_peptides.index.duplicated()]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GTEx_Unique_Peptides</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hgnc_symbol</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SCYL3</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CFH</th>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             GTEx_Unique_Peptides\n",
       "hgnc_symbol                      \n",
       "SCYL3                           3\n",
       "CFH                            76"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtex_unique_peptides = gtex_unique_peptides[~gtex_unique_peptides.index.duplicated(keep='first')]\n",
    "gtex_unique_peptides = gtex_unique_peptides.to_frame(name='GTEx_Unique_Peptides')\n",
    "gtex_unique_peptides.drop(index=[index for index in gtex_unique_peptides.index if type(index) is datetime.datetime])\n",
    "gtex_unique_peptides[:2]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Biochemical Properties"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (7203, 61)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ProteinGroup.id</th>\n",
       "      <th>UniProt identifier(s)</th>\n",
       "      <th>Gene name(s)</th>\n",
       "      <th>Protein name(s)</th>\n",
       "      <th>Potential contaminant</th>\n",
       "      <th>Number of proteins in group</th>\n",
       "      <th>Unique peptides</th>\n",
       "      <th>Sequence coverage [%]</th>\n",
       "      <th>Valid values for K</th>\n",
       "      <th>Cell culture replicate values for K</th>\n",
       "      <th>...</th>\n",
       "      <th>GOCC name</th>\n",
       "      <th>KEGG name</th>\n",
       "      <th>UniProt Keywords</th>\n",
       "      <th>Prosite name</th>\n",
       "      <th>Respiratory Chain Member</th>\n",
       "      <th>Subcellular location (HPA)</th>\n",
       "      <th>Mitochondrial (MitoCharta2.0)</th>\n",
       "      <th>Protein class (HPA)</th>\n",
       "      <th>Enzyme class (HPA)</th>\n",
       "      <th>Other sub-protein class (HPA)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3451</td>\n",
       "      <td>Q13685</td>\n",
       "      <td>AAMP</td>\n",
       "      <td>Angio-associated migratory cell protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>41.2</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>cell part;cell surface;cytoplasm;intracellular...</td>\n",
       "      <td></td>\n",
       "      <td>Angiogenesis;Cellmembrane;Completeproteome;Cyt...</td>\n",
       "      <td>WD_REPEATS_1;WD_REPEATS_2;WD_REPEATS_REGION</td>\n",
       "      <td></td>\n",
       "      <td>Microtubules;Cytosol</td>\n",
       "      <td></td>\n",
       "      <td>plasma;pred. intracellular</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4121</td>\n",
       "      <td>Q5JTZ9</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>Alanine--tRNA ligase;mitochondrial</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>22.4</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>cell part;cytoplasmic part;intracellular membr...</td>\n",
       "      <td>Aminoacyl-tRNA biosynthesis</td>\n",
       "      <td>Aminoacyl-tRNAsynthetase;ATP-binding;Cardiomyo...</td>\n",
       "      <td>AA_TRNA_LIGASE_II_ALA</td>\n",
       "      <td></td>\n",
       "      <td>Mitochondria</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>other enzyme;pred. intracellular;potential dru...</td>\n",
       "      <td>ligase</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 61 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   ProteinGroup.id UniProt identifier(s) Gene name(s)  \\\n",
       "0             3451                Q13685         AAMP   \n",
       "1             4121                Q5JTZ9        AARS2   \n",
       "\n",
       "                           Protein name(s) Potential contaminant  \\\n",
       "0  Angio-associated migratory cell protein                   NaN   \n",
       "1       Alanine--tRNA ligase;mitochondrial                   NaN   \n",
       "\n",
       "   Number of proteins in group  Unique peptides  Sequence coverage [%]  \\\n",
       "0                            1               13                   41.2   \n",
       "1                            1               16                   22.4   \n",
       "\n",
       "   Valid values for K  Cell culture replicate values for K  ...  \\\n",
       "0                   8                                    4  ...   \n",
       "1                   8                                    4  ...   \n",
       "\n",
       "                                           GOCC name  \\\n",
       "0  cell part;cell surface;cytoplasm;intracellular...   \n",
       "1  cell part;cytoplasmic part;intracellular membr...   \n",
       "\n",
       "                     KEGG name  \\\n",
       "0                                \n",
       "1  Aminoacyl-tRNA biosynthesis   \n",
       "\n",
       "                                    UniProt Keywords  \\\n",
       "0  Angiogenesis;Cellmembrane;Completeproteome;Cyt...   \n",
       "1  Aminoacyl-tRNAsynthetase;ATP-binding;Cardiomyo...   \n",
       "\n",
       "                                  Prosite name  Respiratory Chain Member  \\\n",
       "0  WD_REPEATS_1;WD_REPEATS_2;WD_REPEATS_REGION                             \n",
       "1                        AA_TRNA_LIGASE_II_ALA                             \n",
       "\n",
       "  Subcellular location (HPA)  Mitochondrial (MitoCharta2.0)  \\\n",
       "0       Microtubules;Cytosol                                  \n",
       "1               Mitochondria                           TRUE   \n",
       "\n",
       "                                 Protein class (HPA)  Enzyme class (HPA)  \\\n",
       "0                         plasma;pred. intracellular                       \n",
       "1  other enzyme;pred. intracellular;potential dru...              ligase   \n",
       "\n",
       "   Other sub-protein class (HPA)  \n",
       "0                                 \n",
       "1                                 \n",
       "\n",
       "[2 rows x 61 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_properties = pd.read_excel(io=file_protein_properties, sheet_name='Properties and functions')\n",
    "print(\"Dimensions: \", protein_properties.shape)\n",
    "protein_properties[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Columns present: \n",
      " ['ProteinGroup.id', 'UniProt identifier(s)', 'Gene name(s)', 'Protein name(s)', 'Potential contaminant', 'Number of proteins in group', 'Unique peptides', 'Sequence coverage [%]', 'Valid values for K', 'Cell culture replicate values for K', 'Protein sequence length', 'Molecular weight [kDa]', 'K [h-1]', 'k [h-1]', 'T50% [h]', 'T1/2 [h]', 'Copies per cell', 'Copies per h&cell', 'Copies per h&cell corrected for cell doubling', 'Copies (Zeiler et al.)', 'Copies (Nagaraj et al.)', 'Coil [%]', 'Helix [%]', 'Sheet [%]', 'GRAVY Score', 'A [%]', 'C [%]', 'D [%]', 'E [%]', 'G [%]', 'F [%]', 'H [%]', 'I [%]', 'K [%]', 'L [%]', 'M [%]', 'N [%]', 'P [%]', 'Q [%]', 'R [%]', 'S [%]', 'T [%]', 'V [%]', 'W [%]', 'Y [%]', 'Melting temperature (Leuenberger et al.)', 'Thermal Stability Category (Leuenberger et al.)', 'AA after iMet', 'CORUM', 'GOBP name', 'GOMF name', 'GOCC name', 'KEGG name', 'UniProt Keywords', 'Prosite name', 'Respiratory Chain Member', 'Subcellular location (HPA)', 'Mitochondrial (MitoCharta2.0)', 'Protein class (HPA)', 'Enzyme class (HPA)', 'Other sub-protein class (HPA)']\n"
     ]
    }
   ],
   "source": [
    "print('Columns present: \\n', [colname for colname in protein_properties.columns if 'Unnamed:' not in colname])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "biochemical_properties = protein_properties[['Gene name(s)', 'Protein sequence length', \n",
    "                                             'Molecular weight [kDa]', 'T1/2 [h]']].copy()\n",
    "biochemical_properties.replace('inf.', math.inf, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Gene name(s) column contains a few rows with multiple gene names separated by ; => exploding the dataframe\n",
    "biochemical_properties = biochemical_properties.set_index(['Protein sequence length', 'Molecular weight [kDa]', \n",
    "                                                           'T1/2 [h]', ]).apply(lambda x: x.str.split(';').explode()) \\\n",
    "                                                                       .reset_index().set_index('Gene name(s)')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Protein sequence length</th>\n",
       "      <th>Molecular weight [kDa]</th>\n",
       "      <th>T1/2 [h]</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene name(s)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AAMP</th>\n",
       "      <td>434</td>\n",
       "      <td>46.75</td>\n",
       "      <td>10.602299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AARS2</th>\n",
       "      <td>985</td>\n",
       "      <td>107.34</td>\n",
       "      <td>42.752163</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Protein sequence length  Molecular weight [kDa]   T1/2 [h]\n",
       "Gene name(s)                                                            \n",
       "AAMP                              434                   46.75  10.602299\n",
       "AARS2                             985                  107.34  42.752163"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biochemical_properties = biochemical_properties.apply(pd.to_numeric)\n",
    "biochemical_properties[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "biochemical_properties = biochemical_properties[~biochemical_properties.index.duplicated(keep='first')]\n",
    "biochemical_properties = biochemical_properties.drop(index=[index for index in biochemical_properties.index if type(index) is datetime.datetime])\n",
    "biochemical_properties.drop(index=np.nan, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Combine the properties and write into a file \n",
    "all_protein_properties = pd.concat([gtex_abundance_mean, gtex_unique_peptides, biochemical_properties], axis=1)\n",
    "all_protein_properties.to_csv(file_processed_properties)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
