{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "0112573e",
   "metadata": {},
   "source": [
    "### Compute transcriptomic reproducibility correlation for studies with replicate transcriptomic  profiles"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a741a9b6",
   "metadata": {},
   "source": [
    "**Input:** Transcriptomics data (RNA-Seq) of the cancer cell lines from:\n",
    "1. <a href=https://www.nature.com/articles/s41586-019-1186-3>Next-generation characterization of the Cancer Cell Line Encyclopedia</a>     \n",
    "2. <a href=https://www.nature.com/articles/nbt.3080>A comprehensive transcriptional portrait of human cancer cell lines</a>     \n",
    "                            \n",
    "**Output:** Gene-wise correlation between replicate mRNA abundances "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "375907e2",
   "metadata": {},
   "source": [
    "#### Import Packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ready-wrong",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np \n",
    "import os \n",
    "from matplotlib_venn import venn2\n",
    "from matplotlib import pyplot as plt\n",
    "\n",
    "%load_ext autoreload\n",
    "%autoreload 1\n",
    "%aimport standardised_pipeline_utils"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "deluxe-treasurer",
   "metadata": {},
   "outputs": [],
   "source": [
    "get_data_path = lambda folders, fname: os.path.normpath(os.environ['DATA_PATH']+'/'+'/'.join(folders) +'/'+ fname)\n",
    "get_local_data_path = lambda folders, fname: os.path.normpath('../local_data/'+'/'+'/'.join(folders) +'/'+ fname)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "collectible-inspection",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Input files\n",
    "file_ccle_transcriptomics = get_data_path(['depmap','20Q4'], 'CCLE_expression.csv')\n",
    "file_ccle_sample_info = get_data_path(['depmap','20Q4'], 'sample_info.csv')\n",
    "file_klijn_rnaseq = get_data_path(['klijn_2015'], 'E-MTAB-2706-query-results.tpms.tsv')\n",
    "file_ghandi_sample_info = get_data_path(['ghandi_2019'], 'Table_S1.xlsx')\n",
    "\n",
    "# Output files\n",
    "file_transcriptomic_reproducibility = get_local_data_path(['processed', 'correlation_replicates'], \n",
    "                                                          'transcriptomic_reproducibility.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "mediterranean-minimum",
   "metadata": {},
   "source": [
    "#### Next-generation characterization of the Cancer Cell Line Encyclopedia"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "treated-dryer",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (1812, 26)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DepMap_ID</th>\n",
       "      <th>cell_line_name</th>\n",
       "      <th>stripped_cell_line_name</th>\n",
       "      <th>CCLE_Name</th>\n",
       "      <th>alias</th>\n",
       "      <th>COSMICID</th>\n",
       "      <th>sex</th>\n",
       "      <th>source</th>\n",
       "      <th>Achilles_n_replicates</th>\n",
       "      <th>cell_line_NNMD</th>\n",
       "      <th>...</th>\n",
       "      <th>primary_or_metastasis</th>\n",
       "      <th>primary_disease</th>\n",
       "      <th>Subtype</th>\n",
       "      <th>age</th>\n",
       "      <th>Sanger_Model_ID</th>\n",
       "      <th>depmap_public_comments</th>\n",
       "      <th>lineage</th>\n",
       "      <th>lineage_subtype</th>\n",
       "      <th>lineage_sub_subtype</th>\n",
       "      <th>lineage_molecular_subtype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000001</td>\n",
       "      <td>NIH:OVCAR-3</td>\n",
       "      <td>NIHOVCAR3</td>\n",
       "      <td>NIHOVCAR3_OVARY</td>\n",
       "      <td>OVCAR3</td>\n",
       "      <td>905933.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>ATCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>Ovarian Cancer</td>\n",
       "      <td>Adenocarcinoma, high grade serous</td>\n",
       "      <td>60.0</td>\n",
       "      <td>SIDM00105</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ovary</td>\n",
       "      <td>ovary_adenocarcinoma</td>\n",
       "      <td>high_grade_serous</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000002</td>\n",
       "      <td>HL-60</td>\n",
       "      <td>HL60</td>\n",
       "      <td>HL60_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>905938.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>ATCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Leukemia</td>\n",
       "      <td>Acute Myelogenous Leukemia (AML), M3 (Promyelo...</td>\n",
       "      <td>35.0</td>\n",
       "      <td>SIDM00829</td>\n",
       "      <td>NaN</td>\n",
       "      <td>blood</td>\n",
       "      <td>AML</td>\n",
       "      <td>M3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    DepMap_ID cell_line_name stripped_cell_line_name  \\\n",
       "0  ACH-000001    NIH:OVCAR-3               NIHOVCAR3   \n",
       "1  ACH-000002          HL-60                    HL60   \n",
       "\n",
       "                                 CCLE_Name   alias  COSMICID     sex source  \\\n",
       "0                          NIHOVCAR3_OVARY  OVCAR3  905933.0  Female   ATCC   \n",
       "1  HL60_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE     NaN  905938.0  Female   ATCC   \n",
       "\n",
       "   Achilles_n_replicates  cell_line_NNMD  ... primary_or_metastasis  \\\n",
       "0                    NaN             NaN  ...            Metastasis   \n",
       "1                    NaN             NaN  ...               Primary   \n",
       "\n",
       "  primary_disease                                            Subtype   age  \\\n",
       "0  Ovarian Cancer                  Adenocarcinoma, high grade serous  60.0   \n",
       "1        Leukemia  Acute Myelogenous Leukemia (AML), M3 (Promyelo...  35.0   \n",
       "\n",
       "   Sanger_Model_ID depmap_public_comments lineage       lineage_subtype  \\\n",
       "0        SIDM00105                    NaN   ovary  ovary_adenocarcinoma   \n",
       "1        SIDM00829                    NaN   blood                   AML   \n",
       "\n",
       "  lineage_sub_subtype  lineage_molecular_subtype  \n",
       "0   high_grade_serous                        NaN  \n",
       "1                  M3                        NaN  \n",
       "\n",
       "[2 rows x 26 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ccle_sample_info = pd.read_csv(file_ccle_sample_info)\n",
    "print(\"Dimensions: \", ccle_sample_info.shape)\n",
    "ccle_sample_info[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "676626c8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (1096, 32)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CCLE_ID</th>\n",
       "      <th>depMapID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Site_Primary</th>\n",
       "      <th>Site_Subtype1</th>\n",
       "      <th>Site_Subtype2</th>\n",
       "      <th>Site_Subtype3</th>\n",
       "      <th>Histology</th>\n",
       "      <th>Hist_Subtype1</th>\n",
       "      <th>Hist_Subtype2</th>\n",
       "      <th>...</th>\n",
       "      <th>Growth.Medium</th>\n",
       "      <th>Supplements</th>\n",
       "      <th>Freezing.Medium</th>\n",
       "      <th>Doubling.Time.from.Vendor</th>\n",
       "      <th>Doubling.Time.Calculated.hrs</th>\n",
       "      <th>type</th>\n",
       "      <th>type_refined</th>\n",
       "      <th>PATHOLOGIST_ANNOTATION</th>\n",
       "      <th>mutRate</th>\n",
       "      <th>tcga_code</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DMS53_LUNG</td>\n",
       "      <td>ACH-000698</td>\n",
       "      <td>DMS 53</td>\n",
       "      <td>lung</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>small_cell_carcinoma</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>Waymouth's +10%FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5%DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>450.0</td>\n",
       "      <td>lung_small_cell</td>\n",
       "      <td>lung_small_cell</td>\n",
       "      <td>Lung:SCLC</td>\n",
       "      <td>157.241638</td>\n",
       "      <td>SCLC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SW1116_LARGE_INTESTINE</td>\n",
       "      <td>ACH-000489</td>\n",
       "      <td>SW1116</td>\n",
       "      <td>large_intestine</td>\n",
       "      <td>colon</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>adenocarcinoma</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>L15+10%FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5%DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>123.2</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>Colorectal:Carcinoma</td>\n",
       "      <td>94.373192</td>\n",
       "      <td>COAD/READ</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  CCLE_ID    depMapID    Name     Site_Primary Site_Subtype1  \\\n",
       "0              DMS53_LUNG  ACH-000698  DMS 53             lung            NS   \n",
       "1  SW1116_LARGE_INTESTINE  ACH-000489  SW1116  large_intestine         colon   \n",
       "\n",
       "  Site_Subtype2 Site_Subtype3  Histology         Hist_Subtype1 Hist_Subtype2  \\\n",
       "0            NS            NS  carcinoma  small_cell_carcinoma            NS   \n",
       "1            NS            NS  carcinoma        adenocarcinoma            NS   \n",
       "\n",
       "   ...       Growth.Medium Supplements Freezing.Medium  \\\n",
       "0  ...  Waymouth's +10%FBS         NaN          5%DMSO   \n",
       "1  ...          L15+10%FBS         NaN          5%DMSO   \n",
       "\n",
       "   Doubling.Time.from.Vendor Doubling.Time.Calculated.hrs             type  \\\n",
       "0                        NaN                        450.0  lung_small_cell   \n",
       "1                        NaN                        123.2       colorectal   \n",
       "\n",
       "      type_refined PATHOLOGIST_ANNOTATION     mutRate  tcga_code  \n",
       "0  lung_small_cell              Lung:SCLC  157.241638       SCLC  \n",
       "1       colorectal   Colorectal:Carcinoma   94.373192  COAD/READ  \n",
       "\n",
       "[2 rows x 32 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ghandi_sample_info = pd.read_excel(file_ghandi_sample_info, sheet_name='Cell Line Annotations')\n",
    "print(\"Dimensions: \", ghandi_sample_info.shape)\n",
    "ghandi_sample_info[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "aggressive-victoria",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (1376, 19182)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TSPAN6 (7105)</th>\n",
       "      <th>TNMD (64102)</th>\n",
       "      <th>DPM1 (8813)</th>\n",
       "      <th>SCYL3 (57147)</th>\n",
       "      <th>C1orf112 (55732)</th>\n",
       "      <th>FGR (2268)</th>\n",
       "      <th>CFH (3075)</th>\n",
       "      <th>FUCA2 (2519)</th>\n",
       "      <th>GCLC (2729)</th>\n",
       "      <th>NFYA (4800)</th>\n",
       "      <th>...</th>\n",
       "      <th>AC090517.4 (54816)</th>\n",
       "      <th>AL160269.1 (11046)</th>\n",
       "      <th>ABCF2-H2BE1 (114483834)</th>\n",
       "      <th>FO393400.1 (149951)</th>\n",
       "      <th>POLR2J3 (548644)</th>\n",
       "      <th>H2BE1 (114483833)</th>\n",
       "      <th>AL445238.1 (647264)</th>\n",
       "      <th>GET1-SH3BGR (106865373)</th>\n",
       "      <th>AC024598.1 (22891)</th>\n",
       "      <th>AC113348.1 (102724657)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ACH-001113</th>\n",
       "      <td>4.990501</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>7.273702</td>\n",
       "      <td>2.765535</td>\n",
       "      <td>4.480265</td>\n",
       "      <td>0.028569</td>\n",
       "      <td>1.269033</td>\n",
       "      <td>3.058316</td>\n",
       "      <td>6.483171</td>\n",
       "      <td>5.05398</td>\n",
       "      <td>...</td>\n",
       "      <td>0.111031</td>\n",
       "      <td>0.15056</td>\n",
       "      <td>1.427606</td>\n",
       "      <td>0.042644</td>\n",
       "      <td>5.781884</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.799087</td>\n",
       "      <td>0.014355</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACH-001289</th>\n",
       "      <td>5.209843</td>\n",
       "      <td>0.545968</td>\n",
       "      <td>7.070604</td>\n",
       "      <td>2.538538</td>\n",
       "      <td>3.510962</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.176323</td>\n",
       "      <td>3.836934</td>\n",
       "      <td>4.200850</td>\n",
       "      <td>3.83289</td>\n",
       "      <td>...</td>\n",
       "      <td>0.310340</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.807355</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.704319</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.464668</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.070389</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 19182 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            TSPAN6 (7105)  TNMD (64102)  DPM1 (8813)  SCYL3 (57147)  \\\n",
       "ACH-001113       4.990501      0.000000     7.273702       2.765535   \n",
       "ACH-001289       5.209843      0.545968     7.070604       2.538538   \n",
       "\n",
       "            C1orf112 (55732)  FGR (2268)  CFH (3075)  FUCA2 (2519)  \\\n",
       "ACH-001113          4.480265    0.028569    1.269033      3.058316   \n",
       "ACH-001289          3.510962    0.000000    0.176323      3.836934   \n",
       "\n",
       "            GCLC (2729)  NFYA (4800)  ...  AC090517.4 (54816)  \\\n",
       "ACH-001113     6.483171      5.05398  ...            0.111031   \n",
       "ACH-001289     4.200850      3.83289  ...            0.310340   \n",
       "\n",
       "            AL160269.1 (11046)  ABCF2-H2BE1 (114483834)  FO393400.1 (149951)  \\\n",
       "ACH-001113             0.15056                 1.427606             0.042644   \n",
       "ACH-001289             0.00000                 0.807355             0.000000   \n",
       "\n",
       "            POLR2J3 (548644)  H2BE1 (114483833)  AL445238.1 (647264)  \\\n",
       "ACH-001113          5.781884                0.0                  0.0   \n",
       "ACH-001289          4.704319                0.0                  0.0   \n",
       "\n",
       "            GET1-SH3BGR (106865373)  AC024598.1 (22891)  \\\n",
       "ACH-001113                 0.799087            0.014355   \n",
       "ACH-001289                 0.464668            0.000000   \n",
       "\n",
       "            AC113348.1 (102724657)  \n",
       "ACH-001113                0.000000  \n",
       "ACH-001289                0.070389  \n",
       "\n",
       "[2 rows x 19182 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ccle_transcriptomics = pd.read_csv(file_ccle_transcriptomics, index_col=0)\n",
    "print(\"Dimensions: \", ccle_transcriptomics.shape)\n",
    "ccle_transcriptomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "hindu-infection",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>LC1SQSF_LUNG</th>\n",
       "      <th>COGAR359_SOFT_TISSUE</th>\n",
       "      <th>COLO794_SKIN</th>\n",
       "      <th>KKU213_BILIARY_TRACT</th>\n",
       "      <th>RT4_URINARY_TRACT</th>\n",
       "      <th>SNU283_LARGE_INTESTINE</th>\n",
       "      <th>NCIH1395_LUNG</th>\n",
       "      <th>DEL_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>SNU1196_BILIARY_TRACT</th>\n",
       "      <th>LC1F_LUNG</th>\n",
       "      <th>...</th>\n",
       "      <th>BL70_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>U343_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>639V_URINARY_TRACT</th>\n",
       "      <th>MON_SOFT_TISSUE</th>\n",
       "      <th>LOXIMVI_SKIN</th>\n",
       "      <th>TOLEDO_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>SSP25_BILIARY_TRACT</th>\n",
       "      <th>ECC2_BILIARY_TRACT</th>\n",
       "      <th>A673_BONE</th>\n",
       "      <th>TT_OESOPHAGUS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TSPAN6</th>\n",
       "      <td>4.990501</td>\n",
       "      <td>5.209843</td>\n",
       "      <td>3.77926</td>\n",
       "      <td>5.726831</td>\n",
       "      <td>7.465648</td>\n",
       "      <td>4.914086</td>\n",
       "      <td>4.032982</td>\n",
       "      <td>0.097611</td>\n",
       "      <td>4.712596</td>\n",
       "      <td>5.101398</td>\n",
       "      <td>...</td>\n",
       "      <td>0.070389</td>\n",
       "      <td>4.69265</td>\n",
       "      <td>5.0268</td>\n",
       "      <td>6.699052</td>\n",
       "      <td>4.173127</td>\n",
       "      <td>0.097611</td>\n",
       "      <td>5.045268</td>\n",
       "      <td>5.805292</td>\n",
       "      <td>4.870858</td>\n",
       "      <td>5.117695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TNMD</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.545968</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.176323</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 1376 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        LC1SQSF_LUNG  COGAR359_SOFT_TISSUE  COLO794_SKIN  \\\n",
       "TSPAN6      4.990501              5.209843       3.77926   \n",
       "TNMD        0.000000              0.545968       0.00000   \n",
       "\n",
       "        KKU213_BILIARY_TRACT  RT4_URINARY_TRACT  SNU283_LARGE_INTESTINE  \\\n",
       "TSPAN6              5.726831           7.465648                4.914086   \n",
       "TNMD                0.000000           0.000000                0.176323   \n",
       "\n",
       "        NCIH1395_LUNG  DEL_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "TSPAN6       4.032982                                0.097611   \n",
       "TNMD         0.000000                                0.000000   \n",
       "\n",
       "        SNU1196_BILIARY_TRACT  LC1F_LUNG  ...  \\\n",
       "TSPAN6               4.712596   5.101398  ...   \n",
       "TNMD                 0.000000   0.000000  ...   \n",
       "\n",
       "        BL70_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  U343_CENTRAL_NERVOUS_SYSTEM  \\\n",
       "TSPAN6                                 0.070389                      4.69265   \n",
       "TNMD                                   0.000000                      0.00000   \n",
       "\n",
       "        639V_URINARY_TRACT  MON_SOFT_TISSUE  LOXIMVI_SKIN  \\\n",
       "TSPAN6              5.0268         6.699052      4.173127   \n",
       "TNMD                0.0000         0.000000      0.000000   \n",
       "\n",
       "        TOLEDO_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  SSP25_BILIARY_TRACT  \\\n",
       "TSPAN6                                   0.097611             5.045268   \n",
       "TNMD                                     0.000000             0.000000   \n",
       "\n",
       "        ECC2_BILIARY_TRACT  A673_BONE  TT_OESOPHAGUS  \n",
       "TSPAN6            5.805292   4.870858       5.117695  \n",
       "TNMD              0.000000   0.000000       0.000000  \n",
       "\n",
       "[2 rows x 1376 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Rename the cell lines given as DepMap ID to CCLE_Name and columns to just contain the hgnc_symbol\n",
    "ccle_transcriptomics = ccle_transcriptomics.rename(index = dict(zip(ccle_sample_info['DepMap_ID'], \n",
    "                                                                    ccle_sample_info['CCLE_Name'])),\n",
    "                                                   columns = lambda x : str(x).split(' ')[0])\n",
    "# Transpose to obtain a matrix similar to other studies\n",
    "ccle_transcriptomics = ccle_transcriptomics.transpose()\n",
    "ccle_transcriptomics[:2]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "mature-biology",
   "metadata": {},
   "source": [
    "#### A comprehensive transcriptional portrait of human cancer cell lines"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "virtual-infrastructure",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (57711, 623)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>adrenal gland, unknown adrenal cancer, HCC1576</th>\n",
       "      <th>blood, B-cell lymphoma, NU-DUL-1</th>\n",
       "      <th>blood, B-cell lymphoma, RC-K8</th>\n",
       "      <th>blood, Hodgkin's lymphoma, Pfeiffer</th>\n",
       "      <th>blood, acute T cell leukemia, Jurkat, Clone E6-1</th>\n",
       "      <th>blood, acute lymphoblastic leukemia, SUP-B15</th>\n",
       "      <th>blood, acute monocytic leukemia, SIG-M5</th>\n",
       "      <th>blood, acute monocytic leukemia, THP-1</th>\n",
       "      <th>blood, acute myeloid leukemia, HNT-34</th>\n",
       "      <th>...</th>\n",
       "      <th>zone of skin, melanoma, SK23</th>\n",
       "      <th>zone of skin, melanoma, UACC-257</th>\n",
       "      <th>zone of skin, melanoma, UCSD-242l</th>\n",
       "      <th>zone of skin, melanoma, WM-115</th>\n",
       "      <th>zone of skin, melanoma, WM-266-4</th>\n",
       "      <th>zone of skin, metastatic melanoma, COLO 849</th>\n",
       "      <th>zone of skin, metastatic melanoma, COLO-783</th>\n",
       "      <th>zone of skin, metastatic melanoma, COLO-818</th>\n",
       "      <th>zone of skin, metastatic melanoma, SK-MEL-1</th>\n",
       "      <th>zone of skin, skin squamous cell carcinoma, A-431</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000000003</th>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>30.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.5</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>40.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000005</th>\n",
       "      <td>TNMD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 623 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Gene Name  adrenal gland, unknown adrenal cancer, HCC1576  \\\n",
       "Gene ID                                                                     \n",
       "ENSG00000000003    TSPAN6                                            30.0   \n",
       "ENSG00000000005      TNMD                                             NaN   \n",
       "\n",
       "                 blood, B-cell lymphoma, NU-DUL-1  \\\n",
       "Gene ID                                             \n",
       "ENSG00000000003                               NaN   \n",
       "ENSG00000000005                               NaN   \n",
       "\n",
       "                 blood, B-cell lymphoma, RC-K8  \\\n",
       "Gene ID                                          \n",
       "ENSG00000000003                            0.3   \n",
       "ENSG00000000005                            NaN   \n",
       "\n",
       "                 blood, Hodgkin's lymphoma, Pfeiffer  \\\n",
       "Gene ID                                                \n",
       "ENSG00000000003                                  0.3   \n",
       "ENSG00000000005                                  NaN   \n",
       "\n",
       "                 blood, acute T cell leukemia, Jurkat, Clone E6-1  \\\n",
       "Gene ID                                                             \n",
       "ENSG00000000003                                               0.5   \n",
       "ENSG00000000005                                               NaN   \n",
       "\n",
       "                 blood, acute lymphoblastic leukemia, SUP-B15  \\\n",
       "Gene ID                                                         \n",
       "ENSG00000000003                                           0.2   \n",
       "ENSG00000000005                                           NaN   \n",
       "\n",
       "                 blood, acute monocytic leukemia, SIG-M5  \\\n",
       "Gene ID                                                    \n",
       "ENSG00000000003                                      0.5   \n",
       "ENSG00000000005                                      NaN   \n",
       "\n",
       "                 blood, acute monocytic leukemia, THP-1  \\\n",
       "Gene ID                                                   \n",
       "ENSG00000000003                                     1.0   \n",
       "ENSG00000000005                                     NaN   \n",
       "\n",
       "                 blood, acute myeloid leukemia, HNT-34  ...  \\\n",
       "Gene ID                                                 ...   \n",
       "ENSG00000000003                                    1.0  ...   \n",
       "ENSG00000000005                                    NaN  ...   \n",
       "\n",
       "                 zone of skin, melanoma, SK23  \\\n",
       "Gene ID                                         \n",
       "ENSG00000000003                          24.0   \n",
       "ENSG00000000005                           NaN   \n",
       "\n",
       "                 zone of skin, melanoma, UACC-257  \\\n",
       "Gene ID                                             \n",
       "ENSG00000000003                              51.0   \n",
       "ENSG00000000005                               NaN   \n",
       "\n",
       "                 zone of skin, melanoma, UCSD-242l  \\\n",
       "Gene ID                                              \n",
       "ENSG00000000003                               11.0   \n",
       "ENSG00000000005                                NaN   \n",
       "\n",
       "                 zone of skin, melanoma, WM-115  \\\n",
       "Gene ID                                           \n",
       "ENSG00000000003                            15.0   \n",
       "ENSG00000000005                             NaN   \n",
       "\n",
       "                 zone of skin, melanoma, WM-266-4  \\\n",
       "Gene ID                                             \n",
       "ENSG00000000003                              16.0   \n",
       "ENSG00000000005                               NaN   \n",
       "\n",
       "                 zone of skin, metastatic melanoma, COLO 849  \\\n",
       "Gene ID                                                        \n",
       "ENSG00000000003                                         20.0   \n",
       "ENSG00000000005                                          NaN   \n",
       "\n",
       "                 zone of skin, metastatic melanoma, COLO-783  \\\n",
       "Gene ID                                                        \n",
       "ENSG00000000003                                         27.0   \n",
       "ENSG00000000005                                          NaN   \n",
       "\n",
       "                 zone of skin, metastatic melanoma, COLO-818  \\\n",
       "Gene ID                                                        \n",
       "ENSG00000000003                                          7.0   \n",
       "ENSG00000000005                                          NaN   \n",
       "\n",
       "                 zone of skin, metastatic melanoma, SK-MEL-1  \\\n",
       "Gene ID                                                        \n",
       "ENSG00000000003                                         26.0   \n",
       "ENSG00000000005                                          NaN   \n",
       "\n",
       "                 zone of skin, skin squamous cell carcinoma, A-431  \n",
       "Gene ID                                                             \n",
       "ENSG00000000003                                               40.0  \n",
       "ENSG00000000005                                                NaN  \n",
       "\n",
       "[2 rows x 623 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "klijn_transcriptomics = pd.read_csv(file_klijn_rnaseq, index_col=0, sep='\\t', skiprows=4)\n",
    "print(\"Dimensions: \", klijn_transcriptomics.shape)\n",
    "klijn_transcriptomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "herbal-effect",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HCC1576</th>\n",
       "      <th>NUDUL1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>RCK8_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>PFEIFFER_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>JURKAT_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>SUPB15_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>SIGM5_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>HNT34_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>KASUMI1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>...</th>\n",
       "      <th>SK23</th>\n",
       "      <th>UACC257_SKIN</th>\n",
       "      <th>UCSD242L</th>\n",
       "      <th>WM115_SKIN</th>\n",
       "      <th>WM2664_SKIN</th>\n",
       "      <th>COLO 849</th>\n",
       "      <th>COLO783_SKIN</th>\n",
       "      <th>COLO818_SKIN</th>\n",
       "      <th>SKMEL1_SKIN</th>\n",
       "      <th>A431_SKIN</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TSPAN6</th>\n",
       "      <td>30.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.5</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.4</td>\n",
       "      <td>...</td>\n",
       "      <td>24.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>40.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TNMD</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 622 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           HCC1576  NUDUL1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "Gene Name                                                       \n",
       "TSPAN6        30.0                                        NaN   \n",
       "TNMD           NaN                                        NaN   \n",
       "\n",
       "           RCK8_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "Gene Name                                            \n",
       "TSPAN6                                         0.3   \n",
       "TNMD                                           NaN   \n",
       "\n",
       "           PFEIFFER_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "Gene Name                                                \n",
       "TSPAN6                                             0.3   \n",
       "TNMD                                               NaN   \n",
       "\n",
       "           JURKAT_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "Gene Name                                              \n",
       "TSPAN6                                           0.5   \n",
       "TNMD                                             NaN   \n",
       "\n",
       "           SUPB15_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "Gene Name                                              \n",
       "TSPAN6                                           0.2   \n",
       "TNMD                                             NaN   \n",
       "\n",
       "           SIGM5_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "Gene Name                                             \n",
       "TSPAN6                                          0.5   \n",
       "TNMD                                            NaN   \n",
       "\n",
       "           THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "Gene Name                                            \n",
       "TSPAN6                                         1.0   \n",
       "TNMD                                           NaN   \n",
       "\n",
       "           HNT34_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "Gene Name                                             \n",
       "TSPAN6                                          1.0   \n",
       "TNMD                                            NaN   \n",
       "\n",
       "           KASUMI1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  ...  SK23  \\\n",
       "Gene Name                                              ...         \n",
       "TSPAN6                                            0.4  ...  24.0   \n",
       "TNMD                                              NaN  ...   NaN   \n",
       "\n",
       "           UACC257_SKIN  UCSD242L  WM115_SKIN  WM2664_SKIN  COLO 849  \\\n",
       "Gene Name                                                              \n",
       "TSPAN6             51.0      11.0        15.0         16.0      20.0   \n",
       "TNMD                NaN       NaN         NaN          NaN       NaN   \n",
       "\n",
       "           COLO783_SKIN  COLO818_SKIN  SKMEL1_SKIN  A431_SKIN  \n",
       "Gene Name                                                      \n",
       "TSPAN6             27.0           7.0         26.0       40.0  \n",
       "TNMD                NaN           NaN          NaN        NaN  \n",
       "\n",
       "[2 rows x 622 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "klijn_transcriptomics = klijn_transcriptomics.dropna(subset=['Gene Name']).set_index('Gene Name')\n",
    "klijn_transcriptomics.columns = [colname.split(',')[2].strip().upper().replace('-','') \\\n",
    "                                  for colname in klijn_transcriptomics.columns]\n",
    "klijn_transcriptomics = klijn_transcriptomics.rename(columns = dict(zip(ccle_sample_info['stripped_cell_line_name'], \n",
    "                                                                    ccle_sample_info['CCLE_Name'])))\n",
    "klijn_transcriptomics[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "built-antique",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dimensions:  (19180, 1376)\n",
      "Dimensions:  (19492, 622)\n"
     ]
    }
   ],
   "source": [
    "ccle_transcriptomics_processed = standardised_pipeline_utils.process(ccle_transcriptomics)\n",
    "klijn_transcriptomics_processed = standardised_pipeline_utils.process(klijn_transcriptomics)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "radical-ebony",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(19180, 1022)"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# restrict the cell lines to the one published in the paper only, \n",
    "# as depmap portal can have cell lines from other sources too \n",
    "ccle_transcriptomics_processed = ccle_transcriptomics_processed[np.intersect1d(ghandi_sample_info['CCLE_ID'], \n",
    "                                                                               ccle_transcriptomics_processed.columns)]\n",
    "ccle_transcriptomics_processed.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "0678d341",
   "metadata": {},
   "outputs": [],
   "source": [
    "figure, axes = plt.subplots(1,1)\n",
    "plot = venn2([set(ccle_transcriptomics_processed.columns), set(klijn_transcriptomics_processed.columns)], ax=axes, \n",
    "             set_colors=['#ddaa33', '#004488'], set_labels=['CCLE', 'Klijn'])\n",
    "for text in plot.set_labels:\n",
    "    text.set_fontsize(14)\n",
    "for text in plot.subset_labels:\n",
    "    text.set_fontsize(14)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "random-drain",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of common proteins:  13226\n",
      "Number of common samples:  382\n"
     ]
    }
   ],
   "source": [
    "ccle_subset_processed, klijn_subset_processed = standardised_pipeline_utils.match_proteins_samples( \\\n",
    "                                                                            ccle_transcriptomics_processed, \n",
    "                                                                            klijn_transcriptomics_processed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "liked-lithuania",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Median Spearman Correlation:  0.7536\n"
     ]
    }
   ],
   "source": [
    "correlation_subset_transcriptomic_klijn = standardised_pipeline_utils.correlate_genewise(ccle_subset_processed, \n",
    "                                                                            klijn_subset_processed, 'CCLE-Klijn')\n",
    "correlation_subset_transcriptomic_klijn = correlation_subset_transcriptomic_klijn['CCLE-Klijn']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6551d1f5",
   "metadata": {},
   "source": [
    "### Computing correlation using random 18 samples of transcriptomic replicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "148faa50",
   "metadata": {},
   "outputs": [],
   "source": [
    "np.random.seed(10) # specify seed to ensure reproducibility of the analysis\n",
    "def random_sample_correlation(index):\n",
    "    random_sample_columns = ccle_subset_processed.sample(n=18, replace=False, axis=1).columns \n",
    "    ccle_transcriptomics_random = ccle_transcriptomics_processed[random_sample_columns]\n",
    "    klijn_transcriptomics_random = klijn_transcriptomics_processed[random_sample_columns]\n",
    "    print(\"Iteration \", index, ': ', end = '')\n",
    "    return(standardised_pipeline_utils.correlate_genewise(ccle_transcriptomics_random, klijn_transcriptomics_random, \n",
    "                                                          'Iteration'+str(index)).dropna())    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "efc7ab75",
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings\n",
    "from  scipy.stats import SpearmanRConstantInputWarning\n",
    "warnings.filterwarnings(action='ignore', category=SpearmanRConstantInputWarning)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "46b49d9e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Iteration  0 : Median Spearman Correlation:  0.7998\n",
      "Iteration  1 : Median Spearman Correlation:  0.7992\n",
      "Iteration  2 : Median Spearman Correlation:  0.7343\n",
      "Iteration  3 : Median Spearman Correlation:  0.7653\n",
      "Iteration  4 : Median Spearman Correlation:  0.7626\n",
      "Iteration  5 : Median Spearman Correlation:  0.7853\n",
      "Iteration  6 : Median Spearman Correlation:  0.6987\n",
      "Iteration  7 : Median Spearman Correlation:  0.7115\n",
      "Iteration  8 : Median Spearman Correlation:  0.7766\n",
      "Iteration  9 : Median Spearman Correlation:  0.7847\n",
      "Iteration  10 : Median Spearman Correlation:  0.7887\n",
      "Iteration  11 : Median Spearman Correlation:  0.7614\n",
      "Iteration  12 : Median Spearman Correlation:  0.7907\n",
      "Iteration  13 : Median Spearman Correlation:  0.7647\n",
      "Iteration  14 : Median Spearman Correlation:  0.7494\n",
      "Iteration  15 : Median Spearman Correlation:  0.7477\n",
      "Iteration  16 : Median Spearman Correlation:  0.7282\n",
      "Iteration  17 : Median Spearman Correlation:  0.7296\n",
      "Iteration  18 : Median Spearman Correlation:  0.7758\n",
      "Iteration  19 : Median Spearman Correlation:  0.7416\n",
      "Iteration  20 : Median Spearman Correlation:  0.7341\n",
      "Iteration  21 : Median Spearman Correlation:  0.7785\n",
      "Iteration  22 : Median Spearman Correlation:  0.7369\n",
      "Iteration  23 : Median Spearman Correlation:  0.6662\n",
      "Iteration  24 : Median Spearman Correlation:  0.7598\n",
      "Iteration  25 : Median Spearman Correlation:  0.72\n",
      "Iteration  26 : Median Spearman Correlation:  0.7142\n",
      "Iteration  27 : Median Spearman Correlation:  0.7417\n",
      "Iteration  28 : Median Spearman Correlation:  0.7727\n",
      "Iteration  29 : Median Spearman Correlation:  0.7419\n",
      "Iteration  30 : Median Spearman Correlation:  0.7526\n",
      "Iteration  31 : Median Spearman Correlation:  0.7486\n",
      "Iteration  32 : Median Spearman Correlation:  0.7238\n",
      "Iteration  33 : Median Spearman Correlation:  0.7217\n",
      "Iteration  34 : Median Spearman Correlation:  0.776\n",
      "Iteration  35 : Median Spearman Correlation:  0.7244\n",
      "Iteration  36 : Median Spearman Correlation:  0.6963\n",
      "Iteration  37 : Median Spearman Correlation:  0.709\n",
      "Iteration  38 : Median Spearman Correlation:  0.7675\n",
      "Iteration  39 : Median Spearman Correlation:  0.7293\n",
      "Iteration  40 : Median Spearman Correlation:  0.7193\n",
      "Iteration  41 : Median Spearman Correlation:  0.7833\n",
      "Iteration  42 : Median Spearman Correlation:  0.7527\n",
      "Iteration  43 : Median Spearman Correlation:  0.7647\n",
      "Iteration  44 : Median Spearman Correlation:  0.7314\n",
      "Iteration  45 : Median Spearman Correlation:  0.7677\n",
      "Iteration  46 : Median Spearman Correlation:  0.7027\n",
      "Iteration  47 : Median Spearman Correlation:  0.7243\n",
      "Iteration  48 : Median Spearman Correlation:  0.7812\n",
      "Iteration  49 : Median Spearman Correlation:  0.7606\n",
      "Iteration  50 : Median Spearman Correlation:  0.7537\n",
      "Iteration  51 : Median Spearman Correlation:  0.7379\n",
      "Iteration  52 : Median Spearman Correlation:  0.7754\n",
      "Iteration  53 : Median Spearman Correlation:  0.7544\n",
      "Iteration  54 : Median Spearman Correlation:  0.7421\n",
      "Iteration  55 : Median Spearman Correlation:  0.764\n",
      "Iteration  56 : Median Spearman Correlation:  0.7703\n",
      "Iteration  57 : Median Spearman Correlation:  0.7252\n",
      "Iteration  58 : Median Spearman Correlation:  0.7442\n",
      "Iteration  59 : Median Spearman Correlation:  0.8246\n",
      "Iteration  60 : Median Spearman Correlation:  0.7545\n",
      "Iteration  61 : Median Spearman Correlation:  0.7144\n",
      "Iteration  62 : Median Spearman Correlation:  0.7667\n",
      "Iteration  63 : Median Spearman Correlation:  0.7585\n",
      "Iteration  64 : Median Spearman Correlation:  0.77\n",
      "Iteration  65 : Median Spearman Correlation:  0.7022\n",
      "Iteration  66 : Median Spearman Correlation:  0.7308\n",
      "Iteration  67 : Median Spearman Correlation:  0.6829\n",
      "Iteration  68 : Median Spearman Correlation:  0.7738\n",
      "Iteration  69 : Median Spearman Correlation:  0.7961\n",
      "Iteration  70 : Median Spearman Correlation:  0.7517\n",
      "Iteration  71 : Median Spearman Correlation:  0.7681\n",
      "Iteration  72 : Median Spearman Correlation:  0.8002\n",
      "Iteration  73 : Median Spearman Correlation:  0.775\n",
      "Iteration  74 : Median Spearman Correlation:  0.79\n",
      "Iteration  75 : Median Spearman Correlation:  0.7159\n",
      "Iteration  76 : Median Spearman Correlation:  0.8023\n",
      "Iteration  77 : Median Spearman Correlation:  0.7723\n",
      "Iteration  78 : Median Spearman Correlation:  0.7255\n",
      "Iteration  79 : Median Spearman Correlation:  0.8475\n",
      "Iteration  80 : Median Spearman Correlation:  0.7841\n",
      "Iteration  81 : Median Spearman Correlation:  0.7902\n",
      "Iteration  82 : Median Spearman Correlation:  0.742\n",
      "Iteration  83 : Median Spearman Correlation:  0.7892\n",
      "Iteration  84 : Median Spearman Correlation:  0.776\n",
      "Iteration  85 : Median Spearman Correlation:  0.7771\n",
      "Iteration  86 : Median Spearman Correlation:  0.7548\n",
      "Iteration  87 : Median Spearman Correlation:  0.718\n",
      "Iteration  88 : Median Spearman Correlation:  0.7145\n",
      "Iteration  89 : Median Spearman Correlation:  0.7601\n",
      "Iteration  90 : Median Spearman Correlation:  0.7543\n",
      "Iteration  91 : Median Spearman Correlation:  0.7754\n",
      "Iteration  92 : Median Spearman Correlation:  0.7562\n",
      "Iteration  93 : Median Spearman Correlation:  0.7218\n",
      "Iteration  94 : Median Spearman Correlation:  0.7481\n",
      "Iteration  95 : Median Spearman Correlation:  0.7841\n",
      "Iteration  96 : Median Spearman Correlation:  0.7487\n",
      "Iteration  97 : Median Spearman Correlation:  0.7455\n",
      "Iteration  98 : Median Spearman Correlation:  0.7841\n",
      "Iteration  99 : Median Spearman Correlation:  0.8054\n"
     ]
    }
   ],
   "source": [
    "# random sampling to compare the predictive power of transcriptomic and proteomic reproducibility \n",
    "# over the same number of samples / cell lines\n",
    "correlation_random_transcriptomic_klijn = [random_sample_correlation(i) for i in range(100)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "90a34e01",
   "metadata": {},
   "outputs": [],
   "source": [
    "correlation_random_transcriptomic_klijn = pd.concat(correlation_random_transcriptomic_klijn, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "3414b0be",
   "metadata": {},
   "outputs": [],
   "source": [
    "writer = pd.ExcelWriter(file_transcriptomic_reproducibility)\n",
    "correlation_subset_transcriptomic_klijn.to_excel(writer, 'Reproducibility')\n",
    "correlation_random_transcriptomic_klijn.to_excel(writer, 'Random Reproducibility')\n",
    "writer.save()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
